{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculations for retrospective cohorts\n",
    "We perform calculations explicitly mentioned in or related to the paper's section titled, \"Evaluating expanded molecular profiling and actionability in two retrospective cohorts\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob as glob\n",
    "\n",
    "redact_germline = True"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load MOAlmanac outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>ASXL1.Amplification</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ASXL1 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     patient_id       feature_string                almanac_bin phial_bin  \\\n",
       "0  KIRP-2K-A9WE  ASXL1.Amplification      Biologically Relevant       NaN   \n",
       "1  KIRP-2K-A9WE  AURKA.Amplification      Putatively Actionable       NaN   \n",
       "2  KIRP-2K-A9WE  COSMIC Signature 12      Biologically Relevant       NaN   \n",
       "3  KIRP-2K-A9WE   COSMIC Signature 4  Investigate Actionability       NaN   \n",
       "4  KIRP-2K-A9WE   TPX2.Amplification      Putatively Actionable       NaN   \n",
       "\n",
       "   sum cohort sensitive_predictive_implication  \\\n",
       "0    1   KIRP                              NaN   \n",
       "1    1   KIRP                      Preclinical   \n",
       "2    1   KIRP                              NaN   \n",
       "3    1   KIRP                      Inferential   \n",
       "4    1   KIRP                              NaN   \n",
       "\n",
       "  resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "0                               NaN                               NaN   \n",
       "1                               NaN                       Preclinical   \n",
       "2                               NaN                               NaN   \n",
       "3                               NaN                               NaN   \n",
       "4                               NaN                       Preclinical   \n",
       "\n",
       "   clinvar          feature_type   feature_str_simple  \n",
       "0      NaN   Somatic copy number            ASXL1 Amp  \n",
       "1      NaN   Somatic copy number            AURKA Amp  \n",
       "2      NaN  Mutational signature  COSMIC Signature 12  \n",
       "3      NaN  Mutational signature   COSMIC Signature 4  \n",
       "4      NaN   Somatic copy number             TPX2 Amp  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('retrospective.actionability.txt', sep='\\t')\n",
    "if redact_germline:\n",
    "    df = df[~df['feature_type'].eq('Germline')]\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        1531\n",
       "Investigate Actionability    1365\n",
       "Putatively Actionable         563\n",
       "Name: almanac_bin, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['almanac_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        924\n",
       "Investigate Actionability    657\n",
       "Putatively Actionable         81\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MEL     1831\n",
       "SU2C    1322\n",
       "KIRP     532\n",
       "OS       344\n",
       "Name: cohort, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['cohort'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How many patients had > 0 events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Almanac\n",
      "Mel 109 0.990909090909091\n",
      "SU2C 149 0.9933333333333333\n",
      "KIRP 85 0.85\n",
      "OS 50 0.847457627118644\n",
      "\n",
      "PHIAL\n",
      "Mel 101 0.9181818181818182\n",
      "SU2C 131 0.8733333333333333\n",
      "KIRP 27 0.27\n",
      "OS 36 0.6101694915254238\n"
     ]
    }
   ],
   "source": [
    "idx_almanac = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_phial = df['phial_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_mel = df['cohort'].eq('MEL')\n",
    "idx_su2c = df['cohort'].eq('SU2C')\n",
    "idx_kirp = df['cohort'].eq('KIRP')\n",
    "idx_os = df['cohort'].eq('OS')\n",
    "\n",
    "n_almanac_mel = df[idx_almanac & idx_mel]['patient_id'].drop_duplicates().shape[0]\n",
    "n_almanac_su2c = df[idx_almanac & idx_su2c]['patient_id'].drop_duplicates().shape[0]\n",
    "n_almanac_kirp = df[idx_almanac & idx_kirp]['patient_id'].drop_duplicates().shape[0]\n",
    "n_almanac_os = df[idx_almanac & idx_os]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_mel = df[idx_phial & idx_mel]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_su2c = df[idx_phial & idx_su2c]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_kirp = df[idx_phial & idx_kirp]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_os = df[idx_phial & idx_os]['patient_id'].drop_duplicates().shape[0]\n",
    "\n",
    "print('Almanac')\n",
    "print('Mel', n_almanac_mel, n_almanac_mel/110)\n",
    "print('SU2C', n_almanac_su2c, n_almanac_su2c/150)\n",
    "print('KIRP', n_almanac_kirp, n_almanac_kirp/100)\n",
    "print('OS', n_almanac_os, n_almanac_os/59)\n",
    "print('')\n",
    "print('PHIAL')\n",
    "print('Mel', n_phial_mel, n_phial_mel/110)\n",
    "print('SU2C', n_phial_su2c, n_phial_su2c/150)\n",
    "print('KIRP', n_phial_kirp, n_phial_kirp/100)\n",
    "print('OS', n_phial_os, n_phial_os/59)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How many somatic variants and copy number alterations were called"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1445, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>DNMT3A.p.Q350fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     patient_id       feature_string                almanac_bin  \\\n",
       "1  KIRP-2K-A9WE  AURKA.Amplification      Putatively Actionable   \n",
       "4  KIRP-2K-A9WE   TPX2.Amplification      Putatively Actionable   \n",
       "5  KIRP-2Z-A9J1   BRAF.Amplification      Putatively Actionable   \n",
       "7  KIRP-2Z-A9J1      DNMT3A.p.Q350fs  Investigate Actionability   \n",
       "9  KIRP-2Z-A9J1    MET.Amplification      Putatively Actionable   \n",
       "\n",
       "               phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "1                    NaN    1   KIRP                      Preclinical   \n",
       "4                    NaN    1   KIRP                              NaN   \n",
       "5                    NaN    1   KIRP                              NaN   \n",
       "7  Biologically Relevant    2   KIRP                              NaN   \n",
       "9                    NaN    1   KIRP                        Guideline   \n",
       "\n",
       "  resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "1                               NaN                       Preclinical   \n",
       "4                               NaN                       Preclinical   \n",
       "5                 Clinical evidence                               NaN   \n",
       "7                               NaN                         Guideline   \n",
       "9                         Guideline                               NaN   \n",
       "\n",
       "   clinvar         feature_type feature_str_simple  \n",
       "1      NaN  Somatic copy number          AURKA Amp  \n",
       "4      NaN  Somatic copy number           TPX2 Amp  \n",
       "5      NaN  Somatic copy number           BRAF Amp  \n",
       "7      NaN      Somatic variant             DNMT3A  \n",
       "9      NaN  Somatic copy number            MET Amp  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx_v = df['feature_type'].eq('Somatic variant')\n",
    "idx_cn = df['feature_type'].eq('Somatic copy number')\n",
    "idx_a = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_b = df['almanac_bin'].eq('Biologically Relevant')\n",
    "\n",
    "idx = (idx_a) & (idx_v | idx_cn)\n",
    "tmp = df[idx]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "453\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Putatively Actionable         79\n",
       "Investigate Actionability    374\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(tmp['phial_bin'].value_counts().loc[['Putatively Actionable', 'Investigate Actionability']].sum())\n",
    "tmp['phial_bin'].value_counts().loc[['Putatively Actionable', 'Investigate Actionability']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_almanac_pa_ia = tmp['almanac_bin'].value_counts().sum()\n",
    "n_almanac_phial_pa_ia = tmp['phial_bin'].value_counts().loc[['Putatively Actionable', 'Investigate Actionability']].sum()\n",
    "fraction_almanac_phial_pa_ia = (n_almanac_phial_pa_ia / n_almanac_pa_ia)*100\n",
    "n_almanac_phial_br = tmp['phial_bin'].value_counts().loc['Biologically Relevant']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Of the 1445 somatic variants and copy number alterations noted by MOAlmanac, 453 (31.349480968858128%) were highlighted for putative actionability or investigate actionability by PHIAL, with an additional 390 noted by PHIAL for biological relevance.'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"Of the {n_almanac_pa_ia} somatic variants and copy number alterations noted by MOAlmanac, \\\n",
    "{n_almanac_phial_pa_ia} ({fraction_almanac_phial_pa_ia}%) were highlighted for putative actionability or \\\n",
    "investigate actionability by PHIAL, with an additional {n_almanac_phial_br} noted by PHIAL for biological relevance.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Restricting considered features\n",
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: FDA-approved, clinical trials, or clinical evidence  \n",
    "Matches considered: First putatively actionable and then also investigate actionability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(412, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>KIRP-2Z-A9JD</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>KIRP-2Z-A9JD</td>\n",
       "      <td>EGFR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>EGFR Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>KIRP-2Z-A9JD</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4012</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>PIK3CA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>PIK3CA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4013</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>PTEN.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>PTEN Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4014</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>RB1.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>RB1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4018</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>TP53.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TP53 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4021</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AR Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id        feature_string            almanac_bin  \\\n",
       "5     KIRP-2Z-A9J1    BRAF.Amplification  Putatively Actionable   \n",
       "9     KIRP-2Z-A9J1     MET.Amplification  Putatively Actionable   \n",
       "36    KIRP-2Z-A9JD    BRAF.Amplification  Putatively Actionable   \n",
       "39    KIRP-2Z-A9JD    EGFR.Amplification  Putatively Actionable   \n",
       "42    KIRP-2Z-A9JD     MET.Amplification  Putatively Actionable   \n",
       "...            ...                   ...                    ...   \n",
       "4012       TP_2061  PIK3CA.Amplification  Putatively Actionable   \n",
       "4013       TP_2061         PTEN.Deletion  Putatively Actionable   \n",
       "4014       TP_2061          RB1.Deletion  Putatively Actionable   \n",
       "4018       TP_2061         TP53.Deletion  Putatively Actionable   \n",
       "4021       TP_2064      AR.Amplification  Putatively Actionable   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "5                           NaN    1   KIRP                              NaN   \n",
       "9                           NaN    1   KIRP                        Guideline   \n",
       "36                          NaN    1   KIRP                              NaN   \n",
       "39                          NaN    1   KIRP                Clinical evidence   \n",
       "42                          NaN    1   KIRP                        Guideline   \n",
       "...                         ...  ...    ...                              ...   \n",
       "4012  Investigate Actionability    2   SU2C                   Clinical trial   \n",
       "4013  Investigate Actionability    2   SU2C                      Preclinical   \n",
       "4014  Investigate Actionability    2   SU2C                              NaN   \n",
       "4018  Investigate Actionability    2   SU2C                              NaN   \n",
       "4021  Investigate Actionability    2   SU2C                        Guideline   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "5                    Clinical evidence                               NaN   \n",
       "9                            Guideline                               NaN   \n",
       "36                   Clinical evidence                               NaN   \n",
       "39                                 NaN                               NaN   \n",
       "42                           Guideline                               NaN   \n",
       "...                                ...                               ...   \n",
       "4012                               NaN                               NaN   \n",
       "4013                 Clinical evidence                               NaN   \n",
       "4014                               NaN                         Guideline   \n",
       "4018                               NaN                         Guideline   \n",
       "4021                         Guideline                               NaN   \n",
       "\n",
       "      clinvar         feature_type feature_str_simple  \n",
       "5         NaN  Somatic copy number           BRAF Amp  \n",
       "9         NaN  Somatic copy number            MET Amp  \n",
       "36        NaN  Somatic copy number           BRAF Amp  \n",
       "39        NaN  Somatic copy number           EGFR Amp  \n",
       "42        NaN  Somatic copy number            MET Amp  \n",
       "...       ...                  ...                ...  \n",
       "4012      NaN  Somatic copy number         PIK3CA Amp  \n",
       "4013      NaN  Somatic copy number           PTEN Del  \n",
       "4014      NaN  Somatic copy number            RB1 Del  \n",
       "4018      NaN  Somatic copy number           TP53 Del  \n",
       "4021      NaN  Somatic copy number             AR Amp  \n",
       "\n",
       "[412 rows x 12 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "pa_patients = tmp['patient_id'].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Investigate Actionability    167\n",
       "Putatively Actionable         60\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_events = tmp.shape[0]\n",
    "n_patients = pa_patients.shape[0]\n",
    "n_patients_mel = tmp[tmp['cohort'].eq('MEL')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_patients_su2c = tmp[tmp['cohort'].eq('SU2C')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_patients_kirp = tmp[tmp['cohort'].eq('KIRP')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_patients_os = tmp[tmp['cohort'].eq('OS')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial = tmp['phial_bin'].dropna().shape[0]\n",
    "frac = round((n_phial / n_events)*100, 2)\n",
    "tmp['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MOAlmanac identified 412 such putatively actionable events from 253 (73 melanoma, 118 mCRPC, 37 KIRP, 25 OS), 227 (55.1%) of which were flagged by PHIAL for clinical relevance'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"MOAlmanac identified {n_events} such putatively actionable events from {n_patients} ({n_patients_mel} melanoma, \\\n",
    "{n_patients_su2c} mCRPC, {n_patients_kirp} KIRP, {n_patients_os} OS), {n_phial} ({frac}%) of which were flagged by PHIAL for clinical relevance\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP\n",
      "MET.Amplification       18\n",
      "EGFR.Amplification      16\n",
      "BRAF.Amplification      16\n",
      "ERBB2.Amplification      5\n",
      "PIK3CA.Amplification     3\n",
      "Name: feature_string, dtype: int64\n",
      "\n",
      "MEL\n",
      "BRAF.p.V600E         39\n",
      "PTEN.Deletion         9\n",
      "MET.Amplification     9\n",
      "NRAS.p.Q61R           8\n",
      "NRAS.p.Q61L           7\n",
      "Name: feature_string, dtype: int64\n",
      "\n",
      "OS\n",
      "RB1.Deletion          12\n",
      "TP53.Deletion          8\n",
      "MYC.Amplification      5\n",
      "PTEN.Deletion          3\n",
      "CDK4.Amplification     3\n",
      "Name: feature_string, dtype: int64\n",
      "\n",
      "SU2C\n",
      "AR.Amplification     82\n",
      "PTEN.Deletion        40\n",
      "RB1.Deletion         21\n",
      "MYC.Amplification    14\n",
      "TP53.Deletion        10\n",
      "Name: feature_string, dtype: int64\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for label, group in tmp.groupby('cohort'):\n",
    "    print(label)\n",
    "    print(group['feature_string'].value_counts().head())\n",
    "    print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1180, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>DNMT3A.p.Q350fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>KIRP-2Z-A9J2</td>\n",
       "      <td>ARID1A.p.CP1657fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>ARID1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>KIRP-2Z-A9J2</td>\n",
       "      <td>BCR.p.E306D</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>BCR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4014</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>RB1.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>RB1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4018</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>TP53.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TP53 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4021</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AR Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4023</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>BLM.p.V54I</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>BLM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4027</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>TP53.p.H179R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1180 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id      feature_string                almanac_bin  \\\n",
       "5     KIRP-2Z-A9J1  BRAF.Amplification      Putatively Actionable   \n",
       "7     KIRP-2Z-A9J1     DNMT3A.p.Q350fs  Investigate Actionability   \n",
       "9     KIRP-2Z-A9J1   MET.Amplification      Putatively Actionable   \n",
       "12    KIRP-2Z-A9J2   ARID1A.p.CP1657fs  Investigate Actionability   \n",
       "14    KIRP-2Z-A9J2         BCR.p.E306D  Investigate Actionability   \n",
       "...            ...                 ...                        ...   \n",
       "4014       TP_2061        RB1.Deletion      Putatively Actionable   \n",
       "4018       TP_2061       TP53.Deletion      Putatively Actionable   \n",
       "4021       TP_2064    AR.Amplification      Putatively Actionable   \n",
       "4023       TP_2064          BLM.p.V54I  Investigate Actionability   \n",
       "4027       TP_2064        TP53.p.H179R  Investigate Actionability   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "5                           NaN    1   KIRP                              NaN   \n",
       "7         Biologically Relevant    2   KIRP                              NaN   \n",
       "9                           NaN    1   KIRP                        Guideline   \n",
       "12                          NaN    1   KIRP                      Preclinical   \n",
       "14                          NaN    1   KIRP                              NaN   \n",
       "...                         ...  ...    ...                              ...   \n",
       "4014  Investigate Actionability    2   SU2C                              NaN   \n",
       "4018  Investigate Actionability    2   SU2C                              NaN   \n",
       "4021  Investigate Actionability    2   SU2C                        Guideline   \n",
       "4023                        NaN    1   SU2C                              NaN   \n",
       "4027  Investigate Actionability    2   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "5                    Clinical evidence                               NaN   \n",
       "7                                  NaN                         Guideline   \n",
       "9                            Guideline                               NaN   \n",
       "12                                 NaN                 Clinical evidence   \n",
       "14                   Clinical evidence                               NaN   \n",
       "...                                ...                               ...   \n",
       "4014                               NaN                         Guideline   \n",
       "4018                               NaN                         Guideline   \n",
       "4021                         Guideline                               NaN   \n",
       "4023                               NaN                 Clinical evidence   \n",
       "4027                         Guideline                         Guideline   \n",
       "\n",
       "      clinvar         feature_type feature_str_simple  \n",
       "5         NaN  Somatic copy number           BRAF Amp  \n",
       "7         NaN      Somatic variant             DNMT3A  \n",
       "9         NaN  Somatic copy number            MET Amp  \n",
       "12        NaN      Somatic variant             ARID1A  \n",
       "14        NaN      Somatic variant                BCR  \n",
       "...       ...                  ...                ...  \n",
       "4014      NaN  Somatic copy number            RB1 Del  \n",
       "4018      NaN  Somatic copy number           TP53 Del  \n",
       "4021      NaN  Somatic copy number             AR Amp  \n",
       "4023      NaN      Somatic variant                BLM  \n",
       "4027      NaN      Somatic variant               TP53  \n",
       "\n",
       "[1180 rows x 12 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "93"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ia_patients = tmp['patient_id'].drop_duplicates()\n",
    "diff_patients = pd.Index(ia_patients).difference(pd.Index(pa_patients))\n",
    "len(diff_patients)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.222"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_n_patients = 110 + 150 + 100 + 59\n",
    "round(len(diff_patients) / total_n_patients, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ESR1.Deletion      11\n",
       "NRAS.p.Q61K        10\n",
       "BRAF.p.E611*        5\n",
       "BRAF.p.D594Y        5\n",
       "TP53.p.R175H        4\n",
       "                   ..\n",
       "PRPF8.p.E2322K      1\n",
       "TP53.p.G266E        1\n",
       "TP53.p.C238Y        1\n",
       "ASXL1.p.L1170fs     1\n",
       "RET.p.E480K         1\n",
       "Name: feature_string, Length: 693, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['almanac_bin'].eq('Investigate Actionability')]['feature_string'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_string'].eq('NRAS.p.Q61K')]['patient_id'].drop_duplicates().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(43,)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[~tmp['phial_bin'].isnull() & tmp['patient_id'].isin(diff_patients) & tmp['phial_bin'].isin(['Putatively Actionable', 'Investigate Actionability'])]['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How many events did PHIAL call that Almanac didn't?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda3/envs/moalmanac-paper/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "False    225\n",
       "Name: gene, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "target = pd.read_excel('../knowledge-bases/target/almanac-comparison.xlsx')\n",
    "tmp = df[df['phial_bin'].isin(['Putatively Actionable', 'Investigate Actionability']) & df['almanac_bin'].isnull()]\n",
    "tmp['gene'] = tmp['feature_string'].str.split('.').apply(lambda x: x[0])\n",
    "tmp['gene'].isin(target[target['included_in_lift_to_moalmanac'].eq(0)]).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Investigate Actionability    223\n",
       "Putatively Actionable          2\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2646</th>\n",
       "      <td>MO_1232</td>\n",
       "      <td>KRAS.p.Q61H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2647</th>\n",
       "      <td>MO_1232</td>\n",
       "      <td>KRAS.p.Q61L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     patient_id feature_string almanac_bin              phial_bin  sum cohort  \\\n",
       "2646    MO_1232    KRAS.p.Q61H         NaN  Putatively Actionable    1   SU2C   \n",
       "2647    MO_1232    KRAS.p.Q61L         NaN  Putatively Actionable    1   SU2C   \n",
       "\n",
       "     sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "2646                              NaN                               NaN   \n",
       "2647                              NaN                               NaN   \n",
       "\n",
       "     prognostic_predictive_implication  clinvar     feature_type  \\\n",
       "2646                               NaN      NaN  Somatic variant   \n",
       "2647                               NaN      NaN  Somatic variant   \n",
       "\n",
       "     feature_str_simple  gene  \n",
       "2646               KRAS  KRAS  \n",
       "2647               KRAS  KRAS  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['phial_bin'].eq('Putatively Actionable')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Expanding evidence\n",
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: Preclinical  \n",
    "Matches considered: Putatively actionable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(164, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>KIRP-2Z-A9JG</td>\n",
       "      <td>MDM2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MDM2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>KIRP-2Z-A9JP</td>\n",
       "      <td>FGFR1.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>FGFR1 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>KIRP-2Z-A9JQ</td>\n",
       "      <td>KEAP1.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>KEAP1 Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      patient_id       feature_string            almanac_bin phial_bin  sum  \\\n",
       "1   KIRP-2K-A9WE  AURKA.Amplification  Putatively Actionable       NaN    1   \n",
       "4   KIRP-2K-A9WE   TPX2.Amplification  Putatively Actionable       NaN    1   \n",
       "56  KIRP-2Z-A9JG   MDM2.Amplification  Putatively Actionable       NaN    1   \n",
       "84  KIRP-2Z-A9JP  FGFR1.Amplification  Putatively Actionable       NaN    1   \n",
       "97  KIRP-2Z-A9JQ       KEAP1.Deletion  Putatively Actionable       NaN    1   \n",
       "\n",
       "   cohort sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "1    KIRP                      Preclinical                               NaN   \n",
       "4    KIRP                              NaN                               NaN   \n",
       "56   KIRP                      Preclinical                               NaN   \n",
       "84   KIRP                      Preclinical                               NaN   \n",
       "97   KIRP                              NaN                       Preclinical   \n",
       "\n",
       "   prognostic_predictive_implication  clinvar         feature_type  \\\n",
       "1                        Preclinical      NaN  Somatic copy number   \n",
       "4                        Preclinical      NaN  Somatic copy number   \n",
       "56                               NaN      NaN  Somatic copy number   \n",
       "84                               NaN      NaN  Somatic copy number   \n",
       "97                               NaN      NaN  Somatic copy number   \n",
       "\n",
       "   feature_str_simple  \n",
       "1           AURKA Amp  \n",
       "4            TPX2 Amp  \n",
       "56           MDM2 Amp  \n",
       "84          FGFR1 Amp  \n",
       "97          KEAP1 Del  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['Preclinical']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_events = tmp.shape[0]\n",
    "n_patients = tmp['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial = tmp['phial_bin'].dropna().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'On the basis of preclinical evidence, 164 such events from 140 patients were identified-- --91 (55.49%) of which were also highlighted by PHIAL.'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"On the basis of preclinical evidence, {n_events} such events \\\n",
    "from {n_patients} patients were identified-- --{n_phial} ({round(100*(n_phial / n_events), 2)}%) \\\n",
    "of which were also highlighted by PHIAL.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: Inferential  \n",
    "Matches considered: Putatively actionable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(24, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>KIRP-4A-A93Y</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>KIRP-A4-8630</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>KIRP-AL-3466</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>KIRP-B9-4117</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>KIRP-B9-5155</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id       feature_string            almanac_bin phial_bin  sum  \\\n",
       "136  KIRP-4A-A93Y  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "308  KIRP-A4-8630  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "358  KIRP-AL-3466  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "470  KIRP-B9-4117  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "483  KIRP-B9-5155  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "\n",
       "    cohort sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "136   KIRP                     FDA-Approved                       Preclinical   \n",
       "308   KIRP                     FDA-Approved                       Preclinical   \n",
       "358   KIRP                     FDA-Approved                       Preclinical   \n",
       "470   KIRP                     FDA-Approved                       Preclinical   \n",
       "483   KIRP                     FDA-Approved                       Preclinical   \n",
       "\n",
       "    prognostic_predictive_implication  clinvar         feature_type  \\\n",
       "136                       Inferential      NaN  Somatic copy number   \n",
       "308                       Inferential      NaN  Somatic copy number   \n",
       "358                       Inferential      NaN  Somatic copy number   \n",
       "470                       Inferential      NaN  Somatic copy number   \n",
       "483                       Inferential      NaN  Somatic copy number   \n",
       "\n",
       "    feature_str_simple  \n",
       "136          ERBB2 Amp  \n",
       "308          ERBB2 Amp  \n",
       "358          ERBB2 Amp  \n",
       "470          ERBB2 Amp  \n",
       "483          ERBB2 Amp  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['Inferential']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CCND1.Amplification    15\n",
       "ERBB2.Amplification     5\n",
       "CDKN2C.Deletion         3\n",
       "CD274.Amplification     1\n",
       "Name: feature_string, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['feature_string'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Inferential evidence 24 additional putatively actionable copy number alterations from 24 patients, most prominently CCND1 amplifications for reported sensitivity to Palbocicilib.'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_events = tmp.shape[0]\n",
    "n_patients = tmp['patient_id'].drop_duplicates().shape[0]\n",
    "f\"Inferential evidence {n_events} additional putatively actionable copy number alterations from {n_patients} patients, \\\n",
    "most prominently CCND1 amplifications for reported sensitivity to Palbocicilib.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## In total for somatic variants and copy number alterations\n",
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: FDA-approved, clinical trials, clinical evidence, preclinical, inferential  \n",
    "Matches considered: Putatively actionable, Investigate actionability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1445, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>DNMT3A.p.Q350fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4021</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AR Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4022</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4023</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>BLM.p.V54I</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>BLM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4026</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>MPL.p.A134T</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>MPL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4027</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>TP53.p.H179R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1445 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id       feature_string                almanac_bin  \\\n",
       "1     KIRP-2K-A9WE  AURKA.Amplification      Putatively Actionable   \n",
       "4     KIRP-2K-A9WE   TPX2.Amplification      Putatively Actionable   \n",
       "5     KIRP-2Z-A9J1   BRAF.Amplification      Putatively Actionable   \n",
       "7     KIRP-2Z-A9J1      DNMT3A.p.Q350fs  Investigate Actionability   \n",
       "9     KIRP-2Z-A9J1    MET.Amplification      Putatively Actionable   \n",
       "...            ...                  ...                        ...   \n",
       "4021       TP_2064     AR.Amplification      Putatively Actionable   \n",
       "4022       TP_2064  AURKA.Amplification      Putatively Actionable   \n",
       "4023       TP_2064           BLM.p.V54I  Investigate Actionability   \n",
       "4026       TP_2064          MPL.p.A134T  Investigate Actionability   \n",
       "4027       TP_2064         TP53.p.H179R  Investigate Actionability   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "1                           NaN    1   KIRP                      Preclinical   \n",
       "4                           NaN    1   KIRP                              NaN   \n",
       "5                           NaN    1   KIRP                              NaN   \n",
       "7         Biologically Relevant    2   KIRP                              NaN   \n",
       "9                           NaN    1   KIRP                        Guideline   \n",
       "...                         ...  ...    ...                              ...   \n",
       "4021  Investigate Actionability    2   SU2C                        Guideline   \n",
       "4022                        NaN    1   SU2C                      Preclinical   \n",
       "4023                        NaN    1   SU2C                              NaN   \n",
       "4026      Biologically Relevant    2   SU2C                      Preclinical   \n",
       "4027  Investigate Actionability    2   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "1                                  NaN                       Preclinical   \n",
       "4                                  NaN                       Preclinical   \n",
       "5                    Clinical evidence                               NaN   \n",
       "7                                  NaN                         Guideline   \n",
       "9                            Guideline                               NaN   \n",
       "...                                ...                               ...   \n",
       "4021                         Guideline                               NaN   \n",
       "4022                               NaN                       Preclinical   \n",
       "4023                               NaN                 Clinical evidence   \n",
       "4026                               NaN                               NaN   \n",
       "4027                         Guideline                         Guideline   \n",
       "\n",
       "      clinvar         feature_type feature_str_simple  \n",
       "1         NaN  Somatic copy number          AURKA Amp  \n",
       "4         NaN  Somatic copy number           TPX2 Amp  \n",
       "5         NaN  Somatic copy number           BRAF Amp  \n",
       "7         NaN      Somatic variant             DNMT3A  \n",
       "9         NaN  Somatic copy number            MET Amp  \n",
       "...       ...                  ...                ...  \n",
       "4021      NaN  Somatic copy number             AR Amp  \n",
       "4022      NaN  Somatic copy number          AURKA Amp  \n",
       "4023      NaN      Somatic variant                BLM  \n",
       "4026      NaN      Somatic variant                MPL  \n",
       "4027      NaN      Somatic variant               TP53  \n",
       "\n",
       "[1445 rows x 12 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence', 'Preclinical', 'Inferential']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP 72\n",
      "MEL 109\n",
      "OS 42\n",
      "SU2C 142\n",
      "365\n"
     ]
    }
   ],
   "source": [
    "total_n = 0\n",
    "for label, group in tmp.groupby('cohort'):\n",
    "    total_n += group['patient_id'].drop_duplicates().shape[0]\n",
    "    print(label, group['patient_id'].drop_duplicates().shape[0])\n",
    "print(total_n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        390\n",
       "Investigate Actionability    374\n",
       "Putatively Actionable         79\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        0.269896\n",
       "Investigate Actionability    0.258824\n",
       "Putatively Actionable        0.054671\n",
       "Name: phial_bin, dtype: float64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['phial_bin'].value_counts().divide(tmp.shape[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validation sequencing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>score_bin</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature</th>\n",
       "      <th>alteration_type</th>\n",
       "      <th>alteration</th>\n",
       "      <th>tumor_f</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>...</th>\n",
       "      <th>prognostic_url</th>\n",
       "      <th>number_germline_mutations_in_gene</th>\n",
       "      <th>validation_total_coverage</th>\n",
       "      <th>validation_tumor_f</th>\n",
       "      <th>validation_detection_power</th>\n",
       "      <th>feature_display</th>\n",
       "      <th>preclinical_efficacy_observed</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>normal_sample_barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.6073</td>\n",
       "      <td>219.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1921.0</td>\n",
       "      <td>0.9724</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SC_9065</td>\n",
       "      <td>SC_9065-Tumor</td>\n",
       "      <td>SC_9065-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.4224</td>\n",
       "      <td>232.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>290.0</td>\n",
       "      <td>0.4759</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-1115161</td>\n",
       "      <td>PROS01448-1115161-Tumor-SM-5SGU1</td>\n",
       "      <td>PROS01448-1115161-Normal-SM-6CAJL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>217.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1200/JCO.2017.75.7310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.2857</td>\n",
       "      <td>0.9799</td>\n",
       "      <td>PIK3CA p.E545K (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MO_1410</td>\n",
       "      <td>MO_1410-Tumor</td>\n",
       "      <td>MO_1410-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.5703</td>\n",
       "      <td>128.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS12319B-SU2C-06115114</td>\n",
       "      <td>PROS12319B-SU2C-06115114-Tumor-SM-4W2N9</td>\n",
       "      <td>PROS12319B-SU2C-06115114-Normal-SM-4W2NS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.0800</td>\n",
       "      <td>50.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-6115250</td>\n",
       "      <td>PROS01448-6115250-Tumor-SM-67ES4</td>\n",
       "      <td>PROS01448-6115250-Normal-SM-67ET3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.5806</td>\n",
       "      <td>31.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.9394</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-1115244</td>\n",
       "      <td>PROS01448-1115244-Tumor-SM-5SGU6</td>\n",
       "      <td>PROS01448-1115244-Normal-SM-6CAJP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.1190</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>0.2222</td>\n",
       "      <td>0.9822</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-6115219</td>\n",
       "      <td>PROS01448-6115219-Tumor-SM-67ERS</td>\n",
       "      <td>PROS01448-6115219-Normal-SM-67ESS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.2063</td>\n",
       "      <td>63.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1305.0</td>\n",
       "      <td>0.5586</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MO_1316</td>\n",
       "      <td>MO_1316-TM</td>\n",
       "      <td>MO_1316-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.8983</td>\n",
       "      <td>59.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-6115227</td>\n",
       "      <td>PROS01448-6115227-Tumor-SM-67ERU</td>\n",
       "      <td>PROS01448-6115227-Normal-SM-67ESU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>699</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.0758</td>\n",
       "      <td>356.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>640.0</td>\n",
       "      <td>0.1125</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SC_9046</td>\n",
       "      <td>SC_9046-Tumor</td>\n",
       "      <td>SC_9046-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E545A</td>\n",
       "      <td>0.4378</td>\n",
       "      <td>201.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1200/JCO.2017.75.7310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>PIK3CA p.E545A (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MO_1012</td>\n",
       "      <td>MO_1012-Tumor-Subcutaneous_nodule</td>\n",
       "      <td>MO_1012-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.5784</td>\n",
       "      <td>306.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MO_1040</td>\n",
       "      <td>MO_1040-Tumor</td>\n",
       "      <td>MO_1040-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1425</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.7194</td>\n",
       "      <td>449.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>0.9922</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP_2032</td>\n",
       "      <td>TP_2032-Tumor</td>\n",
       "      <td>TP_2032-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.5587</td>\n",
       "      <td>213.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>493.0</td>\n",
       "      <td>0.9899</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SC_9011</td>\n",
       "      <td>SC_9011-Tumor</td>\n",
       "      <td>SC_9011-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1533</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E17K</td>\n",
       "      <td>0.5348</td>\n",
       "      <td>445.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AKT1 p.E17K (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MO_1096</td>\n",
       "      <td>MO_1096-Tumor</td>\n",
       "      <td>MO_1096-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1663</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E17K</td>\n",
       "      <td>0.2219</td>\n",
       "      <td>329.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>445.0</td>\n",
       "      <td>0.2921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AKT1 p.E17K (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SC_9019</td>\n",
       "      <td>SC_9019-Tumor</td>\n",
       "      <td>SC_9019-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1733</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.1910</td>\n",
       "      <td>267.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.4444</td>\n",
       "      <td>0.9979</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat38</td>\n",
       "      <td>MEL-IPI_Pat38-Tumor-SM-53U3Z</td>\n",
       "      <td>MEL-IPI_Pat38-Normal-SM-53U5L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1925</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>218.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1468</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat39</td>\n",
       "      <td>MEL-IPI_Pat39-Tumor-SM-4DK1Z</td>\n",
       "      <td>MEL-IPI_Pat39-Normal-SM-4NFV6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2016</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.8286</td>\n",
       "      <td>245.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>166.0</td>\n",
       "      <td>0.8795</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat08</td>\n",
       "      <td>MEL-IPI_Pat08-Tumor-SM-4DK14</td>\n",
       "      <td>MEL-IPI_Pat08-Normal-SM-4NFUA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2050</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.3945</td>\n",
       "      <td>436.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat37</td>\n",
       "      <td>MEL-IPI_Pat37-Tumor-SM-53U3Y</td>\n",
       "      <td>MEL-IPI_Pat37-Normal-SM-4NFV4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2130</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6027</td>\n",
       "      <td>297.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>109.0</td>\n",
       "      <td>0.5596</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat90</td>\n",
       "      <td>MEL-IPI_Pat90-Tumor-SM-4DK3G</td>\n",
       "      <td>MEL-IPI_Pat90-Normal-SM-4NFWM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2289</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6505</td>\n",
       "      <td>372.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>92.0</td>\n",
       "      <td>0.6739</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat15</td>\n",
       "      <td>MEL-IPI_Pat15-Tumor-SM-4DK1B</td>\n",
       "      <td>MEL-IPI_Pat15-Normal-SM-4NFUH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2313</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61R</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>334.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>118.0</td>\n",
       "      <td>0.4661</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NRAS p.Q61R (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat119</td>\n",
       "      <td>MEL-IPI_Pat119-Tumor-SM-7459N</td>\n",
       "      <td>MEL-IPI_Pat119-Normal-SM-7459Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2423</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q56P</td>\n",
       "      <td>0.2955</td>\n",
       "      <td>264.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.3333</td>\n",
       "      <td>0.0703</td>\n",
       "      <td>MAP2K1 p.Q56P (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEL-IPI_Pat117</td>\n",
       "      <td>MEL-IPI_Pat117-Tumor-SM-5X2QU</td>\n",
       "      <td>MEL-IPI_Pat117-Normal-SM-5X2RC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2446</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.4085</td>\n",
       "      <td>213.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.4444</td>\n",
       "      <td>0.5392</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat128</td>\n",
       "      <td>MEL-IPI_Pat128-Tumor-SM-5X2R6</td>\n",
       "      <td>MEL-IPI_Pat128-Normal-SM-5VWHQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2546</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6174</td>\n",
       "      <td>311.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat49</td>\n",
       "      <td>MEL-IPI_Pat49-Tumor-SM-4DK2A</td>\n",
       "      <td>MEL-IPI_Pat49-Normal-SM-53U5W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2595</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6393</td>\n",
       "      <td>718.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.4000</td>\n",
       "      <td>0.4079</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat129</td>\n",
       "      <td>MEL-IPI_Pat129-Tumor-SM-5X2R7</td>\n",
       "      <td>MEL-IPI_Pat129-Normal-SM-5X2RI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2620</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6593</td>\n",
       "      <td>226.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat25</td>\n",
       "      <td>MEL-IPI_Pat25-Tumor-SM-4DK1L</td>\n",
       "      <td>MEL-IPI_Pat25-Normal-SM-4NFUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.7023</td>\n",
       "      <td>393.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.4000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat118</td>\n",
       "      <td>MEL-IPI_Pat118-Tumor-SM-5X2QV</td>\n",
       "      <td>MEL-IPI_Pat118-Normal-SM-5X2RD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2666</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2834</td>\n",
       "      <td>247.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>118.0</td>\n",
       "      <td>0.2542</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat14</td>\n",
       "      <td>MEL-IPI_Pat14-Tumor-SM-53U3C</td>\n",
       "      <td>MEL-IPI_Pat14-Normal-SM-4NFUG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2680</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>123.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat50</td>\n",
       "      <td>MEL-IPI_Pat50-Tumor-SM-53U69</td>\n",
       "      <td>MEL-IPI_Pat50-Normal-SM-53U5X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2842</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2387</td>\n",
       "      <td>331.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>137.0</td>\n",
       "      <td>0.2044</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat03</td>\n",
       "      <td>MEL-IPI_Pat03-Tumor-SM-4DJZY</td>\n",
       "      <td>MEL-IPI_Pat03-Normal-SM-4NFU5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3027</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6316</td>\n",
       "      <td>152.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1610</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat02</td>\n",
       "      <td>MEL-IPI_Pat02-Tumor-SM-4DJZX</td>\n",
       "      <td>MEL-IPI_Pat02-Normal-SM-4NFU4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3232</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2340</td>\n",
       "      <td>376.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>114.0</td>\n",
       "      <td>0.2456</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat27</td>\n",
       "      <td>MEL-IPI_Pat27-Tumor-SM-4DK1N</td>\n",
       "      <td>MEL-IPI_Pat27-Normal-SM-4NFUT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3288</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>410.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>122.0</td>\n",
       "      <td>0.5574</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat80</td>\n",
       "      <td>MEL-IPI_Pat80-Tumor-SM-4DK36</td>\n",
       "      <td>MEL-IPI_Pat80-Normal-SM-53U53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3367</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61L</td>\n",
       "      <td>0.6401</td>\n",
       "      <td>1592.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>493.0</td>\n",
       "      <td>0.7201</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NRAS p.Q61L (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat19</td>\n",
       "      <td>MEL-IPI_Pat19-Tumor-SM-4DK1F</td>\n",
       "      <td>MEL-IPI_Pat19-Normal-SM-4NFUL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3420</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61R</td>\n",
       "      <td>0.4370</td>\n",
       "      <td>270.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>152.0</td>\n",
       "      <td>0.4803</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NRAS p.Q61R (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat123</td>\n",
       "      <td>MEL-IPI_Pat123-Tumor-SM-5X2R1</td>\n",
       "      <td>MEL-IPI_Pat123-Normal-SM-5VWHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3432</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61R</td>\n",
       "      <td>0.4457</td>\n",
       "      <td>341.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NRAS p.Q61R (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat28</td>\n",
       "      <td>MEL-IPI_Pat28-Tumor-SM-4DK1O</td>\n",
       "      <td>MEL-IPI_Pat28-Normal-SM-4NFUU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3539</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61L</td>\n",
       "      <td>0.3958</td>\n",
       "      <td>331.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NRAS p.Q61L (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat44</td>\n",
       "      <td>MEL-IPI_Pat44-Tumor-SM-4DK25</td>\n",
       "      <td>MEL-IPI_Pat44-Normal-SM-4NFVB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3549</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>KIT</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V559G</td>\n",
       "      <td>0.8811</td>\n",
       "      <td>858.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>KIT p.V559G (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEL-IPI_Pat88</td>\n",
       "      <td>MEL-IPI_Pat88-Tumor-SM-4DK3E</td>\n",
       "      <td>MEL-IPI_Pat88-Normal-SM-53U4C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3610</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2803</td>\n",
       "      <td>635.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.2294</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat43</td>\n",
       "      <td>MEL-IPI_Pat43-Tumor-SM-4DK24</td>\n",
       "      <td>MEL-IPI_Pat43-Normal-SM-4NFVA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3636</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.4099</td>\n",
       "      <td>405.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>106.0</td>\n",
       "      <td>0.6698</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat86</td>\n",
       "      <td>MEL-IPI_Pat86-Tumor-SM-4DK3C</td>\n",
       "      <td>MEL-IPI_Pat86-Normal-SM-53U59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4217</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.G12D</td>\n",
       "      <td>0.4828</td>\n",
       "      <td>29.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1038/ctg.2016.18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.4706</td>\n",
       "      <td>0.9721</td>\n",
       "      <td>KRAS p.G12D (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>KIRP-A4-8312</td>\n",
       "      <td>KIRP-A4-8312-TP</td>\n",
       "      <td>KIRP-A4-8312-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4298</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.3839</td>\n",
       "      <td>112.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.2353</td>\n",
       "      <td>0.9280</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>KIRP-AL-3467</td>\n",
       "      <td>KIRP-AL-3467-TP</td>\n",
       "      <td>KIRP-AL-3467-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4343</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E542K</td>\n",
       "      <td>0.4722</td>\n",
       "      <td>449.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1200/JCO.2017.75.7310</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.0505</td>\n",
       "      <td>PIK3CA p.E542K (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>KIRP-5P-A9K3</td>\n",
       "      <td>KIRP-5P-A9K3-TP</td>\n",
       "      <td>KIRP-5P-A9K3-NB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  score_bin sensitive_predictive_implication  \\\n",
       "70    Putatively Actionable                              NaN   \n",
       "119   Putatively Actionable                              NaN   \n",
       "137   Putatively Actionable                      Preclinical   \n",
       "369   Putatively Actionable                              NaN   \n",
       "424   Putatively Actionable                              NaN   \n",
       "442   Putatively Actionable                              NaN   \n",
       "510   Putatively Actionable                              NaN   \n",
       "533   Putatively Actionable                              NaN   \n",
       "564   Putatively Actionable                              NaN   \n",
       "699   Putatively Actionable                              NaN   \n",
       "879   Putatively Actionable                      Preclinical   \n",
       "1026  Putatively Actionable                              NaN   \n",
       "1425  Putatively Actionable                              NaN   \n",
       "1505  Putatively Actionable                              NaN   \n",
       "1533  Putatively Actionable                      Preclinical   \n",
       "1663  Putatively Actionable                      Preclinical   \n",
       "1733  Putatively Actionable                     FDA-Approved   \n",
       "1925  Putatively Actionable                     FDA-Approved   \n",
       "2016  Putatively Actionable                     FDA-Approved   \n",
       "2050  Putatively Actionable                     FDA-Approved   \n",
       "2130  Putatively Actionable                     FDA-Approved   \n",
       "2289  Putatively Actionable                     FDA-Approved   \n",
       "2313  Putatively Actionable                Clinical evidence   \n",
       "2423  Putatively Actionable                      Preclinical   \n",
       "2446  Putatively Actionable                     FDA-Approved   \n",
       "2546  Putatively Actionable                     FDA-Approved   \n",
       "2595  Putatively Actionable                     FDA-Approved   \n",
       "2620  Putatively Actionable                     FDA-Approved   \n",
       "2653  Putatively Actionable                     FDA-Approved   \n",
       "2666  Putatively Actionable                     FDA-Approved   \n",
       "2680  Putatively Actionable                     FDA-Approved   \n",
       "2842  Putatively Actionable                     FDA-Approved   \n",
       "3027  Putatively Actionable                     FDA-Approved   \n",
       "3232  Putatively Actionable                     FDA-Approved   \n",
       "3288  Putatively Actionable                     FDA-Approved   \n",
       "3367  Putatively Actionable                Clinical evidence   \n",
       "3420  Putatively Actionable                Clinical evidence   \n",
       "3432  Putatively Actionable                Clinical evidence   \n",
       "3539  Putatively Actionable                Clinical evidence   \n",
       "3549  Putatively Actionable                        Guideline   \n",
       "3610  Putatively Actionable                     FDA-Approved   \n",
       "3636  Putatively Actionable                     FDA-Approved   \n",
       "4217  Putatively Actionable                   Clinical trial   \n",
       "4298  Putatively Actionable                     FDA-Approved   \n",
       "4343  Putatively Actionable                      Preclinical   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "70                   Clinical evidence                               NaN   \n",
       "119                  Clinical evidence                               NaN   \n",
       "137                                NaN                    Clinical trial   \n",
       "369                  Clinical evidence                               NaN   \n",
       "424                  Clinical evidence                               NaN   \n",
       "442                  Clinical evidence                               NaN   \n",
       "510                  Clinical evidence                               NaN   \n",
       "533                  Clinical evidence                               NaN   \n",
       "564                  Clinical evidence                               NaN   \n",
       "699                  Clinical evidence                               NaN   \n",
       "879                                NaN                    Clinical trial   \n",
       "1026                 Clinical evidence                               NaN   \n",
       "1425                 Clinical evidence                               NaN   \n",
       "1505                 Clinical evidence                               NaN   \n",
       "1533                               NaN                               NaN   \n",
       "1663                               NaN                               NaN   \n",
       "1733                         Guideline                         Guideline   \n",
       "1925                         Guideline                         Guideline   \n",
       "2016                         Guideline                         Guideline   \n",
       "2050                         Guideline                         Guideline   \n",
       "2130                         Guideline                         Guideline   \n",
       "2289                         Guideline                         Guideline   \n",
       "2313                               NaN                         Guideline   \n",
       "2423                       Preclinical                               NaN   \n",
       "2446                         Guideline                         Guideline   \n",
       "2546                         Guideline                         Guideline   \n",
       "2595                         Guideline                         Guideline   \n",
       "2620                         Guideline                         Guideline   \n",
       "2653                         Guideline                         Guideline   \n",
       "2666                         Guideline                         Guideline   \n",
       "2680                         Guideline                         Guideline   \n",
       "2842                         Guideline                         Guideline   \n",
       "3027                         Guideline                         Guideline   \n",
       "3232                         Guideline                         Guideline   \n",
       "3288                         Guideline                         Guideline   \n",
       "3367                               NaN                         Guideline   \n",
       "3420                               NaN                         Guideline   \n",
       "3432                               NaN                         Guideline   \n",
       "3539                               NaN                         Guideline   \n",
       "3549                               NaN                         Guideline   \n",
       "3610                         Guideline                         Guideline   \n",
       "3636                         Guideline                         Guideline   \n",
       "4217                    Clinical trial                 Clinical evidence   \n",
       "4298                         Guideline                         Guideline   \n",
       "4343                               NaN                    Clinical trial   \n",
       "\n",
       "         feature_type feature alteration_type alteration  tumor_f  \\\n",
       "70    Somatic Variant      AR        Missense    p.T878A   0.6073   \n",
       "119   Somatic Variant      AR        Missense    p.T878A   0.4224   \n",
       "137   Somatic Variant  PIK3CA        Missense    p.E545K   0.3825   \n",
       "369   Somatic Variant      AR        Missense    p.L702H   0.5703   \n",
       "424   Somatic Variant      AR        Missense    p.L702H   0.0800   \n",
       "442   Somatic Variant      AR        Missense    p.L702H   0.5806   \n",
       "510   Somatic Variant      AR        Missense    p.L702H   0.1190   \n",
       "533   Somatic Variant      AR        Missense    p.L702H   0.2063   \n",
       "564   Somatic Variant      AR        Missense    p.L702H   0.8983   \n",
       "699   Somatic Variant      AR        Missense    p.L702H   0.0758   \n",
       "879   Somatic Variant  PIK3CA        Missense    p.E545A   0.4378   \n",
       "1026  Somatic Variant      AR        Missense    p.T878A   0.5784   \n",
       "1425  Somatic Variant      AR        Missense    p.T878A   0.7194   \n",
       "1505  Somatic Variant      AR        Missense    p.T878A   0.5587   \n",
       "1533  Somatic Variant    AKT1        Missense     p.E17K   0.5348   \n",
       "1663  Somatic Variant    AKT1        Missense     p.E17K   0.2219   \n",
       "1733  Somatic Variant    BRAF        Missense    p.V600E   0.1910   \n",
       "1925  Somatic Variant    BRAF        Missense    p.V600E   0.3716   \n",
       "2016  Somatic Variant    BRAF        Missense    p.V600E   0.8286   \n",
       "2050  Somatic Variant    BRAF        Missense    p.V600E   0.3945   \n",
       "2130  Somatic Variant    BRAF        Missense    p.V600E   0.6027   \n",
       "2289  Somatic Variant    BRAF        Missense    p.V600E   0.6505   \n",
       "2313  Somatic Variant    NRAS        Missense     p.Q61R   0.4222   \n",
       "2423  Somatic Variant  MAP2K1        Missense     p.Q56P   0.2955   \n",
       "2446  Somatic Variant    BRAF        Missense    p.V600E   0.4085   \n",
       "2546  Somatic Variant    BRAF        Missense    p.V600E   0.6174   \n",
       "2595  Somatic Variant    BRAF        Missense    p.V600E   0.6393   \n",
       "2620  Somatic Variant    BRAF        Missense    p.V600E   0.6593   \n",
       "2653  Somatic Variant    BRAF        Missense    p.V600E   0.7023   \n",
       "2666  Somatic Variant    BRAF        Missense    p.V600E   0.2834   \n",
       "2680  Somatic Variant    BRAF        Missense    p.V600E   0.8049   \n",
       "2842  Somatic Variant    BRAF        Missense    p.V600E   0.2387   \n",
       "3027  Somatic Variant    BRAF        Missense    p.V600E   0.6316   \n",
       "3232  Somatic Variant    BRAF        Missense    p.V600E   0.2340   \n",
       "3288  Somatic Variant    BRAF        Missense    p.V600E   0.4659   \n",
       "3367  Somatic Variant    NRAS        Missense     p.Q61L   0.6401   \n",
       "3420  Somatic Variant    NRAS        Missense     p.Q61R   0.4370   \n",
       "3432  Somatic Variant    NRAS        Missense     p.Q61R   0.4457   \n",
       "3539  Somatic Variant    NRAS        Missense     p.Q61L   0.3958   \n",
       "3549  Somatic Variant     KIT        Missense    p.V559G   0.8811   \n",
       "3610  Somatic Variant    BRAF        Missense    p.V600E   0.2803   \n",
       "3636  Somatic Variant    BRAF        Missense    p.V600E   0.4099   \n",
       "4217  Somatic Variant    KRAS        Missense     p.G12D   0.4828   \n",
       "4298  Somatic Variant    BRAF        Missense    p.V600E   0.3839   \n",
       "4343  Somatic Variant  PIK3CA        Missense    p.E542K   0.4722   \n",
       "\n",
       "      total_coverage  ...                                     prognostic_url  \\\n",
       "70             219.0  ...                                                NaN   \n",
       "119            232.0  ...                                                NaN   \n",
       "137            217.0  ...           https://doi.org/10.1200/JCO.2017.75.7310   \n",
       "369            128.0  ...                                                NaN   \n",
       "424             50.0  ...                                                NaN   \n",
       "442             31.0  ...                                                NaN   \n",
       "510            210.0  ...                                                NaN   \n",
       "533             63.0  ...                                                NaN   \n",
       "564             59.0  ...                                                NaN   \n",
       "699            356.0  ...                                                NaN   \n",
       "879            201.0  ...           https://doi.org/10.1200/JCO.2017.75.7310   \n",
       "1026           306.0  ...                                                NaN   \n",
       "1425           449.0  ...                                                NaN   \n",
       "1505           213.0  ...                                                NaN   \n",
       "1533           445.0  ...                                                NaN   \n",
       "1663           329.0  ...                                                NaN   \n",
       "1733           267.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "1925           218.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2016           245.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2050           436.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2130           297.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2289           372.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2313           334.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2423           264.0  ...                                                NaN   \n",
       "2446           213.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2546           311.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2595           718.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2620           226.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2653           393.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2666           247.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2680           123.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2842           331.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3027           152.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3232           376.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3288           410.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3367          1592.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3420           270.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3432           341.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3539           331.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3549           858.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3610           635.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3636           405.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "4217            29.0  ...                https://doi.org/10.1038/ctg.2016.18   \n",
       "4298           112.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "4343           449.0  ...           https://doi.org/10.1200/JCO.2017.75.7310   \n",
       "\n",
       "      number_germline_mutations_in_gene validation_total_coverage  \\\n",
       "70                                  1.0                    1921.0   \n",
       "119                                 NaN                     290.0   \n",
       "137                                 NaN                      21.0   \n",
       "369                                 NaN                      21.0   \n",
       "424                                 NaN                       4.0   \n",
       "442                                 NaN                      33.0   \n",
       "510                                 1.0                      81.0   \n",
       "533                                 2.0                    1305.0   \n",
       "564                                 NaN                      66.0   \n",
       "699                                 1.0                     640.0   \n",
       "879                                 NaN                       1.0   \n",
       "1026                                1.0                      64.0   \n",
       "1425                                1.0                     383.0   \n",
       "1505                                2.0                     493.0   \n",
       "1533                                NaN                     102.0   \n",
       "1663                                NaN                     445.0   \n",
       "1733                                NaN                      63.0   \n",
       "1925                                NaN                       6.0   \n",
       "2016                                NaN                     166.0   \n",
       "2050                                NaN                       0.0   \n",
       "2130                                NaN                     109.0   \n",
       "2289                                NaN                      92.0   \n",
       "2313                                NaN                     118.0   \n",
       "2423                                NaN                       6.0   \n",
       "2446                                NaN                       9.0   \n",
       "2546                                NaN                       0.0   \n",
       "2595                                NaN                       5.0   \n",
       "2620                                NaN                       0.0   \n",
       "2653                                NaN                      20.0   \n",
       "2666                                NaN                     118.0   \n",
       "2680                                NaN                       0.0   \n",
       "2842                                NaN                     137.0   \n",
       "3027                                NaN                       4.0   \n",
       "3232                                NaN                     114.0   \n",
       "3288                                NaN                     122.0   \n",
       "3367                                NaN                     493.0   \n",
       "3420                                NaN                     152.0   \n",
       "3432                                NaN                       0.0   \n",
       "3539                                NaN                       0.0   \n",
       "3549                                NaN                       0.0   \n",
       "3610                                NaN                       9.0   \n",
       "3636                                NaN                     106.0   \n",
       "4217                                NaN                      17.0   \n",
       "4298                                NaN                      17.0   \n",
       "4343                                1.0                       4.0   \n",
       "\n",
       "     validation_tumor_f validation_detection_power            feature_display  \\\n",
       "70               0.9724                     1.0000      AR p.T878A (Missense)   \n",
       "119              0.4759                     1.0000      AR p.T878A (Missense)   \n",
       "137              0.2857                     0.9799  PIK3CA p.E545K (Missense)   \n",
       "369              0.7619                     0.9998      AR p.L702H (Missense)   \n",
       "424              0.0000                     0.0002      AR p.L702H (Missense)   \n",
       "442              0.9394                     1.0000      AR p.L702H (Missense)   \n",
       "510              0.2222                     0.9822      AR p.L702H (Missense)   \n",
       "533              0.5586                     1.0000      AR p.L702H (Missense)   \n",
       "564              1.0000                     1.0000      AR p.L702H (Missense)   \n",
       "699              0.1125                     1.0000      AR p.L702H (Missense)   \n",
       "879              0.0000                     0.0000  PIK3CA p.E545A (Missense)   \n",
       "1026             0.2812                     1.0000      AR p.T878A (Missense)   \n",
       "1425             0.9922                     1.0000      AR p.T878A (Missense)   \n",
       "1505             0.9899                     1.0000      AR p.T878A (Missense)   \n",
       "1533             0.0000                     1.0000     AKT1 p.E17K (Missense)   \n",
       "1663             0.2921                     1.0000     AKT1 p.E17K (Missense)   \n",
       "1733             0.4444                     0.9979    BRAF p.V600E (Missense)   \n",
       "1925             0.0000                     0.1468    BRAF p.V600E (Missense)   \n",
       "2016             0.8795                     1.0000    BRAF p.V600E (Missense)   \n",
       "2050             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2130             0.5596                     1.0000    BRAF p.V600E (Missense)   \n",
       "2289             0.6739                     1.0000    BRAF p.V600E (Missense)   \n",
       "2313             0.4661                     1.0000     NRAS p.Q61R (Missense)   \n",
       "2423             0.3333                     0.0703   MAP2K1 p.Q56P (Missense)   \n",
       "2446             0.4444                     0.5392    BRAF p.V600E (Missense)   \n",
       "2546             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2595             0.4000                     0.4079    BRAF p.V600E (Missense)   \n",
       "2620             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2653             0.4000                     1.0000    BRAF p.V600E (Missense)   \n",
       "2666             0.2542                     1.0000    BRAF p.V600E (Missense)   \n",
       "2680             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2842             0.2044                     1.0000    BRAF p.V600E (Missense)   \n",
       "3027             0.0000                     0.1610    BRAF p.V600E (Missense)   \n",
       "3232             0.2456                     1.0000    BRAF p.V600E (Missense)   \n",
       "3288             0.5574                     1.0000    BRAF p.V600E (Missense)   \n",
       "3367             0.7201                     1.0000     NRAS p.Q61L (Missense)   \n",
       "3420             0.4803                     1.0000     NRAS p.Q61R (Missense)   \n",
       "3432             0.0000                     0.0000     NRAS p.Q61R (Missense)   \n",
       "3539             0.0000                     0.0000     NRAS p.Q61L (Missense)   \n",
       "3549             0.0000                     0.0000     KIT p.V559G (Missense)   \n",
       "3610             0.0000                     0.2294    BRAF p.V600E (Missense)   \n",
       "3636             0.6698                     1.0000    BRAF p.V600E (Missense)   \n",
       "4217             0.4706                     0.9721     KRAS p.G12D (Missense)   \n",
       "4298             0.2353                     0.9280    BRAF p.V600E (Missense)   \n",
       "4343             0.5000                     0.0505  PIK3CA p.E542K (Missense)   \n",
       "\n",
       "     preclinical_efficacy_observed                patient_id  \\\n",
       "70                             NaN                   SC_9065   \n",
       "119                            NaN         PROS01448-1115161   \n",
       "137                            1.0                   MO_1410   \n",
       "369                            NaN  PROS12319B-SU2C-06115114   \n",
       "424                            NaN         PROS01448-6115250   \n",
       "442                            NaN         PROS01448-1115244   \n",
       "510                            NaN         PROS01448-6115219   \n",
       "533                            NaN                   MO_1316   \n",
       "564                            NaN         PROS01448-6115227   \n",
       "699                            NaN                   SC_9046   \n",
       "879                            1.0                   MO_1012   \n",
       "1026                           NaN                   MO_1040   \n",
       "1425                           NaN                   TP_2032   \n",
       "1505                           NaN                   SC_9011   \n",
       "1533                           0.0                   MO_1096   \n",
       "1663                           0.0                   SC_9019   \n",
       "1733                           1.0             MEL-IPI_Pat38   \n",
       "1925                           1.0             MEL-IPI_Pat39   \n",
       "2016                           1.0             MEL-IPI_Pat08   \n",
       "2050                           1.0             MEL-IPI_Pat37   \n",
       "2130                           1.0             MEL-IPI_Pat90   \n",
       "2289                           1.0             MEL-IPI_Pat15   \n",
       "2313                           1.0            MEL-IPI_Pat119   \n",
       "2423                           0.0            MEL-IPI_Pat117   \n",
       "2446                           1.0            MEL-IPI_Pat128   \n",
       "2546                           1.0             MEL-IPI_Pat49   \n",
       "2595                           1.0            MEL-IPI_Pat129   \n",
       "2620                           1.0             MEL-IPI_Pat25   \n",
       "2653                           1.0            MEL-IPI_Pat118   \n",
       "2666                           1.0             MEL-IPI_Pat14   \n",
       "2680                           1.0             MEL-IPI_Pat50   \n",
       "2842                           1.0             MEL-IPI_Pat03   \n",
       "3027                           1.0             MEL-IPI_Pat02   \n",
       "3232                           1.0             MEL-IPI_Pat27   \n",
       "3288                           1.0             MEL-IPI_Pat80   \n",
       "3367                           1.0             MEL-IPI_Pat19   \n",
       "3420                           1.0            MEL-IPI_Pat123   \n",
       "3432                           1.0             MEL-IPI_Pat28   \n",
       "3539                           1.0             MEL-IPI_Pat44   \n",
       "3549                           0.0             MEL-IPI_Pat88   \n",
       "3610                           1.0             MEL-IPI_Pat43   \n",
       "3636                           1.0             MEL-IPI_Pat86   \n",
       "4217                           1.0              KIRP-A4-8312   \n",
       "4298                           1.0              KIRP-AL-3467   \n",
       "4343                           1.0              KIRP-5P-A9K3   \n",
       "\n",
       "                         tumor_sample_barcode  \\\n",
       "70                              SC_9065-Tumor   \n",
       "119          PROS01448-1115161-Tumor-SM-5SGU1   \n",
       "137                             MO_1410-Tumor   \n",
       "369   PROS12319B-SU2C-06115114-Tumor-SM-4W2N9   \n",
       "424          PROS01448-6115250-Tumor-SM-67ES4   \n",
       "442          PROS01448-1115244-Tumor-SM-5SGU6   \n",
       "510          PROS01448-6115219-Tumor-SM-67ERS   \n",
       "533                                MO_1316-TM   \n",
       "564          PROS01448-6115227-Tumor-SM-67ERU   \n",
       "699                             SC_9046-Tumor   \n",
       "879         MO_1012-Tumor-Subcutaneous_nodule   \n",
       "1026                            MO_1040-Tumor   \n",
       "1425                            TP_2032-Tumor   \n",
       "1505                            SC_9011-Tumor   \n",
       "1533                            MO_1096-Tumor   \n",
       "1663                            SC_9019-Tumor   \n",
       "1733             MEL-IPI_Pat38-Tumor-SM-53U3Z   \n",
       "1925             MEL-IPI_Pat39-Tumor-SM-4DK1Z   \n",
       "2016             MEL-IPI_Pat08-Tumor-SM-4DK14   \n",
       "2050             MEL-IPI_Pat37-Tumor-SM-53U3Y   \n",
       "2130             MEL-IPI_Pat90-Tumor-SM-4DK3G   \n",
       "2289             MEL-IPI_Pat15-Tumor-SM-4DK1B   \n",
       "2313            MEL-IPI_Pat119-Tumor-SM-7459N   \n",
       "2423            MEL-IPI_Pat117-Tumor-SM-5X2QU   \n",
       "2446            MEL-IPI_Pat128-Tumor-SM-5X2R6   \n",
       "2546             MEL-IPI_Pat49-Tumor-SM-4DK2A   \n",
       "2595            MEL-IPI_Pat129-Tumor-SM-5X2R7   \n",
       "2620             MEL-IPI_Pat25-Tumor-SM-4DK1L   \n",
       "2653            MEL-IPI_Pat118-Tumor-SM-5X2QV   \n",
       "2666             MEL-IPI_Pat14-Tumor-SM-53U3C   \n",
       "2680             MEL-IPI_Pat50-Tumor-SM-53U69   \n",
       "2842             MEL-IPI_Pat03-Tumor-SM-4DJZY   \n",
       "3027             MEL-IPI_Pat02-Tumor-SM-4DJZX   \n",
       "3232             MEL-IPI_Pat27-Tumor-SM-4DK1N   \n",
       "3288             MEL-IPI_Pat80-Tumor-SM-4DK36   \n",
       "3367             MEL-IPI_Pat19-Tumor-SM-4DK1F   \n",
       "3420            MEL-IPI_Pat123-Tumor-SM-5X2R1   \n",
       "3432             MEL-IPI_Pat28-Tumor-SM-4DK1O   \n",
       "3539             MEL-IPI_Pat44-Tumor-SM-4DK25   \n",
       "3549             MEL-IPI_Pat88-Tumor-SM-4DK3E   \n",
       "3610             MEL-IPI_Pat43-Tumor-SM-4DK24   \n",
       "3636             MEL-IPI_Pat86-Tumor-SM-4DK3C   \n",
       "4217                          KIRP-A4-8312-TP   \n",
       "4298                          KIRP-AL-3467-TP   \n",
       "4343                          KIRP-5P-A9K3-TP   \n",
       "\n",
       "                         normal_sample_barcode  \n",
       "70                                  SC_9065-NB  \n",
       "119          PROS01448-1115161-Normal-SM-6CAJL  \n",
       "137                             MO_1410-Normal  \n",
       "369   PROS12319B-SU2C-06115114-Normal-SM-4W2NS  \n",
       "424          PROS01448-6115250-Normal-SM-67ET3  \n",
       "442          PROS01448-1115244-Normal-SM-6CAJP  \n",
       "510          PROS01448-6115219-Normal-SM-67ESS  \n",
       "533                                 MO_1316-NB  \n",
       "564          PROS01448-6115227-Normal-SM-67ESU  \n",
       "699                             SC_9046-Normal  \n",
       "879                             MO_1012-Normal  \n",
       "1026                            MO_1040-Normal  \n",
       "1425                            TP_2032-Normal  \n",
       "1505                            SC_9011-Normal  \n",
       "1533                            MO_1096-Normal  \n",
       "1663                            SC_9019-Normal  \n",
       "1733             MEL-IPI_Pat38-Normal-SM-53U5L  \n",
       "1925             MEL-IPI_Pat39-Normal-SM-4NFV6  \n",
       "2016             MEL-IPI_Pat08-Normal-SM-4NFUA  \n",
       "2050             MEL-IPI_Pat37-Normal-SM-4NFV4  \n",
       "2130             MEL-IPI_Pat90-Normal-SM-4NFWM  \n",
       "2289             MEL-IPI_Pat15-Normal-SM-4NFUH  \n",
       "2313            MEL-IPI_Pat119-Normal-SM-7459Q  \n",
       "2423            MEL-IPI_Pat117-Normal-SM-5X2RC  \n",
       "2446            MEL-IPI_Pat128-Normal-SM-5VWHQ  \n",
       "2546             MEL-IPI_Pat49-Normal-SM-53U5W  \n",
       "2595            MEL-IPI_Pat129-Normal-SM-5X2RI  \n",
       "2620             MEL-IPI_Pat25-Normal-SM-4NFUR  \n",
       "2653            MEL-IPI_Pat118-Normal-SM-5X2RD  \n",
       "2666             MEL-IPI_Pat14-Normal-SM-4NFUG  \n",
       "2680             MEL-IPI_Pat50-Normal-SM-53U5X  \n",
       "2842             MEL-IPI_Pat03-Normal-SM-4NFU5  \n",
       "3027             MEL-IPI_Pat02-Normal-SM-4NFU4  \n",
       "3232             MEL-IPI_Pat27-Normal-SM-4NFUT  \n",
       "3288             MEL-IPI_Pat80-Normal-SM-53U53  \n",
       "3367             MEL-IPI_Pat19-Normal-SM-4NFUL  \n",
       "3420            MEL-IPI_Pat123-Normal-SM-5VWHL  \n",
       "3432             MEL-IPI_Pat28-Normal-SM-4NFUU  \n",
       "3539             MEL-IPI_Pat44-Normal-SM-4NFVB  \n",
       "3549             MEL-IPI_Pat88-Normal-SM-53U4C  \n",
       "3610             MEL-IPI_Pat43-Normal-SM-4NFVA  \n",
       "3636             MEL-IPI_Pat86-Normal-SM-53U59  \n",
       "4217                           KIRP-A4-8312-NB  \n",
       "4298                           KIRP-AL-3467-NB  \n",
       "4343                           KIRP-5P-A9K3-NB  \n",
       "\n",
       "[45 rows x 41 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handles = (glob.glob('2015-Robinson/data/*/*/*.actionable.txt') + \n",
    "           glob.glob('2015-VanAllen/data/*/*/*.actionable.txt') + \n",
    "           glob.glob('2016-TCGA/data/*/*/*.actionable.txt') + \n",
    "           glob.glob('2014-Perry/data/*/*/*.actionable.txt')\n",
    "          )\n",
    "tmp = pd.concat([pd.read_csv(handle, sep='\\t') for handle in handles], ignore_index=True)\n",
    "tmp = tmp[tmp['feature_type'].eq('Somatic Variant') & tmp['score_bin'].isin(['Putatively Actionable', 'Investigate Actionability - High', 'Investigate Actionability - Low']) & ~tmp['validation_total_coverage'].isnull()]\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(29, 41)\n"
     ]
    }
   ],
   "source": [
    "print(tmp[tmp['validation_detection_power'].ge(0.80)].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BRAF p.V600E (Missense)      11\n",
       "AR p.L702H (Missense)         6\n",
       "AR p.T878A (Missense)         5\n",
       "NRAS p.Q61R (Missense)        2\n",
       "AKT1 p.E17K (Missense)        2\n",
       "KRAS p.G12D (Missense)        1\n",
       "NRAS p.Q61L (Missense)        1\n",
       "PIK3CA p.E545K (Missense)     1\n",
       "Name: feature_display, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['validation_detection_power'].ge(0.80)]['feature_display'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extended features\n",
    "Features considered: Germline variants, fusions, mutational burden, mutational signatures, aneuploidy  \n",
    "Evidence considered: FDA-approved, clinical trials, or clinical evidence  \n",
    "Matches considered: First putatively actionable and then also investigate actionability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(483, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>KIRP-2Z-A9J5</td>\n",
       "      <td>COSMIC Signature 3</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>KIRP-2Z-A9J7</td>\n",
       "      <td>COSMIC Signature 2</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>KIRP-2Z-A9J9</td>\n",
       "      <td>COSMIC Signature 3</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>KIRP-2Z-A9JG</td>\n",
       "      <td>COSMIC Signature 5</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      patient_id      feature_string                almanac_bin phial_bin  \\\n",
       "3   KIRP-2K-A9WE  COSMIC Signature 4  Investigate Actionability       NaN   \n",
       "19  KIRP-2Z-A9J5  COSMIC Signature 3  Investigate Actionability       NaN   \n",
       "28  KIRP-2Z-A9J7  COSMIC Signature 2  Investigate Actionability       NaN   \n",
       "33  KIRP-2Z-A9J9  COSMIC Signature 3  Investigate Actionability       NaN   \n",
       "52  KIRP-2Z-A9JG  COSMIC Signature 5  Investigate Actionability       NaN   \n",
       "\n",
       "    sum cohort sensitive_predictive_implication  \\\n",
       "3     1   KIRP                      Inferential   \n",
       "19    1   KIRP                      Inferential   \n",
       "28    1   KIRP                      Inferential   \n",
       "33    1   KIRP                      Inferential   \n",
       "52    1   KIRP                      Inferential   \n",
       "\n",
       "   resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "3                                NaN                               NaN   \n",
       "19                               NaN                               NaN   \n",
       "28                               NaN                               NaN   \n",
       "33                               NaN                               NaN   \n",
       "52                               NaN                               NaN   \n",
       "\n",
       "    clinvar          feature_type  feature_str_simple  \n",
       "3       NaN  Mutational signature  COSMIC Signature 4  \n",
       "19      NaN  Mutational signature  COSMIC Signature 3  \n",
       "28      NaN  Mutational signature  COSMIC Signature 2  \n",
       "33      NaN  Mutational signature  COSMIC Signature 3  \n",
       "52      NaN  Mutational signature  COSMIC Signature 5  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx_f = df['feature_type'].isin(['Germline', 'Rearrangement', 'Aneuploidy', \n",
    "                                 'Tumor mutational burden', 'Mutational signature'])\n",
    "idx_a = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "\n",
    "tmp = df[idx_f & idx_a]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'By considering these feature types, MOAlmanac identified an additional 483 clinically relevant molecular features in 315 patients, resulting in 393 patients with at least one event associated with therapeutic sensitivity, resistance, or prognosis.'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_cr_total = n_almanac_mel + n_almanac_su2c + n_almanac_kirp + n_almanac_os\n",
    "\n",
    "f\"By considering these feature types, MOAlmanac identified an additional {tmp.shape[0]} clinically relevant \\\n",
    "molecular features in {tmp['patient_id'].drop_duplicates().shape[0]} patients, resulting in \\\n",
    "{n_cr_total} patients with at least one event associated with therapeutic sensitivity, resistance, or prognosis.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(483, 12)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(315,)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 12)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_germline = tmp[tmp['feature_type'].eq('Germline') & tmp['clinvar'].isin(['Pathogenic', 'Pathogenic/Likely pathogenic', 'Likely pathogenic'])]\n",
    "tmp_germline.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0,)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_germline['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [patient_id, feature_string, almanac_bin, phial_bin, sum, cohort, sensitive_predictive_implication, resistance_predictive_implication, prognostic_predictive_implication, clinvar, feature_type, feature_str_simple]\n",
       "Index: []"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_germline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(137, 12)\n",
      "(91,)\n"
     ]
    }
   ],
   "source": [
    "tmp_rearrangement = tmp[tmp['feature_type'].eq('Rearrangement')]\n",
    "print(tmp_rearrangement.shape)\n",
    "print(tmp_rearrangement['patient_id'].drop_duplicates().shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erg_patients = tmp[tmp['feature_string'].eq('TMPRSS2--ERG')]['patient_id'].drop_duplicates()\n",
    "len(pd.Index(erg_patients).difference(pd.Index(pa_patients)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2371</th>\n",
       "      <td>MO_1008</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2483</th>\n",
       "      <td>MO_1015</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2492</th>\n",
       "      <td>MO_1040</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>MO_1071</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2531</th>\n",
       "      <td>MO_1095</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2550</th>\n",
       "      <td>MO_1114</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2556</th>\n",
       "      <td>MO_1118</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2564</th>\n",
       "      <td>MO_1124</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2589</th>\n",
       "      <td>MO_1162</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2595</th>\n",
       "      <td>MO_1176</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2614</th>\n",
       "      <td>MO_1192</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2627</th>\n",
       "      <td>MO_1202</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2661</th>\n",
       "      <td>MO_1241</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2670</th>\n",
       "      <td>MO_1244</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2676</th>\n",
       "      <td>MO_1249</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2690</th>\n",
       "      <td>MO_1263</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2698</th>\n",
       "      <td>MO_1277</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2707</th>\n",
       "      <td>MO_1316</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2722</th>\n",
       "      <td>MO_1337</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3107</th>\n",
       "      <td>PROS01448-1115183</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3205</th>\n",
       "      <td>PROS10448-1115154</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3290</th>\n",
       "      <td>SC_9003</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3353</th>\n",
       "      <td>SC_9009</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3377</th>\n",
       "      <td>SC_9016</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3390</th>\n",
       "      <td>SC_9018</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3403</th>\n",
       "      <td>SC_9020</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3424</th>\n",
       "      <td>SC_9022</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3503</th>\n",
       "      <td>SC_9035</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3514</th>\n",
       "      <td>SC_9037</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3548</th>\n",
       "      <td>SC_9043</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3555</th>\n",
       "      <td>SC_9046</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3591</th>\n",
       "      <td>SC_9049</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3624</th>\n",
       "      <td>SC_9056</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3653</th>\n",
       "      <td>SC_9059</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3660</th>\n",
       "      <td>SC_9060</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3672</th>\n",
       "      <td>SC_9061</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3693</th>\n",
       "      <td>SC_9063</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3709</th>\n",
       "      <td>SC_9066</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3809</th>\n",
       "      <td>SC_9086</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3843</th>\n",
       "      <td>SC_9092</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3902</th>\n",
       "      <td>SC_9097</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3935</th>\n",
       "      <td>SC_9104</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3943</th>\n",
       "      <td>SC_9107</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3955</th>\n",
       "      <td>SC_9109</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3965</th>\n",
       "      <td>TP_2001</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3996</th>\n",
       "      <td>TP_2054</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient_id feature_string            almanac_bin phial_bin  sum  \\\n",
       "2371            MO_1008   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2483            MO_1015   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2492            MO_1040   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2503            MO_1071   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2531            MO_1095   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2550            MO_1114   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2556            MO_1118   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2564            MO_1124   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2589            MO_1162   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2595            MO_1176   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2614            MO_1192   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2627            MO_1202   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2661            MO_1241   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2670            MO_1244   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2676            MO_1249   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2690            MO_1263   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2698            MO_1277   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2707            MO_1316   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "2722            MO_1337   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3107  PROS01448-1115183   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3205  PROS10448-1115154   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3290            SC_9003   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3353            SC_9009   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3377            SC_9016   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3390            SC_9018   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3403            SC_9020   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3424            SC_9022   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3503            SC_9035   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3514            SC_9037   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3548            SC_9043   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3555            SC_9046   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3591            SC_9049   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3624            SC_9056   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3653            SC_9059   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3660            SC_9060   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3672            SC_9061   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3693            SC_9063   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3709            SC_9066   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3809            SC_9086   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3843            SC_9092   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3902            SC_9097   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3935            SC_9104   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3943            SC_9107   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3955            SC_9109   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3965            TP_2001   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3996            TP_2054   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "\n",
       "     cohort sensitive_predictive_implication  \\\n",
       "2371   SU2C                      Preclinical   \n",
       "2483   SU2C                      Preclinical   \n",
       "2492   SU2C                      Preclinical   \n",
       "2503   SU2C                      Preclinical   \n",
       "2531   SU2C                      Preclinical   \n",
       "2550   SU2C                      Preclinical   \n",
       "2556   SU2C                      Preclinical   \n",
       "2564   SU2C                      Preclinical   \n",
       "2589   SU2C                      Preclinical   \n",
       "2595   SU2C                      Preclinical   \n",
       "2614   SU2C                      Preclinical   \n",
       "2627   SU2C                      Preclinical   \n",
       "2661   SU2C                      Preclinical   \n",
       "2670   SU2C                      Preclinical   \n",
       "2676   SU2C                      Preclinical   \n",
       "2690   SU2C                      Preclinical   \n",
       "2698   SU2C                      Preclinical   \n",
       "2707   SU2C                      Preclinical   \n",
       "2722   SU2C                      Preclinical   \n",
       "3107   SU2C                      Preclinical   \n",
       "3205   SU2C                      Preclinical   \n",
       "3290   SU2C                      Preclinical   \n",
       "3353   SU2C                      Preclinical   \n",
       "3377   SU2C                      Preclinical   \n",
       "3390   SU2C                      Preclinical   \n",
       "3403   SU2C                      Preclinical   \n",
       "3424   SU2C                      Preclinical   \n",
       "3503   SU2C                      Preclinical   \n",
       "3514   SU2C                      Preclinical   \n",
       "3548   SU2C                      Preclinical   \n",
       "3555   SU2C                      Preclinical   \n",
       "3591   SU2C                      Preclinical   \n",
       "3624   SU2C                      Preclinical   \n",
       "3653   SU2C                      Preclinical   \n",
       "3660   SU2C                      Preclinical   \n",
       "3672   SU2C                      Preclinical   \n",
       "3693   SU2C                      Preclinical   \n",
       "3709   SU2C                      Preclinical   \n",
       "3809   SU2C                      Preclinical   \n",
       "3843   SU2C                      Preclinical   \n",
       "3902   SU2C                      Preclinical   \n",
       "3935   SU2C                      Preclinical   \n",
       "3943   SU2C                      Preclinical   \n",
       "3955   SU2C                      Preclinical   \n",
       "3965   SU2C                      Preclinical   \n",
       "3996   SU2C                      Preclinical   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "2371                               NaN                 Clinical evidence   \n",
       "2483                               NaN                 Clinical evidence   \n",
       "2492                               NaN                 Clinical evidence   \n",
       "2503                               NaN                 Clinical evidence   \n",
       "2531                               NaN                 Clinical evidence   \n",
       "2550                               NaN                 Clinical evidence   \n",
       "2556                               NaN                 Clinical evidence   \n",
       "2564                               NaN                 Clinical evidence   \n",
       "2589                               NaN                 Clinical evidence   \n",
       "2595                               NaN                 Clinical evidence   \n",
       "2614                               NaN                 Clinical evidence   \n",
       "2627                               NaN                 Clinical evidence   \n",
       "2661                               NaN                 Clinical evidence   \n",
       "2670                               NaN                 Clinical evidence   \n",
       "2676                               NaN                 Clinical evidence   \n",
       "2690                               NaN                 Clinical evidence   \n",
       "2698                               NaN                 Clinical evidence   \n",
       "2707                               NaN                 Clinical evidence   \n",
       "2722                               NaN                 Clinical evidence   \n",
       "3107                               NaN                 Clinical evidence   \n",
       "3205                               NaN                 Clinical evidence   \n",
       "3290                               NaN                 Clinical evidence   \n",
       "3353                               NaN                 Clinical evidence   \n",
       "3377                               NaN                 Clinical evidence   \n",
       "3390                               NaN                 Clinical evidence   \n",
       "3403                               NaN                 Clinical evidence   \n",
       "3424                               NaN                 Clinical evidence   \n",
       "3503                               NaN                 Clinical evidence   \n",
       "3514                               NaN                 Clinical evidence   \n",
       "3548                               NaN                 Clinical evidence   \n",
       "3555                               NaN                 Clinical evidence   \n",
       "3591                               NaN                 Clinical evidence   \n",
       "3624                               NaN                 Clinical evidence   \n",
       "3653                               NaN                 Clinical evidence   \n",
       "3660                               NaN                 Clinical evidence   \n",
       "3672                               NaN                 Clinical evidence   \n",
       "3693                               NaN                 Clinical evidence   \n",
       "3709                               NaN                 Clinical evidence   \n",
       "3809                               NaN                 Clinical evidence   \n",
       "3843                               NaN                 Clinical evidence   \n",
       "3902                               NaN                 Clinical evidence   \n",
       "3935                               NaN                 Clinical evidence   \n",
       "3943                               NaN                 Clinical evidence   \n",
       "3955                               NaN                 Clinical evidence   \n",
       "3965                               NaN                 Clinical evidence   \n",
       "3996                               NaN                 Clinical evidence   \n",
       "\n",
       "      clinvar   feature_type feature_str_simple  \n",
       "2371      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2483      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2492      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2503      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2531      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2550      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2556      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2564      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2589      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2595      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2614      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2627      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2661      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2670      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2676      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2690      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2698      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2707      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "2722      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3107      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3205      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3290      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3353      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3377      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3390      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3403      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3424      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3503      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3514      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3548      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3555      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3591      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3624      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3653      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3660      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3672      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3693      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3709      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3809      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3843      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3902      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3935      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3943      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3955      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3965      NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3996      NaN  Rearrangement       TMPRSS2--ERG  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_string'].eq('TMPRSS2--ERG')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MEL 44\n",
      "SU2C 4\n"
     ]
    }
   ],
   "source": [
    "tmb = tmp[tmp['feature_string'].eq('High Mutational Burden')]\n",
    "for label, group in tmb.groupby('cohort'):\n",
    "    print(label, group['patient_id'].drop_duplicates().shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(116,)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_type'].eq('Mutational signature')]['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "180"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_type'].eq('Aneuploidy')]['patient_id'].drop_duplicates().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.0\n",
      "count    315.000000\n",
      "mean       1.533333\n",
      "std        1.450236\n",
      "min        1.000000\n",
      "25%        1.000000\n",
      "50%        1.000000\n",
      "75%        2.000000\n",
      "max       23.000000\n",
      "Name: patient_id, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "# Range of additional considerations per patient\n",
    "print(tmp['patient_id'].value_counts().median())\n",
    "print(tmp['patient_id'].value_counts().describe())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.loc[tmp['cohort'].isin(['MEL', 'SU2C']), 'patient_id'].value_counts().median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    210.000000\n",
       "mean       1.676190\n",
       "std        1.725104\n",
       "min        1.000000\n",
       "25%        1.000000\n",
       "50%        1.000000\n",
       "75%        2.000000\n",
       "max       23.000000\n",
       "Name: patient_id, dtype: float64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.loc[tmp['cohort'].isin(['MEL', 'SU2C']), 'patient_id'].value_counts().describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAOAAAAEoCAYAAABB4f2MAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3Xt8XHWd//HXm1jSQhHlDgqUmxgSQGhBhMi9XGT1p4K4BRUw4iK2eIHlYgQKbCsgRVDQIhaLCxtdQX+CoEBNChtBoEXdIgUUKLcKFOXSKy3ls398v9OeDJPMN5OZTDLzeT4e59GZc86c80k6n5xzvleZGc656lin2gE4V888AZ2rIk9A56rIE9C5KvIEdK6KPAGdqyJPQOeqyBPQuSryBHSuit5R7QDKZZNNNrExY8ZUOwznAJg7d+7LZrZpsf1qJgHHjBnDnDlzqh2GcwBIejplP78Fda6KPAGdqyJPQOeqKCkBJR0g6YOZ9ydK6pZ0jaTRlQvPudqWegW8AtgCQNLOwDXA/wIfAr5dmdCcq32pCbgDMC++Phq4y8xOBU4GPlqJwJyrB6kJaEBDfH0I8Nv4+gVg43IH5QZfR0cHLS0tNDQ00NLSQkdHR7VDqgup9YAPAudKugv4MPDFuH4MIQndMNbR0UF7ezszZsygtbWV7u5u2traAJgwYUKVo6txZlZ0AVoIz3yvAedn1l8F3JhyjEovY8eONVea5uZm6+zs7LGus7PTmpubqxTR8AfMsYTvrWwAgzJJGgmsNrNVA/5LMEDjxo0zbwlTmoaGBlasWMGIESPWrFu1ahUjR45k9erVVYxs+JI018zGFduvX/WAksZJ+rSk9eOqBsLzoRvGmpqa6O7u7rGuu7ubpqamKkVUP1LrATeXdD/wAPBfwOZx0+XAtArF5gZJe3s7bW1tdHV1sWrVKrq6umhra6O9vb3aodW81EKY77C2xPOZzPqfA98rd1BucOUKWiZNmsT8+fNpampiypQpXgAzCFIT8BDgEDN7RVJ2/RPANmWPyg26CRMmeMJVQeoz4ChgZYH1mwIryheOc/UlNQHvAU7MvDdJDcBZwO/KHZRz9SL1FvRM4G5JewGNhIKXZmBDYL8KxeZczUu6AprZI8CuwH3AncBIQgHMHmb2ROXCc662JQ9JYWYvAOdVMBbn6k5yAkraEvgSsEtc9Qgw3cwWViIw5+pBakX8eEKVw6eBZXE5FvibpMMqF55ztS31Cvhd4EfAVyzTeFTSlcCVgLdZcq4EqdUQY4Cr7O0tt68Gti1rRM7VkdQEnEMoBc23K/DH8oXjXH1JvQX9PvAdSTsBf4jr9iEUypwtac/cjmb2UHlDdK52pSbgjfHfqX1sg55DVzjnikhNwO0qGoVzdSo1AZeZ2aKKRuJcHUothHle0s2SjlRefyTnXOlSE/Ao4A3gZuBZSRdJ2qFyYTlXH1IbY99lZscBWwHfAo4EHpfUKen4ODiTc66f+jUok5m9amZXx9GeTgP2Bf4TWCjpYp8nwrn+6dcEnbFB9gnAScB7gJ8CMwhXxnOAccChZY7RuZqVlICSPgl8HjgMeJjQ/vMGM3s9s8884E+VCNK5WpV6Bfwx0AF8yMzm9rLPU8CUskTlXJ1IfQbc0sxO6SP5MLPlZnZBb9slnSPpQUmvS1ok6VZJLXn7SNJkSQslLZc0W1JzYozODTupCbhY0mb5KyVtLCl17PIDCW1K9wUOBt4EZknaKLPPmcDpwCRgL+Al4C5JGySew7lhJfUWtLfK90YKD1f4NmZ2eI8DSp8lTPayH3BrrOD/KnCxmd0c9zmBkITHESYFda6m9JmAkr4eXxpwiqQlmc0NhKnKHi3x3BsQrsCvxPfbEWbhvTO3g5ktl3QP4arpCehqTrEr4KT4r4AvANnbzZXAAuCUEs99JaHU9L74fov474t5+71IqPJwrub0mYBmth2ApC7gk2b2Sl/7p5J0OdAKtJpZ/jNkfq97FVjnXE1IbYp2UBmT7zvABOBgM3sysyk30+4WeR/ZjLdfFZ2rCf1qijZQcRCn4wjJl//s+BQhCcdn9h9JeM68d9CCdG4Q9asp2kBIuhr4LPBx4BVJuSvdEjNbYmYm6QqgXdKjwOPAN4ElhDkJnas5g5aAwKnx3/zJXC4AJsfXlxJmYroaeDdwP3CYmS0ejACdG2yDloBmVrQjbxz2cDJrE9K5mpb8DBinqT5D0g8kbRLX7SfJx4txrkSpQ9OPBR4DjgfagHfGTePxBtjOlSz1CngZcKWZ7UEYmiLnDnx+QOdKlpqAY4HrC6z/O7B5+cJxrr6kJuByQqlkvvcTGks750qQmoC/As6X1Bjfm6QxwCWEkdKccyVITcAzgI2ARcB6QDfwN+BVQmW5c64ESfWAceyXVkkHA3sSEvchM5tVyeCcq3VFE1DSCMIV73Nm1gl0Vjwq5+pE0VtQM1tF6CzrXYKcK7PUZ8DrgZMrGYhz9Si1Lej6wPGSxgNzgaXZjWZ2WrkDc64epCZgE5Cb+Xb7vG1+a+pciVJLQQ+qdCDO1aNB7RHvnOspdW6IW/rabmYfK084ztWX1GfAf+S9HwHsDmwN/KKsETlXR1KfAU8qtF7SNMCHi3CuRAN9BryGtWO9OOf6aaAJuHNZonCuTqUWwnw3fxWwJWGu+OvKHZRz9SK1EGbXvPdvEbomfQ1PQOdK5hXxzlVR6qho1xWaJFPS+pL8CuhciVILYU4gjFidbxTwufKF41x9KTZB50aEAhcB75b0ZmZzA3AUPnORcyUr9gz4MqG3gwGPFNhuwPnlDsq5elEsAQ8iXP06gaOBf2a2rQSeNrOFFYrNuZpXbIbcuwHi/A/PmtlbgxKVc3UitRriaQBJWwHbAOvmbb+n/KE5V/tSW8JsRZgkc3/Cc1/+vO0N5Q/NudqXWg1xBbAa2AVYRpg2+lPAfOCIyoTmXO1LbYp2AHCUmT0qyYBFZvZ7SW8AFwF3VSxC52pY6hVwFKFKAkJJ6Gbx9SPAbuUOyrl6kZqAjxJmQgL4E3CKpG2BLwPPVyIw5+pB6i3olcAW8fWFwG+BCYTJOk+oQFzO1YXUaogbM68filOTvR94xsxe7u1zzrm+pV4B15C0OaEQ5qGiOzvn+pTaHWmEpEslLSY8842J6y+R5GPCOFei1EKY84GPAp8hPPflPACcmHoySftLukXS85JM0ol522fG9dnlD6nHd264Sb0FnQB83szulpRtD/ow8L5+nG90/MxP4lLILOCzmfcr+3F854aV1ATcCni6l88nP0ea2e3A7RCudr3s9oaZvZB6TOeGs9Rb0L8Q2oHmO5YwXVk5tUp6SdLjkq6VtFnxjzg3PKVevS4AbpC0NaHh9ackvR84jtArvlx+Sxjq/ilCQc9/AJ2SxprZG3190LnhKLUe8FZJxwLfIAxJeD5hvsCPmtmscgVjZj/NvJ0naS7h1vcofA4KV4N6TUBJ5wGXmdkySdsAd5rZHYMXGpjZQknPATsN5nmdGyx9PQOeRyi1hHBLuGnlw+lJ0ibAe4C/D/a5nRsMfd2CPg8cI+k2Qgfc90oaWWhHM3sm5WSSRgM7xrfrANtI+gChh8U/gcnAzYSEGwN8C3gJ+GXK8Z0bbvpKwCnAVcD3CL3fHyywT65nfGqP+HFAV+b9BXG5HvgSYQj8zwHvIiRhF3CsmfkUaK4m9ZqAZvZDSf9NuBI9ROj5nj9RZ7+Y2WxC0vbm8IEc37nhptioaK8Cf5J0EnC3VwU4V16p1RDXVzoQ5+rRQCfodM4NgCegc1XkCehcFXkCOldFxZqiJTGzC8sTjnP1pa9S0E/lvd8WWA/IzYa0FWGU7AWEkdKcc/3UV0X8rrnXsR7wc8AJuWZnsYH2j4EbCx/BOVdM6jPgecBXs20+4+vT8Qk6nStZagJuTuE54kcCm5QvHOfqS2oC3gVcK2kfSQ1x2Qe4Bp+YxbmSpSbgF4BngXuBFXH5PaHL0smVCc252pfaFnQR8BFJ7yMMSS9gvpk9XsngnKt1/Rqa3swel/QaYWh6ny/euQHyoemdq6JBHZreOdfTYA9N75zLSL0ClmVoeudcT0NxaHrn6sZQG5reubqSdAU0s1sJV7vDWDs0/U6UeWh65+pNf6YWuwMY1KHpnat13iPeuSrqq0f8W4RRr4sys9SRsZ1zGX3dgh7L2gTcnNDr/ZfAfXHdh4CP4/0BnStZXz3ib8q9lnQLcI6ZXZvZ5TpJDxCS8PuVC9G52pX6DHgwPSdVyekCDixbNM7VmdQEfBk4psD6Y4BF5QvHufqSWg1xHvBjSQex9hlwH+BQoK0SgTlXD1I75P5E0mPAacDHCB1yHwH2M7P7KxifczWtPxXx9wPHVzAW5+pOv3oySNoK2Iy8Z0cze6icQTlXL5ISUNIewA2sHQ8mqz9TVDvnMlKvgD8kjIp2MmFo+qQWMs65vqUm4C7AHj4KmnPllVoPOA/YopKBOFePUhPwG8Clkg6VtLmkjbJLJQN0rpal3oLmOt3eSc/nP+GFMM6VLDUBDyrHySTtD5wBjCUM9HSSmc3MbBehd8UXgXcD9wNfNrO/lOP8zg01qS1h7i7T+UYThjL8SVzynUmY8uxE4DFCE7i7JO1sZovLFINzQ0Zyj3hJu0q6StJvJG0Z13081hEmMbPbzewbsatTj6Ht49Xvq8DFZnazmT0MnABsQBj8ybmakzo0/WHAg8B7CF2TcnMF7kD5OuRuRyhpvTO3wsyWA/cA+5bpHM4NKalXwIuAr5vZJ4CVmfWzgb3LFEuumuPFvPUv4lUgrkalJmAzcHuB9f8Eyl0Nkd/KRgXWOVcTUhPwFcLtZ749gefKFMsL8d/8q91mvP2q6FxNSE3A/wK+Lem9hKvROyQdAFxG4dLMUjxFSMLxuRWSRgIfJszM61zNSa0H/CYwkzBBS64zrgiJOSX1ZJJGAzvGt+sA20j6APBPM3tG0hVAu6RHgcfjeZfE8zhXc1LrAVcBx0s6l3DbuQ7wRzP7az/PN46egztdEJfrCXV/lxJKWK9mbUX8YV4H6GpVan/A84DLzOxJ4MnM+lHAv5vZhSnHMbPZvL0/YXa7AZPj4lzN688MuaMLrF8PH5i3JnR0dNDS0kJDQwMtLS10dHRUO6S6kPoM2FtVwB6Eqgg3jHV0dNDe3s6MGTNobW2lu7ubtrYw2N2ECROqHF2NM7NeF2Ax8DqwOvM6tyyN66/u6xiDtYwdO9ZcaZqbm62zs7PHus7OTmtubq5SRMMfMMcSvrcK+xYm6QTC1e86QjvN1zKbVwILzOy+Qp8dbOPGjbM5c+ZUO4xhqaGhgRUrVjBixIg161atWsXIkSNZvXp1FSMbviTNNbNxxfbr8xbUzK6PB3sKuNdCaairMU1NTXR3d3PQQWt7nXV3d9PU1FTFqOpDaiHM2ELJJ2mkpB+UOSY3yNrb22lra6Orq4tVq1bR1dVFW1sb7e3t1Q6t5qUWwpwj6Qjgc2b2AqwZqrCDvG5FbvjJFbRMmjSJ+fPn09TUxJQpU7wAZhD0+Qy4ZqcwIO9PgN2BfyPMD38hoXXMVy10G6oqfwZ0Q0lZngFzzGwhcKikacDPCaWfnzazXw4sTOfqW396xB8NnEToA/gy8DVJ21QoLufqQmqP+BmEoekvMLNDgN0IXZT+LMknbHGuRKlXwL2Bvc3sSgAze9nM/h9wDjC9UsE5V+tSS0HHmdkb+SvNbLqkzjLH5FzdSLoC5pJP0jhJn5a0fny/PpneEc65/kntjrQ5cAuwF6FR9k6ExLscWAF8pVIBOlfLUp8Bv0MYLmJjYFlm/c+Bw8odlHP1IvUZ8BDgEDN7JYyfu8YTgFdFOFei1CvgKHqOB5qzKeEW1DlXgtQEvIcwZkuOSWoAzgJ+V+6gnKsXqbegZwJ3S9oLaASmEQbr3RDYr0KxOVfzUqshHgF2JYzPeScwklAAs4eZPVG58JyrbalXQGI3JB+AybkySm6M7ZwrP09A56rIE9C5Kuo1ASXtLyn5GdE51399XQG7iHP/SXpS0saDE5Jz9aOvBHyFMG00wJgi+zrnStDXLebNhMr3vxN6QMyRVHCUVjPbvhLBOVfr+krAUwhdkHYidDv6MWF4eudcmfSagHF8+9sAJO0OTDOfp8+5skodlvAkWDNl9I6EW9InzMx7Qjg3AKmjor1D0reJI6EB84BXJF0qaUTfn3bO9Sa1nu9SYALhubA7rvsw8C1CEp9R/tCcq32pCXgc8Hkzuz2z7glJi4Af4QnoXElS6/Y2JAw/ke8J4F3lC8e5+pKagH8GTiuw/ivAn8oXjnP1pT894m+XNB64j1AK+iFgK+DICsXmXM1L7RF/D/A+Qi/40cA74+udzay7r88653rXnx7xC4GKTpkqaTJv73X/opltUcnzOlctQ7GB9WPAlpll1+qGUx8mTZrEyJEjkcTIkSOZNGlStUOqC0MxAd80sxcyy6JqB1TrJk2axPTp05k6dSpLly5l6tSpTJ8+3ZNwMJjZkFmAyYSh758HngJ+Cmyf8tmxY8eaK01jY6NNmzatx7pp06ZZY2NjlSIa/oA5lvC9TZojfrBIOhLYAHgU2Az4JvB+oNnM/tHXZ32O+NJJYunSpay33npr1i1btoz111+fofT9GE5S54gv6RZU0ihJh0ratpTP98bMfmNm/21m/2tms4B/iTGeUM7zuJ4aGxuZPr3nPKvTp0+nsbGxShHVj9TG2DMlnRpfrws8QBig97F41aoIM1sC/IXQJ9FVyMknn8xZZ53F5ZdfzrJly7j88ss566yzOPnkk6sdWu1LuU8F/g7sGV8fAywg3CKeDdyfcoxSFsII3H8Hziu2rz8DDszEiROtsbHRAGtsbLSJEydWO6RhjXI+A0paAexoZs9J+hHwmpmdLmkMMM/MNijHHwNJlwG3As/EBD8X2B/Y1cye7uuz/gzohpJyPwO+ALTEGZEOB2bF9aOBVaWFWNB7gQ5CXeAvgDeAfYoln3PDVWpLmOuAnwELgdWsnZLsg4QSy7Iws38t17GcGw5S24JeCHwe+CHQama5yTrfBC6pUGxuEHV0dNDS0kJDQwMtLS10dHRUO6S6kHQFlLQ/8CszezNv043AvmWPyg2qjo4O2tvbmTFjBq2trXR3d9PW1gbAhAkTqhxdjUspqSHcdm5WYP3GwOqUY1R68VLQ0jU3N1tnZ2ePdZ2dndbc3FyliIY/EktBUwthROgDmG9jYGkZ/g64Kpo/fz6tra091rW2tjJ//vwqRVQ/+rwFlXRLfGnADZLeyGxuAFoIs+a6YaypqYnu7m4OOuigNeu6u7tpamqqYlT1odgV8B9xEWFIwn9klueA6cBnKhmgq7z29nba2tro6upi1apVdHV10dbWRnt7Rbt/OopcAW3tgLwLgMvMzG83a1CuoGXSpEnMnz+fpqYmpkyZ4gUwg2BI9YYYCG8J44aS1JYwqdUQGwFTgEMITcR63Lqa2TtLCdK5epfaEmYGsAehIn4hhUtEnXP9lJqAhwDjzez+SgbjXL1JrQd8CVhSyUCcq0epCdgOXChpdCWDca7epN6CfpMwT/xLkp4mrwuSme1W5rhchUnq92dqpcR8KElNwJsqGoUbdL0l05izb2PBxUcNcjT1K3WG3AsqHYhz9WgoDszrXN1IrYhfTB91f14R71xpUp8BJ+a9H0GomD+a0ELGOVeC1GfA6wutl/QQoZL+e+UMyrl6MdBnwC7go+UIxLl6NNAE/Ffg5XIE4lw9Si2EmUfPQhgBmwMbAV+qQFzO1YVSK+LfAhYBs82sbOOCuvLb/YI7eW15/8ZOHnP2bcn7bjhqBH8+/7D+huUir4ivca8tX1XRli39SVb3dslzxANIOhjYhXA7+hczm12JoJyrF6nPgO8BfgmMJXTIBdhK0hzgE2a2sNcPO+d6lVoK+l3C4Lw7mtnWZrY1Yc6+1XGbc64Eqbeg44EDzeyp3Aoze1LSaaydqMU5108DrQd8qyxROFenUhPwd8B3JW2dWyFpG+BK/AroXMlSE/A0YD3gSUlPx4F6n4jrTqtQbM7VvNR6wGeBPSWNB95PaAnziJnN6vuTzrm+9Kse0MzuAu6qUCyuAjZoOptdrz+7gscH8CEsSlVsdqQjgR8Au5vZa3nbNgT+DHzRzO6sXIhuIOadMK/aIbg+FHsGnAh8Oz/5AOK6S4CvVCIw5+pBsQTcDejrOa8T2L184ThXX4ol4Kb0XddnhFlynXMlKJaAzxGugr3ZDXi+fOE4V1+KJeBtwEWSRuVvkLQecGHcp6wknSrpKUkrJM2V9OFyn8O5oaBYNcQU4Bjgr5K+B+Q63zYRCmgETC1nQJI+TWhhcyrQHf/9jaRdzOyZcp7LuWorNkX1S5L2JVRFTCUkHIRnvzuAU83sxTLH9HVgppldG99PknQEYeiLc8p8LueqqmhFvJk9DXxE0ruBHQlJ+Fcze6XcwUhal9Dn8LK8TXcC+5b7fM5VW3JLmJhwD1YwFoBNgAYg/6r6InBohc/t3KDrV1O0QZQ/DL4KrOth7ty5L8ep05wbCrZN2WmoJeDLhF72W+St34y3XxV7MLNNKxWUc5UypGZHMrOVwFxCD/ys8cC9gx+Rc5U11K6AAJcD/ynpAeD3wCnAVsD0qkblXAUMuQQ0s59J2pgwLfaWwMPAR2JprHM1RT7vt3PVM6SeAZ2rN56ANUDSryXNHOAxxkkySWP62Gc9STdJeq3Yvi6NJ2AeSTPjlyt/+UM/jjFb0lWVjLNKPg/sD7QSns+fLcdB4+/81+U41nAz5AphhohZwGfz1q2sRiBDzI7AfDMbsuNcSBphZv2bDqqazMyXzALMBH7dx/YDgFWEkcJz604BXge2j5+3vGVM3G8XQvetxcBLQAewRf65CcN8PA+8AvwYWC+zz3pxvyWExgnfiJ+ZmdlnXcJwIc8BSwlNCA/P+zmOIPRuWQH8D3BcNtYCP/fsvJ9pdsq5CE0LZwBPAcuBvwJnAuvE7ZML/L4OBMbE1+Py4jDgmPg6t88EwugMy4GJcdu+wN3Asvi7/AHwzsxx9gf+EH+PrwH3Ay2D/n2r9hd+qC3FEjDuM5Vw+7URYZjGpcAJcduGhEYD1xFa9GwRv4RbElr6XELozrUbcCvwQObLODN+Ga6N+xwGvAqckzn39+MX6nCgBfg5IfmzCXhj/HLtT/ijMJFwBd89bt86Jt73YvzHxgTqKwE3ij/TvfFn2ijxXCMI/Ub3iglzbPyZ2uL20cDPCKPt5X5f6/YzARcQus1tB7wX2DUm1umEOUw+CNwH3BQ/9w7CH7fLgB3i7+A4oMkTcGgk4JvxPzC7XJLZZwThL/0vgIeAn+UdYzZwVd66C4Hf5a17d/wC7Z0597PAOzL7XAvMynxZ3wCOz2wfHb/QM+P7HQjDiGyTd67/D3w/vp4KPE6shorrvtlXAsZ9riJe+VLP1ctxLs79TJmf+9d5+/QnAU/P2+cnwIy8dR+I+25G+GNiwAHV/r75M2Bh9wBfzFv3au6Fma2SdBzwF8Kt5MEJxxwL7C9pSYFtOxCuhBAGPH4zs20h4S94br91CX/Nc7EsiVOI5+xJHDhZUmY1jYTbNAhX1z9Y/GZG99F/KedC0inAFwgNlEcR/oCVs2HFnLz3Y4EdY+fuNWHEf3cws/tiqfEdkn5HmF7h5xYGoB5UnoCFLTOzvxXZZx9CKfK7CINXvdr37qxDeP47o8C2bEPz/AIEY21ptShunfiZvQoca3k/jpOi6LliElxB+LnvJdwufxn4RJFj5wYDWxOrpBG97Lu0QFw/Ar5TYN/nAczsJElXEJ6FPwZMkfRxM7ujSFxl5QlYglj/dRXhi3QEcKOkfTNXrpWE576shwjPP09b6aV0fyN80fcBnoyxrE94Fnwi7vNHwpd2CzPr6uU4jwBHS1LmKrhPCfGknKsVuN/M1lTLSNohb59Cv69F8d8tM+s+kBjXQ0BzsT+iZvZnwuDSl0j6DXACYaSHQeP1gIU1Stoib9kUQFIDcANwt5ldQ7i1ei9wfubzC4C9JY2RtImkdYCrCQU0P5P0QUnbSzpU0g8lbZASlJktIZQoXiJpvKRmQsFIQ2afxwkFIzMlHRPPM07SGZI+GXebTnh+ukLSzpKOIZTk9kviuR4nzCtypKSdJJ1LKEnOWgC0xFg2iVUJywmFO2dJao5Do+SPlNCbSwi//+mS9pC0o6R/kXQNgKTtJF0saV9J20o6iFAo9kh/fwcDVu2H0KG2ULgawYDn4vZzgReATTOfGU+4MrXG9+8jPFMto2c1xE7ATYQSuOXAY4SSyHUz584vjJgMPJx5vz6hkGEJ4fnzXN5eDTEifu5JwtXlBeAWYGxmn6Pi+VcQep0cTz8LYVLORXhmnRF/5lfj6/OABZljbEoYdmRxjOHAuL4pxrYMmAd8mMKFMOMKxDoO+C3hlndp/PyFcdvmhAK05wmFWs8AlwIjBvv75o2xnasivwV1roo8AZ2rIk9A56rIE9C5KvIEdAMmaYGkQg0MXBGegC6ZpMmSHi6waS9CI/HBOFdN8ZYwNaYa/eHMbFHxvVwhfgWskthrfrqkKyW9Epdvx1YzuX3WlXSJpOckLZX0oKTDM9sPjL31PyLpAUkrCd2UCp3PJE2UdJukZZKelvSZvH0ulvSYpOXxtvJSSSPjthMJrX2aM6MEnBi39bgFlbRhbOHzkqTFku6WNC6z/URJSyQdIunh+LN1Sdqu2LlqTrVbntTrQuiytJieffJeA76e2adYX7sDCS1B5hH6Dm5PpoVO3vkM+Afwb4SWOu2EBs/jMvucC+xHaGHyEUILkYvitlGEpmCPsrbf3qi4bQFwRnwtwrRytwF7E3rRX0RokbJl3OdEQsuhWXGf3QjtSu8odq5aW6oeQL0uMQEL9cnLNXlL6deXS8CjE85nwLV562YBN/TxmVOAv2XeTybTLC6zPpuABxOayY3K2+dPwJnx9Ykxnp0z24+Pf1zW6etctbb4M2B1FeqTd5Gkd5LY1y7K7w/Xm/w+f/cR2oQCEBtlf5Vw1RpNaOSd30uhmLGEYTMW5cU9kvBHJecNM3ss834hoV3pu4B/9vOcw5Yn4NCV0q8vJ78/XL9J2gf4KXAB8DXtQhaJAAABdElEQVRCw+mPkd4DIWcdQv/GQtOKv555/Wbettwforoql/AErK4PFuiTt9DMXpeU0teuv/YhdF/Kvp8fX+8HPG9mF+U2Sto27/OF+u3le4jQ2+AtM3tyALGmnGvY8wSsrq0IffK+TxhI6N+B/4DQ105Srq/d6YQv9kaE574nzewXJZzvk5IeJDx/HgMcwtrhLh4H3iPpeMKt6eGE0cayFgDbStqTUECz2MzeyNtnFqEL0a8kncnagpQjCOPA/E9irCnnGv6q/RBarwshCaYT+ti9SugvNw1oyOxTrK/dgYRbt00SzmeEUtTfEm5hnyGO5JbZ51uEnuhLCP3lvhS+Imu2N7K2P6MBJ8b1C4iFMPH9BsCVhJHWVhIGmvopYTwWCIUwS/LO3eNn6e1ctbZ4f8AqkTSbUMo3cZDOZ8CnzOymwTifS1NXD7zODTWegM5Vkd+COldFfgV0roo8AZ2rIk9A56rIE9C5KvIEdK6KPAGdq6L/A8LoSvLqrIBVAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 216x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "\n",
    "fig = plt.figure(figsize = (3, 4), constrained_layout=True)\n",
    "ax = plt.subplot()\n",
    "plt.tick_params(axis = 'both', which = 'both',\n",
    "            top = False, bottom = False, right = False, left = False, \n",
    "            labeltop = False, labelbottom = True, labelright = False, labelleft = True)\n",
    "\n",
    "p = tmp['patient_id'].value_counts().plot.box()\n",
    "plt.ylabel('Counts of extended feature types', fontsize=14)\n",
    "\n",
    "ax.set_xticklabels(['Extended features\\nper patient'], fontsize = 14)\n",
    "ax.tick_params(axis='y', which='major', labelsize=14)\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Therapeutic sensitivity by evidence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3       KIRP-2K-A9WE\n",
       "19      KIRP-2Z-A9J5\n",
       "28      KIRP-2Z-A9J7\n",
       "33      KIRP-2Z-A9J9\n",
       "52      KIRP-2Z-A9JG\n",
       "            ...     \n",
       "3998         TP_2060\n",
       "4000         TP_2060\n",
       "4009         TP_2061\n",
       "4019         TP_2061\n",
       "4028         TP_2064\n",
       "Name: patient_id, Length: 483, dtype: object"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['patient_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda3/envs/moalmanac-paper/lib/python3.7/site-packages/ipykernel_launcher.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  del sys.path[0]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "FDA-Approved         70\n",
       "Clinical evidence    16\n",
       "Preclinical          10\n",
       "Guideline             9\n",
       "Clinical trial        1\n",
       "Inferential           1\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp = df[df['sensitive_predictive_implication'].notnull()]\n",
    "tmp['sensitive_predictive_implication'].value_counts()\n",
    "\n",
    "evidence_map = {\n",
    "    'FDA-Approved': 0,\n",
    "    'Guideline': 1,\n",
    "    'Clinical trial': 2,\n",
    "    'Clinical evidence': 3,\n",
    "    'Preclinical': 4,\n",
    "    'Inferential': 5\n",
    "}\n",
    "\n",
    "tmp['evidence_map'] = tmp['sensitive_predictive_implication'].replace(evidence_map)\n",
    "tmp = (tmp\n",
    " .loc[:, ['patient_id', 'sensitive_predictive_implication', 'evidence_map']]\n",
    " .sort_values('evidence_map')\n",
    " .drop_duplicates(subset='patient_id', keep='first')\n",
    ")\n",
    "\n",
    "idx_mel = tmp['patient_id'].str.contains('MEL-IPI')\n",
    "idx_kirp = tmp['patient_id'].str.contains('KIRP')\n",
    "idx_os = tmp['patient_id'].str.contains('Osteosarcoma')\n",
    "idx_prad = tmp.index.difference(idx_mel[idx_mel].index.union(idx_kirp[idx_kirp].index).union(idx_os[idx_os].index))\n",
    "tmp_mel_vcs = tmp[idx_mel]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_mel_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "79"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_mel_vcs['FDA-Approved'] + tmp_mel_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Guideline         78\n",
       "FDA-Approved      29\n",
       "Preclinical       29\n",
       "Clinical trial     4\n",
       "Inferential        1\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_su2c_vcs = tmp.loc[idx_prad, :]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_su2c_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "107"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_su2c_vcs['FDA-Approved'] + tmp_su2c_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Guideline            20\n",
       "Inferential          18\n",
       "FDA-Approved         16\n",
       "Preclinical          16\n",
       "Clinical evidence     9\n",
       "Clinical trial        4\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_kirp_vcs = tmp[idx_kirp]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_kirp_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_kirp_vcs['FDA-Approved'] + tmp_kirp_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Preclinical          17\n",
       "Inferential           9\n",
       "FDA-Approved          5\n",
       "Guideline             4\n",
       "Clinical trial        2\n",
       "Clinical evidence     1\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_os_vcs = tmp[idx_os]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_os_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_os_vcs['FDA-Approved'] + tmp_os_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "231"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "tmp_mel_vcs['FDA-Approved'] + tmp_mel_vcs['Guideline'] +\n",
    "tmp_su2c_vcs['FDA-Approved'] + tmp_su2c_vcs['Guideline'] +\n",
    "tmp_kirp_vcs['FDA-Approved'] + tmp_kirp_vcs['Guideline'] + \n",
    "tmp_os_vcs['FDA-Approved'] + tmp_os_vcs['Guideline']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "419"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "110 + 150 + 100 + 59"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FDA-Approved         120\n",
       "Guideline            111\n",
       "Preclinical           72\n",
       "Inferential           29\n",
       "Clinical evidence     26\n",
       "Clinical trial        11\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['sensitive_predictive_implication'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "patients_fda_guideline = tmp[tmp['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline'])]['patient_id'].tolist()\n",
    "patients_all = tmp['patient_id'].tolist()\n",
    "patients_fda_guideline_clinical = tmp[tmp['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence'])]['patient_id'].tolist()\n",
    "patients_preclinical = tmp[tmp['sensitive_predictive_implication'].isin(['Preclinical'])]['patient_id'].tolist()\n",
    "patients_inferential = tmp[tmp['sensitive_predictive_implication'].isin(['Inferential'])]['patient_id'].tolist()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "patients_all = tmp['patient_id'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda3/envs/moalmanac-paper/lib/python3.7/site-packages/ipykernel_launcher.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  del sys.path[0]\n"
     ]
    }
   ],
   "source": [
    "tmp = df[df['sensitive_predictive_implication'].notnull() & df['feature_type'].isin(['Somatic variant', 'Somatic copy number'])]\n",
    "tmp['sensitive_predictive_implication'].value_counts()\n",
    "\n",
    "evidence_map = {\n",
    "    'FDA-Approved': 0,\n",
    "    'Guideline': 1,\n",
    "    'Clinical trial': 2,\n",
    "    'Clinical evidence': 3,\n",
    "    'Preclinical': 4,\n",
    "    'Inferential': 5\n",
    "}\n",
    "\n",
    "tmp['evidence_map'] = tmp['sensitive_predictive_implication'].replace(evidence_map)\n",
    "tmp = (tmp\n",
    " .loc[:, ['patient_id', 'sensitive_predictive_implication', 'evidence_map']]\n",
    " .sort_values('evidence_map')\n",
    " .drop_duplicates(subset='patient_id', keep='first')\n",
    ")\n",
    "\n",
    "patients_fda_guideline_somatic = (tmp[\n",
    "    tmp['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline'])]['patient_id'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['KIRP-2Z-A9JJ', 'MEL-IPI_Pat06', 'Osteosarcoma-CCM201',\n",
      "       'Osteosarcoma-IAS513', 'Osteosarcoma-RCP466', 'Osteosarcoma-WABS378',\n",
      "       'PROS01448-1115244', 'SC_9034', 'SC_9066', 'SC_9086', 'TP_2032',\n",
      "       'TP_2060'],\n",
      "      dtype='object')\n",
      "(12,)\n"
     ]
    }
   ],
   "source": [
    "tmp_patients = pd.Index(patients_fda_guideline).difference(pd.Index(patients_fda_guideline_somatic))\n",
    "print(tmp_patients)\n",
    "print(tmp_patients.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>KIRP-2Z-A9JJ</td>\n",
       "      <td>EML4--ALK</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>EML4--ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>MEL-IPI_Pat06</td>\n",
       "      <td>COL1A1--TGM2</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>MEL</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--TGM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2820</th>\n",
       "      <td>Osteosarcoma-CCM201</td>\n",
       "      <td>COL1A1--MMP9</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--MMP9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2846</th>\n",
       "      <td>Osteosarcoma-IAS513</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2931</th>\n",
       "      <td>Osteosarcoma-RCP466</td>\n",
       "      <td>COL1A1--USP6</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--USP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3067</th>\n",
       "      <td>Osteosarcoma-WABS378</td>\n",
       "      <td>ABL1--TTF1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>ABL1--TTF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3119</th>\n",
       "      <td>PROS01448-1115244</td>\n",
       "      <td>COL1A1--KLK3</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--KLK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3493</th>\n",
       "      <td>SC_9034</td>\n",
       "      <td>COL1A1--CPE</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--CPE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3703</th>\n",
       "      <td>SC_9066</td>\n",
       "      <td>CDK5RAP2--COL1A1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>CDK5RAP2--COL1A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3704</th>\n",
       "      <td>SC_9066</td>\n",
       "      <td>COL1A1--C1R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--C1R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3803</th>\n",
       "      <td>SC_9086</td>\n",
       "      <td>COL1A1--LRP1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--LRP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3807</th>\n",
       "      <td>SC_9086</td>\n",
       "      <td>MT-RNR2--COL1A1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>MT-RNR2--COL1A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3985</th>\n",
       "      <td>TP_2032</td>\n",
       "      <td>COL1A1--MT-CO2</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--MT-CO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3998</th>\n",
       "      <td>TP_2060</td>\n",
       "      <td>COL1A1--BCL2L2-PABPN1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--BCL2L2-PABPN1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                patient_id         feature_string                almanac_bin  \\\n",
       "65            KIRP-2Z-A9JJ              EML4--ALK      Putatively Actionable   \n",
       "580          MEL-IPI_Pat06           COL1A1--TGM2  Investigate Actionability   \n",
       "2820   Osteosarcoma-CCM201           COL1A1--MMP9  Investigate Actionability   \n",
       "2846   Osteosarcoma-IAS513             NTRK2--AUH  Investigate Actionability   \n",
       "2931   Osteosarcoma-RCP466           COL1A1--USP6  Investigate Actionability   \n",
       "3067  Osteosarcoma-WABS378             ABL1--TTF1  Investigate Actionability   \n",
       "3119     PROS01448-1115244           COL1A1--KLK3  Investigate Actionability   \n",
       "3493               SC_9034            COL1A1--CPE  Investigate Actionability   \n",
       "3703               SC_9066       CDK5RAP2--COL1A1  Investigate Actionability   \n",
       "3704               SC_9066            COL1A1--C1R  Investigate Actionability   \n",
       "3803               SC_9086           COL1A1--LRP1  Investigate Actionability   \n",
       "3807               SC_9086        MT-RNR2--COL1A1  Investigate Actionability   \n",
       "3985               TP_2032         COL1A1--MT-CO2  Investigate Actionability   \n",
       "3998               TP_2060  COL1A1--BCL2L2-PABPN1  Investigate Actionability   \n",
       "\n",
       "     phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "65         NaN    1   KIRP                        Guideline   \n",
       "580        NaN    1    MEL                        Guideline   \n",
       "2820       NaN    1     OS                        Guideline   \n",
       "2846       NaN    1     OS                     FDA-Approved   \n",
       "2931       NaN    1     OS                        Guideline   \n",
       "3067       NaN    1     OS                     FDA-Approved   \n",
       "3119       NaN    1   SU2C                        Guideline   \n",
       "3493       NaN    1   SU2C                        Guideline   \n",
       "3703       NaN    1   SU2C                        Guideline   \n",
       "3704       NaN    1   SU2C                        Guideline   \n",
       "3803       NaN    1   SU2C                        Guideline   \n",
       "3807       NaN    1   SU2C                        Guideline   \n",
       "3985       NaN    1   SU2C                        Guideline   \n",
       "3998       NaN    1   SU2C                        Guideline   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "65                                 NaN                               NaN   \n",
       "580                                NaN                               NaN   \n",
       "2820                               NaN                               NaN   \n",
       "2846                               NaN                               NaN   \n",
       "2931                               NaN                               NaN   \n",
       "3067                               NaN                               NaN   \n",
       "3119                               NaN                               NaN   \n",
       "3493                               NaN                               NaN   \n",
       "3703                               NaN                               NaN   \n",
       "3704                               NaN                               NaN   \n",
       "3803                               NaN                               NaN   \n",
       "3807                               NaN                               NaN   \n",
       "3985                               NaN                               NaN   \n",
       "3998                               NaN                               NaN   \n",
       "\n",
       "      clinvar   feature_type     feature_str_simple  \n",
       "65        NaN  Rearrangement              EML4--ALK  \n",
       "580       NaN  Rearrangement           COL1A1--TGM2  \n",
       "2820      NaN  Rearrangement           COL1A1--MMP9  \n",
       "2846      NaN  Rearrangement             NTRK2--AUH  \n",
       "2931      NaN  Rearrangement           COL1A1--USP6  \n",
       "3067      NaN  Rearrangement             ABL1--TTF1  \n",
       "3119      NaN  Rearrangement           COL1A1--KLK3  \n",
       "3493      NaN  Rearrangement            COL1A1--CPE  \n",
       "3703      NaN  Rearrangement       CDK5RAP2--COL1A1  \n",
       "3704      NaN  Rearrangement            COL1A1--C1R  \n",
       "3803      NaN  Rearrangement           COL1A1--LRP1  \n",
       "3807      NaN  Rearrangement        MT-RNR2--COL1A1  \n",
       "3985      NaN  Rearrangement         COL1A1--MT-CO2  \n",
       "3998      NaN  Rearrangement  COL1A1--BCL2L2-PABPN1  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp = (df[df['patient_id'].isin(tmp_patients) \n",
    "          & df['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline'])])\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "COL1A1--C1R              1\n",
       "COL1A1--LRP1             1\n",
       "COL1A1--TGM2             1\n",
       "COL1A1--KLK3             1\n",
       "CDK5RAP2--COL1A1         1\n",
       "MT-RNR2--COL1A1          1\n",
       "COL1A1--BCL2L2-PABPN1    1\n",
       "ABL1--TTF1               1\n",
       "COL1A1--CPE              1\n",
       "EML4--ALK                1\n",
       "NTRK2--AUH               1\n",
       "COL1A1--USP6             1\n",
       "COL1A1--MT-CO2           1\n",
       "COL1A1--MMP9             1\n",
       "Name: feature_string, dtype: int64"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['feature_string'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2846</th>\n",
       "      <td>Osteosarcoma-IAS513</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               patient_id feature_string                almanac_bin phial_bin  \\\n",
       "2846  Osteosarcoma-IAS513     NTRK2--AUH  Investigate Actionability       NaN   \n",
       "\n",
       "      sum cohort sensitive_predictive_implication  \\\n",
       "2846    1     OS                     FDA-Approved   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "2846                               NaN                               NaN   \n",
       "\n",
       "      clinvar   feature_type feature_str_simple  \n",
       "2846      NaN  Rearrangement         NTRK2--AUH  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_string'].str.contains('NTRK')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(72,)"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(pd.Index(patients_preclinical)\n",
    " .difference(pd.Index(patients_fda_guideline_clinical))\n",
    " .shape\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(29,)"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(pd.Index(patients_inferential)\n",
    " .difference(pd.Index(patients_fda_guideline_clinical))\n",
    " .difference(pd.Index(patients_preclinical))\n",
    " .shape\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PTEN.Deletion          18\n",
       "CDKN2A.Deletion        14\n",
       "TMPRSS2--ERG           11\n",
       "AURKB.Amplification     4\n",
       "FGFR2.Deletion          3\n",
       "Name: feature_string, dtype: int64"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(df[df['patient_id'].isin(patients_preclinical) & df['sensitive_predictive_implication'].eq('Preclinical')]['feature_string']\n",
    " .value_counts()\n",
    " .head()\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table of sample counts by assertion type, evidence, and match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Somatic variant        17\n",
       "Somatic copy number     3\n",
       "Name: feature_type, dtype: int64"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(df[(df['cohort'].eq('KIRP') &\n",
    " df['almanac_bin'].isin(['Investigate Actionability', 'Putatively Actionable']) &\n",
    " df['prognostic_predictive_implication'].isin(['Guideline'])\n",
    ")]\n",
    "#.shape\n",
    "#['patient_id'].value_counts()\n",
    "['feature_type'].value_counts()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       KIRP-2K-A9WE\n",
       "1       KIRP-2K-A9WE\n",
       "2       KIRP-2K-A9WE\n",
       "3       KIRP-2K-A9WE\n",
       "4       KIRP-2K-A9WE\n",
       "            ...     \n",
       "4024         TP_2064\n",
       "4025         TP_2064\n",
       "4026         TP_2064\n",
       "4027         TP_2064\n",
       "4028         TP_2064\n",
       "Name: patient_id, Length: 4029, dtype: object"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['patient_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "columns = df.columns[df.columns.str.contains('predictive_implication')]\n",
    "\n",
    "evidence_columns = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence', 'Preclinical', 'Inferential']\n",
    "match_columns = ['Putatively Actionable', 'Investigate Actionability', 'Biologically Relevant']\n",
    "types_columns = ['Somatic variant', 'Somatic copy number', 'Germline', 'Rearrangement', 'Mutational signature', 'Tumor mutational burden', 'Aneuploidy']\n",
    "\n",
    "outname = 'retrospective.feature_types_by_evidence.xlsx'\n",
    "\n",
    "with pd.ExcelWriter(outname) as writer: \n",
    "    for label, group in df.groupby('cohort'):\n",
    "        for column in columns:\n",
    "            sheetname = '{} {} profile counts'.format(label, column.split('_')[0])\n",
    "            (\n",
    "                df[df[column].isin(evidence_columns) & df['cohort'].eq(label)]\n",
    "                .loc[:, ['patient_id', 'feature_type', column]]\n",
    "                .drop_duplicates()\n",
    "                .pivot_table(index='feature_type', columns=column, values='patient_id', aggfunc=lambda x: len(x.unique()))\n",
    "                .reindex(evidence_columns, axis=1)\n",
    "                .reindex(types_columns)\n",
    "                .fillna(0)\n",
    "                .astype(int)\n",
    "                .to_excel(writer, sheet_name=sheetname)\n",
    "            )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## In conclusion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP 85 0.85\n",
      "MEL 109 0.991\n",
      "OS 50 0.847\n",
      "SU2C 149 0.993\n"
     ]
    }
   ],
   "source": [
    "n_size = {\n",
    "    'KIRP': 100,\n",
    "    'MEL': 110,\n",
    "    'SU2C': 150,\n",
    "    'OS': 59\n",
    "}\n",
    "\n",
    "tmp = df[df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])]\n",
    "for label, group in tmp.groupby('cohort'):\n",
    "    print(label, \n",
    "          group['patient_id'].drop_duplicates().shape[0], \n",
    "          round(group['patient_id'].drop_duplicates().shape[0] / n_size[label], 3)\n",
    "         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP 27 0.27\n",
      "MEL 101 0.918\n",
      "OS 36 0.61\n",
      "SU2C 131 0.873\n"
     ]
    }
   ],
   "source": [
    "tmp = df[df['phial_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])]\n",
    "for label, group in tmp.groupby('cohort'):\n",
    "    print(label, \n",
    "          group['patient_id'].drop_duplicates().shape[0], \n",
    "          round(group['patient_id'].drop_duplicates().shape[0] / n_size[label], 3)\n",
    "         )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2 caption"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(365,)\n",
      "(295,)\n"
     ]
    }
   ],
   "source": [
    "idx_v = df['feature_type'].eq('Somatic variant')\n",
    "idx_cn = df['feature_type'].eq('Somatic copy number')\n",
    "idx_a = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_p = df['phial_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "\n",
    "idx = (idx_a) & (idx_v | idx_cn)\n",
    "tmp = df[idx]['patient_id'].drop_duplicates()\n",
    "print(tmp.shape)\n",
    "\n",
    "idx = (idx_p) & (idx_v | idx_cn)\n",
    "tmp = df[idx]['patient_id'].drop_duplicates()\n",
    "print(tmp.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>KIRP-2Z-A9J3</td>\n",
       "      <td>FGFR1.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>FGFR1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>KIRP-2Z-A9JE</td>\n",
       "      <td>SMARCA4.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>SMARCA4 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>KIRP-2Z-A9JG</td>\n",
       "      <td>ABL1.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ABL1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>KIRP-2Z-A9JP</td>\n",
       "      <td>SMARCA4.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>SMARCA4 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>KIRP-2Z-A9JR</td>\n",
       "      <td>EWSR1.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>EWSR1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3963</th>\n",
       "      <td>TP_2001</td>\n",
       "      <td>MITF.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MITF Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3976</th>\n",
       "      <td>TP_2010</td>\n",
       "      <td>TMPRSS2.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TMPRSS2 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3979</th>\n",
       "      <td>TP_2020</td>\n",
       "      <td>AR.p.Q58L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>AR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3983</th>\n",
       "      <td>TP_2020</td>\n",
       "      <td>NOTCH2.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>NOTCH2 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4011</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>MAP2K4.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MAP2K4 Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>225 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id    feature_string almanac_bin                  phial_bin  \\\n",
       "15    KIRP-2Z-A9J3    FGFR1.Deletion         NaN  Investigate Actionability   \n",
       "49    KIRP-2Z-A9JE  SMARCA4.Deletion         NaN  Investigate Actionability   \n",
       "50    KIRP-2Z-A9JG     ABL1.Deletion         NaN  Investigate Actionability   \n",
       "89    KIRP-2Z-A9JP  SMARCA4.Deletion         NaN  Investigate Actionability   \n",
       "107   KIRP-2Z-A9JR    EWSR1.Deletion         NaN  Investigate Actionability   \n",
       "...            ...               ...         ...                        ...   \n",
       "3963       TP_2001     MITF.Deletion         NaN  Investigate Actionability   \n",
       "3976       TP_2010  TMPRSS2.Deletion         NaN  Investigate Actionability   \n",
       "3979       TP_2020         AR.p.Q58L         NaN  Investigate Actionability   \n",
       "3983       TP_2020   NOTCH2.Deletion         NaN  Investigate Actionability   \n",
       "4011       TP_2061   MAP2K4.Deletion         NaN  Investigate Actionability   \n",
       "\n",
       "      sum cohort sensitive_predictive_implication  \\\n",
       "15      1   KIRP                              NaN   \n",
       "49      1   KIRP                              NaN   \n",
       "50      1   KIRP                              NaN   \n",
       "89      1   KIRP                              NaN   \n",
       "107     1   KIRP                              NaN   \n",
       "...   ...    ...                              ...   \n",
       "3963    1   SU2C                              NaN   \n",
       "3976    1   SU2C                              NaN   \n",
       "3979    1   SU2C                              NaN   \n",
       "3983    1   SU2C                              NaN   \n",
       "4011    1   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "15                                 NaN                               NaN   \n",
       "49                                 NaN                               NaN   \n",
       "50                                 NaN                               NaN   \n",
       "89                                 NaN                               NaN   \n",
       "107                                NaN                               NaN   \n",
       "...                                ...                               ...   \n",
       "3963                               NaN                               NaN   \n",
       "3976                               NaN                               NaN   \n",
       "3979                               NaN                               NaN   \n",
       "3983                               NaN                               NaN   \n",
       "4011                               NaN                               NaN   \n",
       "\n",
       "      clinvar         feature_type feature_str_simple  \n",
       "15        NaN  Somatic copy number          FGFR1 Del  \n",
       "49        NaN  Somatic copy number        SMARCA4 Del  \n",
       "50        NaN  Somatic copy number           ABL1 Del  \n",
       "89        NaN  Somatic copy number        SMARCA4 Del  \n",
       "107       NaN  Somatic copy number          EWSR1 Del  \n",
       "...       ...                  ...                ...  \n",
       "3963      NaN  Somatic copy number           MITF Del  \n",
       "3976      NaN  Somatic copy number        TMPRSS2 Del  \n",
       "3979      NaN      Somatic variant                 AR  \n",
       "3983      NaN  Somatic copy number         NOTCH2 Del  \n",
       "4011      NaN  Somatic copy number         MAP2K4 Del  \n",
       "\n",
       "[225 rows x 12 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[idx & df['almanac_bin'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_patients_preclinical = (pd.Index(patients_preclinical)\n",
    "                               .difference(pd.Index(patients_fda_guideline_clinical))\n",
    "                              )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SU2C    29\n",
       "OS      17\n",
       "KIRP    16\n",
       "MEL     10\n",
       "Name: cohort, dtype: int64"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    df[df['patient_id'].isin(\n",
    "        pd.Index(patients_preclinical)\n",
    "        .difference(pd.Index(patients_fda_guideline_clinical))\n",
    "    )]\n",
    " .loc[:, ['patient_id', 'cohort']]\n",
    " .drop_duplicates()\n",
    " .loc[:, 'cohort']\n",
    " .value_counts()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>ASXL1.Amplification</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ASXL1 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4024</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>COSMIC Signature 1</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4025</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>COSMIC Signature 6</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4026</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>MPL.p.A134T</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>MPL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4027</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>TP53.p.H179R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4028</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>Whole genome doubling</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aneuploidy</td>\n",
       "      <td>WGD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4029 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id         feature_string                almanac_bin  \\\n",
       "0     KIRP-2K-A9WE    ASXL1.Amplification      Biologically Relevant   \n",
       "1     KIRP-2K-A9WE    AURKA.Amplification      Putatively Actionable   \n",
       "2     KIRP-2K-A9WE    COSMIC Signature 12      Biologically Relevant   \n",
       "3     KIRP-2K-A9WE     COSMIC Signature 4  Investigate Actionability   \n",
       "4     KIRP-2K-A9WE     TPX2.Amplification      Putatively Actionable   \n",
       "...            ...                    ...                        ...   \n",
       "4024       TP_2064     COSMIC Signature 1      Biologically Relevant   \n",
       "4025       TP_2064     COSMIC Signature 6      Biologically Relevant   \n",
       "4026       TP_2064            MPL.p.A134T  Investigate Actionability   \n",
       "4027       TP_2064           TP53.p.H179R  Investigate Actionability   \n",
       "4028       TP_2064  Whole genome doubling  Investigate Actionability   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "0                           NaN    1   KIRP                              NaN   \n",
       "1                           NaN    1   KIRP                      Preclinical   \n",
       "2                           NaN    1   KIRP                              NaN   \n",
       "3                           NaN    1   KIRP                      Inferential   \n",
       "4                           NaN    1   KIRP                              NaN   \n",
       "...                         ...  ...    ...                              ...   \n",
       "4024                        NaN    1   SU2C                              NaN   \n",
       "4025                        NaN    1   SU2C                              NaN   \n",
       "4026      Biologically Relevant    2   SU2C                      Preclinical   \n",
       "4027  Investigate Actionability    2   SU2C                              NaN   \n",
       "4028                        NaN    1   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "0                                  NaN                               NaN   \n",
       "1                                  NaN                       Preclinical   \n",
       "2                                  NaN                               NaN   \n",
       "3                                  NaN                               NaN   \n",
       "4                                  NaN                       Preclinical   \n",
       "...                                ...                               ...   \n",
       "4024                               NaN                               NaN   \n",
       "4025                               NaN                               NaN   \n",
       "4026                               NaN                               NaN   \n",
       "4027                         Guideline                         Guideline   \n",
       "4028                               NaN                       Inferential   \n",
       "\n",
       "      clinvar          feature_type   feature_str_simple  \n",
       "0         NaN   Somatic copy number            ASXL1 Amp  \n",
       "1         NaN   Somatic copy number            AURKA Amp  \n",
       "2         NaN  Mutational signature  COSMIC Signature 12  \n",
       "3         NaN  Mutational signature   COSMIC Signature 4  \n",
       "4         NaN   Somatic copy number             TPX2 Amp  \n",
       "...       ...                   ...                  ...  \n",
       "4024      NaN  Mutational signature   COSMIC Signature 1  \n",
       "4025      NaN  Mutational signature   COSMIC Signature 6  \n",
       "4026      NaN       Somatic variant                  MPL  \n",
       "4027      NaN       Somatic variant                 TP53  \n",
       "4028      NaN            Aneuploidy                  WGD  \n",
       "\n",
       "[4029 rows x 12 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p ../../tables/supplementary-table-03\n",
    "\n",
    "outname = '../../tables/supplementary-table-03/supplementary-table-3.xlsx'\n",
    "with pd.ExcelWriter(outname) as writer:\n",
    "    df.drop('sum', axis=1).to_excel(writer, sheet_name='MOAlmanac & PHIAL calls', index=False)\n",
    "    counts = pd.read_csv('retrospective.counts-by-category.txt', sep='\\t')\n",
    "    counts.to_excel(writer, sheet_name='summarized counts', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "moalmanac-paper",
   "language": "python",
   "name": "moalmanac-paper"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
