{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculations for retrospective cohorts\n",
    "We perform calculations explicitly mentioned in or related to the paper's section titled, \"Evaluating expanded molecular profiling and actionability in two retrospective cohorts\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob as glob\n",
    "\n",
    "redact_germline = True"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load MOAlmanac outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AR.p.QQQQQ73del.Germline</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Germline</td>\n",
       "      <td>AR Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>ASXL1.Amplification</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ASXL1 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     patient_id            feature_string                almanac_bin  \\\n",
       "0  KIRP-2K-A9WE  AR.p.QQQQQ73del.Germline      Biologically Relevant   \n",
       "1  KIRP-2K-A9WE       ASXL1.Amplification      Biologically Relevant   \n",
       "2  KIRP-2K-A9WE       AURKA.Amplification      Putatively Actionable   \n",
       "3  KIRP-2K-A9WE       COSMIC Signature 12      Biologically Relevant   \n",
       "4  KIRP-2K-A9WE        COSMIC Signature 4  Investigate Actionability   \n",
       "\n",
       "  phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "0       NaN    1   KIRP                              NaN   \n",
       "1       NaN    1   KIRP                              NaN   \n",
       "2       NaN    1   KIRP                      Preclinical   \n",
       "3       NaN    1   KIRP                              NaN   \n",
       "4       NaN    1   KIRP                      Inferential   \n",
       "\n",
       "  resistance_predictive_implication prognostic_predictive_implication clinvar  \\\n",
       "0                               NaN                               NaN     NaN   \n",
       "1                               NaN                               NaN     NaN   \n",
       "2                               NaN                       Preclinical     NaN   \n",
       "3                               NaN                               NaN     NaN   \n",
       "4                               NaN                               NaN     NaN   \n",
       "\n",
       "           feature_type   feature_str_simple  \n",
       "0              Germline          AR Germline  \n",
       "1   Somatic copy number            ASXL1 Amp  \n",
       "2   Somatic copy number            AURKA Amp  \n",
       "3  Mutational signature  COSMIC Signature 12  \n",
       "4  Mutational signature   COSMIC Signature 4  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('retrospective.actionability.txt', sep='\\t')\n",
    "if redact_germline:\n",
    "    df = df[~df['feature_type'].eq('Germline')]\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        2519\n",
       "Investigate Actionability    1439\n",
       "Putatively Actionable         563\n",
       "Name: almanac_bin, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['almanac_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        924\n",
       "Investigate Actionability    657\n",
       "Putatively Actionable         81\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MEL     2113\n",
       "SU2C    1700\n",
       "KIRP     810\n",
       "OS       468\n",
       "Name: cohort, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['cohort'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How many patients had > 0 events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Almanac\n",
      "Mel 110 1.0\n",
      "SU2C 149 0.9933333333333333\n",
      "KIRP 85 0.85\n",
      "OS 51 0.864406779661017\n",
      "\n",
      "PHIAL\n",
      "Mel 101 0.9181818181818182\n",
      "SU2C 131 0.8733333333333333\n",
      "KIRP 27 0.27\n",
      "OS 36 0.6101694915254238\n"
     ]
    }
   ],
   "source": [
    "idx_almanac = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_phial = df['phial_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_mel = df['cohort'].eq('MEL')\n",
    "idx_su2c = df['cohort'].eq('SU2C')\n",
    "idx_kirp = df['cohort'].eq('KIRP')\n",
    "idx_os = df['cohort'].eq('OS')\n",
    "\n",
    "n_almanac_mel = df[idx_almanac & idx_mel]['patient_id'].drop_duplicates().shape[0]\n",
    "n_almanac_su2c = df[idx_almanac & idx_su2c]['patient_id'].drop_duplicates().shape[0]\n",
    "n_almanac_kirp = df[idx_almanac & idx_kirp]['patient_id'].drop_duplicates().shape[0]\n",
    "n_almanac_os = df[idx_almanac & idx_os]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_mel = df[idx_phial & idx_mel]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_su2c = df[idx_phial & idx_su2c]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_kirp = df[idx_phial & idx_kirp]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial_os = df[idx_phial & idx_os]['patient_id'].drop_duplicates().shape[0]\n",
    "\n",
    "print('Almanac')\n",
    "print('Mel', n_almanac_mel, n_almanac_mel/110)\n",
    "print('SU2C', n_almanac_su2c, n_almanac_su2c/150)\n",
    "print('KIRP', n_almanac_kirp, n_almanac_kirp/100)\n",
    "print('OS', n_almanac_os, n_almanac_os/59)\n",
    "print('')\n",
    "print('PHIAL')\n",
    "print('Mel', n_phial_mel, n_phial_mel/110)\n",
    "print('SU2C', n_phial_su2c, n_phial_su2c/150)\n",
    "print('KIRP', n_phial_kirp, n_phial_kirp/100)\n",
    "print('OS', n_phial_os, n_phial_os/59)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How many somatic variants and copy number alterations were called"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1445, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>DNMT3A.p.Q350fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      patient_id       feature_string                almanac_bin  \\\n",
       "2   KIRP-2K-A9WE  AURKA.Amplification      Putatively Actionable   \n",
       "7   KIRP-2K-A9WE   TPX2.Amplification      Putatively Actionable   \n",
       "9   KIRP-2Z-A9J1   BRAF.Amplification      Putatively Actionable   \n",
       "11  KIRP-2Z-A9J1      DNMT3A.p.Q350fs  Investigate Actionability   \n",
       "15  KIRP-2Z-A9J1    MET.Amplification      Putatively Actionable   \n",
       "\n",
       "                phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "2                     NaN    1   KIRP                      Preclinical   \n",
       "7                     NaN    1   KIRP                              NaN   \n",
       "9                     NaN    1   KIRP                              NaN   \n",
       "11  Biologically Relevant    2   KIRP                              NaN   \n",
       "15                    NaN    1   KIRP                        Guideline   \n",
       "\n",
       "   resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "2                                NaN                       Preclinical   \n",
       "7                                NaN                       Preclinical   \n",
       "9                  Clinical evidence                               NaN   \n",
       "11                               NaN                         Guideline   \n",
       "15                         Guideline                               NaN   \n",
       "\n",
       "   clinvar         feature_type feature_str_simple  \n",
       "2      NaN  Somatic copy number          AURKA Amp  \n",
       "7      NaN  Somatic copy number           TPX2 Amp  \n",
       "9      NaN  Somatic copy number           BRAF Amp  \n",
       "11     NaN      Somatic variant             DNMT3A  \n",
       "15     NaN  Somatic copy number            MET Amp  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx_v = df['feature_type'].eq('Somatic variant')\n",
    "idx_cn = df['feature_type'].eq('Somatic copy number')\n",
    "idx_a = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_b = df['almanac_bin'].eq('Biologically Relevant')\n",
    "\n",
    "idx = (idx_a) & (idx_v | idx_cn)\n",
    "tmp = df[idx]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "453\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Putatively Actionable         79\n",
       "Investigate Actionability    374\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(tmp['phial_bin'].value_counts().loc[['Putatively Actionable', 'Investigate Actionability']].sum())\n",
    "tmp['phial_bin'].value_counts().loc[['Putatively Actionable', 'Investigate Actionability']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_almanac_pa_ia = tmp['almanac_bin'].value_counts().sum()\n",
    "n_almanac_phial_pa_ia = tmp['phial_bin'].value_counts().loc[['Putatively Actionable', 'Investigate Actionability']].sum()\n",
    "fraction_almanac_phial_pa_ia = (n_almanac_phial_pa_ia / n_almanac_pa_ia)*100\n",
    "n_almanac_phial_br = tmp['phial_bin'].value_counts().loc['Biologically Relevant']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Of the 1445 somatic variants and copy number alterations noted by MOAlmanac, 453 (31.349480968858128%) were highlighted for putative actionability or investigate actionability by PHIAL, with an additional 390 noted by PHIAL for biological relevance.'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"Of the {n_almanac_pa_ia} somatic variants and copy number alterations noted by MOAlmanac, \\\n",
    "{n_almanac_phial_pa_ia} ({fraction_almanac_phial_pa_ia}%) were highlighted for putative actionability or \\\n",
    "investigate actionability by PHIAL, with an additional {n_almanac_phial_br} noted by PHIAL for biological relevance.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Restricting considered features\n",
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: FDA-approved, clinical trials, or clinical evidence  \n",
    "Matches considered: First putatively actionable and then also investigate actionability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(412, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>KIRP-2Z-A9JD</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>KIRP-2Z-A9JD</td>\n",
       "      <td>EGFR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>EGFR Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>KIRP-2Z-A9JD</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5070</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>PIK3CA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>PIK3CA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5071</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>PTEN.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>PTEN Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5072</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>RB1.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>RB1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5077</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>TP53.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TP53 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5080</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AR Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id        feature_string            almanac_bin  \\\n",
       "9     KIRP-2Z-A9J1    BRAF.Amplification  Putatively Actionable   \n",
       "15    KIRP-2Z-A9J1     MET.Amplification  Putatively Actionable   \n",
       "72    KIRP-2Z-A9JD    BRAF.Amplification  Putatively Actionable   \n",
       "76    KIRP-2Z-A9JD    EGFR.Amplification  Putatively Actionable   \n",
       "80    KIRP-2Z-A9JD     MET.Amplification  Putatively Actionable   \n",
       "...            ...                   ...                    ...   \n",
       "5070       TP_2061  PIK3CA.Amplification  Putatively Actionable   \n",
       "5071       TP_2061         PTEN.Deletion  Putatively Actionable   \n",
       "5072       TP_2061          RB1.Deletion  Putatively Actionable   \n",
       "5077       TP_2061         TP53.Deletion  Putatively Actionable   \n",
       "5080       TP_2064      AR.Amplification  Putatively Actionable   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "9                           NaN    1   KIRP                              NaN   \n",
       "15                          NaN    1   KIRP                        Guideline   \n",
       "72                          NaN    1   KIRP                              NaN   \n",
       "76                          NaN    1   KIRP                Clinical evidence   \n",
       "80                          NaN    1   KIRP                        Guideline   \n",
       "...                         ...  ...    ...                              ...   \n",
       "5070  Investigate Actionability    2   SU2C                   Clinical trial   \n",
       "5071  Investigate Actionability    2   SU2C                      Preclinical   \n",
       "5072  Investigate Actionability    2   SU2C                              NaN   \n",
       "5077  Investigate Actionability    2   SU2C                              NaN   \n",
       "5080  Investigate Actionability    2   SU2C                        Guideline   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "9                    Clinical evidence                               NaN   \n",
       "15                           Guideline                               NaN   \n",
       "72                   Clinical evidence                               NaN   \n",
       "76                                 NaN                               NaN   \n",
       "80                           Guideline                               NaN   \n",
       "...                                ...                               ...   \n",
       "5070                               NaN                               NaN   \n",
       "5071                 Clinical evidence                               NaN   \n",
       "5072                               NaN                         Guideline   \n",
       "5077                               NaN                         Guideline   \n",
       "5080                         Guideline                               NaN   \n",
       "\n",
       "     clinvar         feature_type feature_str_simple  \n",
       "9        NaN  Somatic copy number           BRAF Amp  \n",
       "15       NaN  Somatic copy number            MET Amp  \n",
       "72       NaN  Somatic copy number           BRAF Amp  \n",
       "76       NaN  Somatic copy number           EGFR Amp  \n",
       "80       NaN  Somatic copy number            MET Amp  \n",
       "...      ...                  ...                ...  \n",
       "5070     NaN  Somatic copy number         PIK3CA Amp  \n",
       "5071     NaN  Somatic copy number           PTEN Del  \n",
       "5072     NaN  Somatic copy number            RB1 Del  \n",
       "5077     NaN  Somatic copy number           TP53 Del  \n",
       "5080     NaN  Somatic copy number             AR Amp  \n",
       "\n",
       "[412 rows x 12 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "pa_patients = tmp['patient_id'].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Investigate Actionability    167\n",
       "Putatively Actionable         60\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_events = tmp.shape[0]\n",
    "n_patients = pa_patients.shape[0]\n",
    "n_patients_mel = tmp[tmp['cohort'].eq('MEL')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_patients_su2c = tmp[tmp['cohort'].eq('SU2C')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_patients_kirp = tmp[tmp['cohort'].eq('KIRP')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_patients_os = tmp[tmp['cohort'].eq('OS')]['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial = tmp['phial_bin'].dropna().shape[0]\n",
    "frac = round((n_phial / n_events)*100, 2)\n",
    "tmp['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MOAlmanac identified 412 such putatively actionable events from 253 (73 melanoma, 118 mCRPC, 37 KIRP, 25 OS), 227 (55.1%) of which were flagged by PHIAL for clinical relevance'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"MOAlmanac identified {n_events} such putatively actionable events from {n_patients} ({n_patients_mel} melanoma, \\\n",
    "{n_patients_su2c} mCRPC, {n_patients_kirp} KIRP, {n_patients_os} OS), {n_phial} ({frac}%) of which were flagged by PHIAL for clinical relevance\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP\n",
      "MET.Amplification       18\n",
      "BRAF.Amplification      16\n",
      "EGFR.Amplification      16\n",
      "ERBB2.Amplification      5\n",
      "PIK3CA.Amplification     3\n",
      "Name: feature_string, dtype: int64\n",
      "\n",
      "MEL\n",
      "BRAF.p.V600E         39\n",
      "PTEN.Deletion         9\n",
      "MET.Amplification     9\n",
      "NRAS.p.Q61R           8\n",
      "NRAS.p.Q61L           7\n",
      "Name: feature_string, dtype: int64\n",
      "\n",
      "OS\n",
      "RB1.Deletion          12\n",
      "TP53.Deletion          8\n",
      "MYC.Amplification      5\n",
      "PTEN.Deletion          3\n",
      "CDK4.Amplification     3\n",
      "Name: feature_string, dtype: int64\n",
      "\n",
      "SU2C\n",
      "AR.Amplification     82\n",
      "PTEN.Deletion        40\n",
      "RB1.Deletion         21\n",
      "MYC.Amplification    14\n",
      "TP53.Deletion        10\n",
      "Name: feature_string, dtype: int64\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for label, group in tmp.groupby('cohort'):\n",
    "    print(label)\n",
    "    print(group['feature_string'].value_counts().head())\n",
    "    print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1180, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>DNMT3A.p.Q350fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>KIRP-2Z-A9J2</td>\n",
       "      <td>ARID1A.p.CP1657fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>ARID1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>KIRP-2Z-A9J2</td>\n",
       "      <td>BCR.p.E306D</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>BCR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5072</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>RB1.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>RB1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5077</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>TP53.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TP53 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5080</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AR Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5085</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>BLM.p.V54I</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>BLM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5089</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>TP53.p.H179R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1180 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id      feature_string                almanac_bin  \\\n",
       "9     KIRP-2Z-A9J1  BRAF.Amplification      Putatively Actionable   \n",
       "11    KIRP-2Z-A9J1     DNMT3A.p.Q350fs  Investigate Actionability   \n",
       "15    KIRP-2Z-A9J1   MET.Amplification      Putatively Actionable   \n",
       "20    KIRP-2Z-A9J2   ARID1A.p.CP1657fs  Investigate Actionability   \n",
       "22    KIRP-2Z-A9J2         BCR.p.E306D  Investigate Actionability   \n",
       "...            ...                 ...                        ...   \n",
       "5072       TP_2061        RB1.Deletion      Putatively Actionable   \n",
       "5077       TP_2061       TP53.Deletion      Putatively Actionable   \n",
       "5080       TP_2064    AR.Amplification      Putatively Actionable   \n",
       "5085       TP_2064          BLM.p.V54I  Investigate Actionability   \n",
       "5089       TP_2064        TP53.p.H179R  Investigate Actionability   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "9                           NaN    1   KIRP                              NaN   \n",
       "11        Biologically Relevant    2   KIRP                              NaN   \n",
       "15                          NaN    1   KIRP                        Guideline   \n",
       "20                          NaN    1   KIRP                      Preclinical   \n",
       "22                          NaN    1   KIRP                              NaN   \n",
       "...                         ...  ...    ...                              ...   \n",
       "5072  Investigate Actionability    2   SU2C                              NaN   \n",
       "5077  Investigate Actionability    2   SU2C                              NaN   \n",
       "5080  Investigate Actionability    2   SU2C                        Guideline   \n",
       "5085                        NaN    1   SU2C                              NaN   \n",
       "5089  Investigate Actionability    2   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "9                    Clinical evidence                               NaN   \n",
       "11                                 NaN                         Guideline   \n",
       "15                           Guideline                               NaN   \n",
       "20                                 NaN                 Clinical evidence   \n",
       "22                   Clinical evidence                               NaN   \n",
       "...                                ...                               ...   \n",
       "5072                               NaN                         Guideline   \n",
       "5077                               NaN                         Guideline   \n",
       "5080                         Guideline                               NaN   \n",
       "5085                               NaN                 Clinical evidence   \n",
       "5089                         Guideline                         Guideline   \n",
       "\n",
       "     clinvar         feature_type feature_str_simple  \n",
       "9        NaN  Somatic copy number           BRAF Amp  \n",
       "11       NaN      Somatic variant             DNMT3A  \n",
       "15       NaN  Somatic copy number            MET Amp  \n",
       "20       NaN      Somatic variant             ARID1A  \n",
       "22       NaN      Somatic variant                BCR  \n",
       "...      ...                  ...                ...  \n",
       "5072     NaN  Somatic copy number            RB1 Del  \n",
       "5077     NaN  Somatic copy number           TP53 Del  \n",
       "5080     NaN  Somatic copy number             AR Amp  \n",
       "5085     NaN      Somatic variant                BLM  \n",
       "5089     NaN      Somatic variant               TP53  \n",
       "\n",
       "[1180 rows x 12 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "93"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ia_patients = tmp['patient_id'].drop_duplicates()\n",
    "diff_patients = pd.Index(ia_patients).difference(pd.Index(pa_patients))\n",
    "len(diff_patients)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.222"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_n_patients = 110 + 150 + 100 + 59\n",
    "round(len(diff_patients) / total_n_patients, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ESR1.Deletion         11\n",
       "NRAS.p.Q61K           10\n",
       "BRAF.p.D594Y           5\n",
       "BRAF.p.E611*           5\n",
       "TP53.Amplification     4\n",
       "                      ..\n",
       "TET2.p.P576fs          1\n",
       "BRCA1.p.E914K          1\n",
       "CTNNB1.p.D299N         1\n",
       "BRIP1.p.P368L          1\n",
       "PBRM1.p.V804A          1\n",
       "Name: feature_string, Length: 693, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['almanac_bin'].eq('Investigate Actionability')]['feature_string'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_string'].eq('NRAS.p.Q61K')]['patient_id'].drop_duplicates().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(43,)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[~tmp['phial_bin'].isnull() & tmp['patient_id'].isin(diff_patients) & tmp['phial_bin'].isin(['Putatively Actionable', 'Investigate Actionability'])]['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How many events did PHIAL call that Almanac didn't?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda3/envs/moalmanac-paper/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "False    225\n",
       "Name: gene, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "target = pd.read_excel('../knowledge-bases/target/almanac-comparison.xlsx')\n",
    "tmp = df[df['phial_bin'].isin(['Putatively Actionable', 'Investigate Actionability']) & df['almanac_bin'].isnull()]\n",
    "tmp['gene'] = tmp['feature_string'].str.split('.').apply(lambda x: x[0])\n",
    "tmp['gene'].isin(target[target['included_in_lift_to_moalmanac'].eq(0)]).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Investigate Actionability    223\n",
       "Putatively Actionable          2\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3279</th>\n",
       "      <td>MO_1232</td>\n",
       "      <td>KRAS.p.Q61H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3280</th>\n",
       "      <td>MO_1232</td>\n",
       "      <td>KRAS.p.Q61L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     patient_id feature_string almanac_bin              phial_bin  sum cohort  \\\n",
       "3279    MO_1232    KRAS.p.Q61H         NaN  Putatively Actionable    1   SU2C   \n",
       "3280    MO_1232    KRAS.p.Q61L         NaN  Putatively Actionable    1   SU2C   \n",
       "\n",
       "     sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "3279                              NaN                               NaN   \n",
       "3280                              NaN                               NaN   \n",
       "\n",
       "     prognostic_predictive_implication clinvar     feature_type  \\\n",
       "3279                               NaN     NaN  Somatic variant   \n",
       "3280                               NaN     NaN  Somatic variant   \n",
       "\n",
       "     feature_str_simple  gene  \n",
       "3279               KRAS  KRAS  \n",
       "3280               KRAS  KRAS  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['phial_bin'].eq('Putatively Actionable')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Expanding evidence\n",
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: Preclinical  \n",
    "Matches considered: Putatively actionable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(164, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>KIRP-2Z-A9JG</td>\n",
       "      <td>MDM2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MDM2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>KIRP-2Z-A9JP</td>\n",
       "      <td>FGFR1.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>FGFR1 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>KIRP-2Z-A9JQ</td>\n",
       "      <td>KEAP1.Deletion</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>KEAP1 Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id       feature_string            almanac_bin phial_bin  sum  \\\n",
       "2    KIRP-2K-A9WE  AURKA.Amplification  Putatively Actionable       NaN    1   \n",
       "7    KIRP-2K-A9WE   TPX2.Amplification  Putatively Actionable       NaN    1   \n",
       "100  KIRP-2Z-A9JG   MDM2.Amplification  Putatively Actionable       NaN    1   \n",
       "161  KIRP-2Z-A9JP  FGFR1.Amplification  Putatively Actionable       NaN    1   \n",
       "178  KIRP-2Z-A9JQ       KEAP1.Deletion  Putatively Actionable       NaN    1   \n",
       "\n",
       "    cohort sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "2     KIRP                      Preclinical                               NaN   \n",
       "7     KIRP                              NaN                               NaN   \n",
       "100   KIRP                      Preclinical                               NaN   \n",
       "161   KIRP                      Preclinical                               NaN   \n",
       "178   KIRP                              NaN                       Preclinical   \n",
       "\n",
       "    prognostic_predictive_implication clinvar         feature_type  \\\n",
       "2                         Preclinical     NaN  Somatic copy number   \n",
       "7                         Preclinical     NaN  Somatic copy number   \n",
       "100                               NaN     NaN  Somatic copy number   \n",
       "161                               NaN     NaN  Somatic copy number   \n",
       "178                               NaN     NaN  Somatic copy number   \n",
       "\n",
       "    feature_str_simple  \n",
       "2            AURKA Amp  \n",
       "7             TPX2 Amp  \n",
       "100           MDM2 Amp  \n",
       "161          FGFR1 Amp  \n",
       "178          KEAP1 Del  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['Preclinical']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_events = tmp.shape[0]\n",
    "n_patients = tmp['patient_id'].drop_duplicates().shape[0]\n",
    "n_phial = tmp['phial_bin'].dropna().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'On the basis of preclinical evidence, 164 such events from 140 patients were identified-- --91 (55.49%) of which were also highlighted by PHIAL.'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"On the basis of preclinical evidence, {n_events} such events \\\n",
    "from {n_patients} patients were identified-- --{n_phial} ({round(100*(n_phial / n_events), 2)}%) \\\n",
    "of which were also highlighted by PHIAL.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: Inferential  \n",
    "Matches considered: Putatively actionable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(24, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>KIRP-4A-A93Y</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>KIRP-A4-8630</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>KIRP-AL-3466</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>723</th>\n",
       "      <td>KIRP-B9-4117</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>747</th>\n",
       "      <td>KIRP-B9-5155</td>\n",
       "      <td>ERBB2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ERBB2 Amp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id       feature_string            almanac_bin phial_bin  sum  \\\n",
       "227  KIRP-4A-A93Y  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "477  KIRP-A4-8630  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "555  KIRP-AL-3466  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "723  KIRP-B9-4117  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "747  KIRP-B9-5155  ERBB2.Amplification  Putatively Actionable       NaN    1   \n",
       "\n",
       "    cohort sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "227   KIRP                     FDA-Approved                       Preclinical   \n",
       "477   KIRP                     FDA-Approved                       Preclinical   \n",
       "555   KIRP                     FDA-Approved                       Preclinical   \n",
       "723   KIRP                     FDA-Approved                       Preclinical   \n",
       "747   KIRP                     FDA-Approved                       Preclinical   \n",
       "\n",
       "    prognostic_predictive_implication clinvar         feature_type  \\\n",
       "227                       Inferential     NaN  Somatic copy number   \n",
       "477                       Inferential     NaN  Somatic copy number   \n",
       "555                       Inferential     NaN  Somatic copy number   \n",
       "723                       Inferential     NaN  Somatic copy number   \n",
       "747                       Inferential     NaN  Somatic copy number   \n",
       "\n",
       "    feature_str_simple  \n",
       "227          ERBB2 Amp  \n",
       "477          ERBB2 Amp  \n",
       "555          ERBB2 Amp  \n",
       "723          ERBB2 Amp  \n",
       "747          ERBB2 Amp  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['Inferential']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CCND1.Amplification    15\n",
       "ERBB2.Amplification     5\n",
       "CDKN2C.Deletion         3\n",
       "CD274.Amplification     1\n",
       "Name: feature_string, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['feature_string'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Inferential evidence 24 additional putatively actionable copy number alterations from 24 patients, most prominently CCND1 amplifications for reported sensitivity to Palbocicilib.'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_events = tmp.shape[0]\n",
    "n_patients = tmp['patient_id'].drop_duplicates().shape[0]\n",
    "f\"Inferential evidence {n_events} additional putatively actionable copy number alterations from {n_patients} patients, \\\n",
    "most prominently CCND1 amplifications for reported sensitivity to Palbocicilib.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## In total for somatic variants and copy number alterations\n",
    "Features considered: Somatic variants, copy number alterations  \n",
    "Evidence considered: FDA-approved, clinical trials, clinical evidence, preclinical, inferential  \n",
    "Matches considered: Putatively actionable, Investigate actionability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1445, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>TPX2.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TPX2 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>BRAF.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>BRAF Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>DNMT3A.p.Q350fs</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>KIRP-2Z-A9J1</td>\n",
       "      <td>MET.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MET Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5080</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AR.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AR Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5084</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5085</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>BLM.p.V54I</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>BLM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5088</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>MPL.p.A134T</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>MPL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5089</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>TP53.p.H179R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1445 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id       feature_string                almanac_bin  \\\n",
       "2     KIRP-2K-A9WE  AURKA.Amplification      Putatively Actionable   \n",
       "7     KIRP-2K-A9WE   TPX2.Amplification      Putatively Actionable   \n",
       "9     KIRP-2Z-A9J1   BRAF.Amplification      Putatively Actionable   \n",
       "11    KIRP-2Z-A9J1      DNMT3A.p.Q350fs  Investigate Actionability   \n",
       "15    KIRP-2Z-A9J1    MET.Amplification      Putatively Actionable   \n",
       "...            ...                  ...                        ...   \n",
       "5080       TP_2064     AR.Amplification      Putatively Actionable   \n",
       "5084       TP_2064  AURKA.Amplification      Putatively Actionable   \n",
       "5085       TP_2064           BLM.p.V54I  Investigate Actionability   \n",
       "5088       TP_2064          MPL.p.A134T  Investigate Actionability   \n",
       "5089       TP_2064         TP53.p.H179R  Investigate Actionability   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "2                           NaN    1   KIRP                      Preclinical   \n",
       "7                           NaN    1   KIRP                              NaN   \n",
       "9                           NaN    1   KIRP                              NaN   \n",
       "11        Biologically Relevant    2   KIRP                              NaN   \n",
       "15                          NaN    1   KIRP                        Guideline   \n",
       "...                         ...  ...    ...                              ...   \n",
       "5080  Investigate Actionability    2   SU2C                        Guideline   \n",
       "5084                        NaN    1   SU2C                      Preclinical   \n",
       "5085                        NaN    1   SU2C                              NaN   \n",
       "5088      Biologically Relevant    2   SU2C                      Preclinical   \n",
       "5089  Investigate Actionability    2   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "2                                  NaN                       Preclinical   \n",
       "7                                  NaN                       Preclinical   \n",
       "9                    Clinical evidence                               NaN   \n",
       "11                                 NaN                         Guideline   \n",
       "15                           Guideline                               NaN   \n",
       "...                                ...                               ...   \n",
       "5080                         Guideline                               NaN   \n",
       "5084                               NaN                       Preclinical   \n",
       "5085                               NaN                 Clinical evidence   \n",
       "5088                               NaN                               NaN   \n",
       "5089                         Guideline                         Guideline   \n",
       "\n",
       "     clinvar         feature_type feature_str_simple  \n",
       "2        NaN  Somatic copy number          AURKA Amp  \n",
       "7        NaN  Somatic copy number           TPX2 Amp  \n",
       "9        NaN  Somatic copy number           BRAF Amp  \n",
       "11       NaN      Somatic variant             DNMT3A  \n",
       "15       NaN  Somatic copy number            MET Amp  \n",
       "...      ...                  ...                ...  \n",
       "5080     NaN  Somatic copy number             AR Amp  \n",
       "5084     NaN  Somatic copy number          AURKA Amp  \n",
       "5085     NaN      Somatic variant                BLM  \n",
       "5088     NaN      Somatic variant                MPL  \n",
       "5089     NaN      Somatic variant               TP53  \n",
       "\n",
       "[1445 rows x 12 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_types = ['Somatic variant', 'Somatic copy number']\n",
    "evidence_types = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence', 'Preclinical', 'Inferential']\n",
    "idx_match = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_features = df['feature_type'].isin(feature_types)\n",
    "idx_evidence_s = df['sensitive_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_r = df['resistance_predictive_implication'].isin(evidence_types)\n",
    "idx_evidence_p = df['prognostic_predictive_implication'].isin(evidence_types)\n",
    "\n",
    "tmp = df[idx_match & idx_features & (idx_evidence_s | idx_evidence_r | idx_evidence_p)]\n",
    "print(tmp.shape)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP 72\n",
      "MEL 109\n",
      "OS 42\n",
      "SU2C 142\n",
      "365\n"
     ]
    }
   ],
   "source": [
    "total_n = 0\n",
    "for label, group in tmp.groupby('cohort'):\n",
    "    total_n += group['patient_id'].drop_duplicates().shape[0]\n",
    "    print(label, group['patient_id'].drop_duplicates().shape[0])\n",
    "print(total_n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        390\n",
       "Investigate Actionability    374\n",
       "Putatively Actionable         79\n",
       "Name: phial_bin, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['phial_bin'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biologically Relevant        0.269896\n",
       "Investigate Actionability    0.258824\n",
       "Putatively Actionable        0.054671\n",
       "Name: phial_bin, dtype: float64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['phial_bin'].value_counts().divide(tmp.shape[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validation sequencing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>score_bin</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature</th>\n",
       "      <th>alteration_type</th>\n",
       "      <th>alteration</th>\n",
       "      <th>tumor_f</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>...</th>\n",
       "      <th>prognostic_url</th>\n",
       "      <th>number_germline_mutations_in_gene</th>\n",
       "      <th>validation_total_coverage</th>\n",
       "      <th>validation_tumor_f</th>\n",
       "      <th>validation_detection_power</th>\n",
       "      <th>feature_display</th>\n",
       "      <th>preclinical_efficacy_observed</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>normal_sample_barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.6073</td>\n",
       "      <td>219.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1921.0</td>\n",
       "      <td>0.9724</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SC_9065</td>\n",
       "      <td>SC_9065-Tumor</td>\n",
       "      <td>SC_9065-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.4224</td>\n",
       "      <td>232.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>290.0</td>\n",
       "      <td>0.4759</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-1115161</td>\n",
       "      <td>PROS01448-1115161-Tumor-SM-5SGU1</td>\n",
       "      <td>PROS01448-1115161-Normal-SM-6CAJL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>217.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1200/JCO.2017.75.7310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.2857</td>\n",
       "      <td>0.9799</td>\n",
       "      <td>PIK3CA p.E545K (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MO_1410</td>\n",
       "      <td>MO_1410-Tumor</td>\n",
       "      <td>MO_1410-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.5703</td>\n",
       "      <td>128.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.7619</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS12319B-SU2C-06115114</td>\n",
       "      <td>PROS12319B-SU2C-06115114-Tumor-SM-4W2N9</td>\n",
       "      <td>PROS12319B-SU2C-06115114-Normal-SM-4W2NS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.0800</td>\n",
       "      <td>50.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-6115250</td>\n",
       "      <td>PROS01448-6115250-Tumor-SM-67ES4</td>\n",
       "      <td>PROS01448-6115250-Normal-SM-67ET3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.5806</td>\n",
       "      <td>31.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.9394</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-1115244</td>\n",
       "      <td>PROS01448-1115244-Tumor-SM-5SGU6</td>\n",
       "      <td>PROS01448-1115244-Normal-SM-6CAJP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.1190</td>\n",
       "      <td>210.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>0.2222</td>\n",
       "      <td>0.9822</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-6115219</td>\n",
       "      <td>PROS01448-6115219-Tumor-SM-67ERS</td>\n",
       "      <td>PROS01448-6115219-Normal-SM-67ESS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.2063</td>\n",
       "      <td>63.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1305.0</td>\n",
       "      <td>0.5586</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MO_1316</td>\n",
       "      <td>MO_1316-TM</td>\n",
       "      <td>MO_1316-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.8983</td>\n",
       "      <td>59.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROS01448-6115227</td>\n",
       "      <td>PROS01448-6115227-Tumor-SM-67ERU</td>\n",
       "      <td>PROS01448-6115227-Normal-SM-67ESU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>699</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.L702H</td>\n",
       "      <td>0.0758</td>\n",
       "      <td>356.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>640.0</td>\n",
       "      <td>0.1125</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.L702H (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SC_9046</td>\n",
       "      <td>SC_9046-Tumor</td>\n",
       "      <td>SC_9046-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E545A</td>\n",
       "      <td>0.4378</td>\n",
       "      <td>201.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1200/JCO.2017.75.7310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>PIK3CA p.E545A (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MO_1012</td>\n",
       "      <td>MO_1012-Tumor-Subcutaneous_nodule</td>\n",
       "      <td>MO_1012-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.5784</td>\n",
       "      <td>306.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MO_1040</td>\n",
       "      <td>MO_1040-Tumor</td>\n",
       "      <td>MO_1040-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1425</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.7194</td>\n",
       "      <td>449.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>0.9922</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP_2032</td>\n",
       "      <td>TP_2032-Tumor</td>\n",
       "      <td>TP_2032-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AR</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.T878A</td>\n",
       "      <td>0.5587</td>\n",
       "      <td>213.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>493.0</td>\n",
       "      <td>0.9899</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AR p.T878A (Missense)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SC_9011</td>\n",
       "      <td>SC_9011-Tumor</td>\n",
       "      <td>SC_9011-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1533</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E17K</td>\n",
       "      <td>0.5348</td>\n",
       "      <td>445.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>102.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AKT1 p.E17K (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MO_1096</td>\n",
       "      <td>MO_1096-Tumor</td>\n",
       "      <td>MO_1096-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1663</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E17K</td>\n",
       "      <td>0.2219</td>\n",
       "      <td>329.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>445.0</td>\n",
       "      <td>0.2921</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>AKT1 p.E17K (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SC_9019</td>\n",
       "      <td>SC_9019-Tumor</td>\n",
       "      <td>SC_9019-Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1733</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.1910</td>\n",
       "      <td>267.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.4444</td>\n",
       "      <td>0.9979</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat38</td>\n",
       "      <td>MEL-IPI_Pat38-Tumor-SM-53U3Z</td>\n",
       "      <td>MEL-IPI_Pat38-Normal-SM-53U5L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1925</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.3716</td>\n",
       "      <td>218.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1468</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat39</td>\n",
       "      <td>MEL-IPI_Pat39-Tumor-SM-4DK1Z</td>\n",
       "      <td>MEL-IPI_Pat39-Normal-SM-4NFV6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2016</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.8286</td>\n",
       "      <td>245.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>166.0</td>\n",
       "      <td>0.8795</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat08</td>\n",
       "      <td>MEL-IPI_Pat08-Tumor-SM-4DK14</td>\n",
       "      <td>MEL-IPI_Pat08-Normal-SM-4NFUA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2050</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.3945</td>\n",
       "      <td>436.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat37</td>\n",
       "      <td>MEL-IPI_Pat37-Tumor-SM-53U3Y</td>\n",
       "      <td>MEL-IPI_Pat37-Normal-SM-4NFV4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2130</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6027</td>\n",
       "      <td>297.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>109.0</td>\n",
       "      <td>0.5596</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat90</td>\n",
       "      <td>MEL-IPI_Pat90-Tumor-SM-4DK3G</td>\n",
       "      <td>MEL-IPI_Pat90-Normal-SM-4NFWM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2289</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6505</td>\n",
       "      <td>372.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>92.0</td>\n",
       "      <td>0.6739</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat15</td>\n",
       "      <td>MEL-IPI_Pat15-Tumor-SM-4DK1B</td>\n",
       "      <td>MEL-IPI_Pat15-Normal-SM-4NFUH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2313</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61R</td>\n",
       "      <td>0.4222</td>\n",
       "      <td>334.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>118.0</td>\n",
       "      <td>0.4661</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NRAS p.Q61R (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat119</td>\n",
       "      <td>MEL-IPI_Pat119-Tumor-SM-7459N</td>\n",
       "      <td>MEL-IPI_Pat119-Normal-SM-7459Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2423</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q56P</td>\n",
       "      <td>0.2955</td>\n",
       "      <td>264.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.3333</td>\n",
       "      <td>0.0703</td>\n",
       "      <td>MAP2K1 p.Q56P (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEL-IPI_Pat117</td>\n",
       "      <td>MEL-IPI_Pat117-Tumor-SM-5X2QU</td>\n",
       "      <td>MEL-IPI_Pat117-Normal-SM-5X2RC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2446</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.4085</td>\n",
       "      <td>213.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.4444</td>\n",
       "      <td>0.5392</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat128</td>\n",
       "      <td>MEL-IPI_Pat128-Tumor-SM-5X2R6</td>\n",
       "      <td>MEL-IPI_Pat128-Normal-SM-5VWHQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2546</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6174</td>\n",
       "      <td>311.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat49</td>\n",
       "      <td>MEL-IPI_Pat49-Tumor-SM-4DK2A</td>\n",
       "      <td>MEL-IPI_Pat49-Normal-SM-53U5W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2595</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6393</td>\n",
       "      <td>718.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.4000</td>\n",
       "      <td>0.4079</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat129</td>\n",
       "      <td>MEL-IPI_Pat129-Tumor-SM-5X2R7</td>\n",
       "      <td>MEL-IPI_Pat129-Normal-SM-5X2RI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2620</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6593</td>\n",
       "      <td>226.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat25</td>\n",
       "      <td>MEL-IPI_Pat25-Tumor-SM-4DK1L</td>\n",
       "      <td>MEL-IPI_Pat25-Normal-SM-4NFUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.7023</td>\n",
       "      <td>393.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.4000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat118</td>\n",
       "      <td>MEL-IPI_Pat118-Tumor-SM-5X2QV</td>\n",
       "      <td>MEL-IPI_Pat118-Normal-SM-5X2RD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2666</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2834</td>\n",
       "      <td>247.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>118.0</td>\n",
       "      <td>0.2542</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat14</td>\n",
       "      <td>MEL-IPI_Pat14-Tumor-SM-53U3C</td>\n",
       "      <td>MEL-IPI_Pat14-Normal-SM-4NFUG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2680</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.8049</td>\n",
       "      <td>123.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat50</td>\n",
       "      <td>MEL-IPI_Pat50-Tumor-SM-53U69</td>\n",
       "      <td>MEL-IPI_Pat50-Normal-SM-53U5X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2842</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2387</td>\n",
       "      <td>331.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>137.0</td>\n",
       "      <td>0.2044</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat03</td>\n",
       "      <td>MEL-IPI_Pat03-Tumor-SM-4DJZY</td>\n",
       "      <td>MEL-IPI_Pat03-Normal-SM-4NFU5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3027</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.6316</td>\n",
       "      <td>152.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1610</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat02</td>\n",
       "      <td>MEL-IPI_Pat02-Tumor-SM-4DJZX</td>\n",
       "      <td>MEL-IPI_Pat02-Normal-SM-4NFU4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3232</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2340</td>\n",
       "      <td>376.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>114.0</td>\n",
       "      <td>0.2456</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat27</td>\n",
       "      <td>MEL-IPI_Pat27-Tumor-SM-4DK1N</td>\n",
       "      <td>MEL-IPI_Pat27-Normal-SM-4NFUT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3288</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.4659</td>\n",
       "      <td>410.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>122.0</td>\n",
       "      <td>0.5574</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat80</td>\n",
       "      <td>MEL-IPI_Pat80-Tumor-SM-4DK36</td>\n",
       "      <td>MEL-IPI_Pat80-Normal-SM-53U53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3367</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61L</td>\n",
       "      <td>0.6401</td>\n",
       "      <td>1592.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>493.0</td>\n",
       "      <td>0.7201</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NRAS p.Q61L (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat19</td>\n",
       "      <td>MEL-IPI_Pat19-Tumor-SM-4DK1F</td>\n",
       "      <td>MEL-IPI_Pat19-Normal-SM-4NFUL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3420</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61R</td>\n",
       "      <td>0.4370</td>\n",
       "      <td>270.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>152.0</td>\n",
       "      <td>0.4803</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NRAS p.Q61R (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat123</td>\n",
       "      <td>MEL-IPI_Pat123-Tumor-SM-5X2R1</td>\n",
       "      <td>MEL-IPI_Pat123-Normal-SM-5VWHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3432</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61R</td>\n",
       "      <td>0.4457</td>\n",
       "      <td>341.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NRAS p.Q61R (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat28</td>\n",
       "      <td>MEL-IPI_Pat28-Tumor-SM-4DK1O</td>\n",
       "      <td>MEL-IPI_Pat28-Normal-SM-4NFUU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3539</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.Q61L</td>\n",
       "      <td>0.3958</td>\n",
       "      <td>331.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NRAS p.Q61L (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat44</td>\n",
       "      <td>MEL-IPI_Pat44-Tumor-SM-4DK25</td>\n",
       "      <td>MEL-IPI_Pat44-Normal-SM-4NFVB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3549</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>KIT</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V559G</td>\n",
       "      <td>0.8811</td>\n",
       "      <td>858.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>KIT p.V559G (Missense)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEL-IPI_Pat88</td>\n",
       "      <td>MEL-IPI_Pat88-Tumor-SM-4DK3E</td>\n",
       "      <td>MEL-IPI_Pat88-Normal-SM-53U4C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3610</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.2803</td>\n",
       "      <td>635.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.2294</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat43</td>\n",
       "      <td>MEL-IPI_Pat43-Tumor-SM-4DK24</td>\n",
       "      <td>MEL-IPI_Pat43-Normal-SM-4NFVA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3636</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.4099</td>\n",
       "      <td>405.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>106.0</td>\n",
       "      <td>0.6698</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MEL-IPI_Pat86</td>\n",
       "      <td>MEL-IPI_Pat86-Tumor-SM-4DK3C</td>\n",
       "      <td>MEL-IPI_Pat86-Normal-SM-53U59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4217</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.G12D</td>\n",
       "      <td>0.4828</td>\n",
       "      <td>29.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1038/ctg.2016.18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.4706</td>\n",
       "      <td>0.9721</td>\n",
       "      <td>KRAS p.G12D (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>KIRP-A4-8312</td>\n",
       "      <td>KIRP-A4-8312-TP</td>\n",
       "      <td>KIRP-A4-8312-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4298</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>0.3839</td>\n",
       "      <td>112.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.nccn.org/professionals/physician_g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.2353</td>\n",
       "      <td>0.9280</td>\n",
       "      <td>BRAF p.V600E (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>KIRP-AL-3467</td>\n",
       "      <td>KIRP-AL-3467-TP</td>\n",
       "      <td>KIRP-AL-3467-NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4343</th>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical trial</td>\n",
       "      <td>Somatic Variant</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E542K</td>\n",
       "      <td>0.4722</td>\n",
       "      <td>449.0</td>\n",
       "      <td>...</td>\n",
       "      <td>https://doi.org/10.1200/JCO.2017.75.7310</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.0505</td>\n",
       "      <td>PIK3CA p.E542K (Missense)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>KIRP-5P-A9K3</td>\n",
       "      <td>KIRP-5P-A9K3-TP</td>\n",
       "      <td>KIRP-5P-A9K3-NB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  score_bin sensitive_predictive_implication  \\\n",
       "70    Putatively Actionable                              NaN   \n",
       "119   Putatively Actionable                              NaN   \n",
       "137   Putatively Actionable                      Preclinical   \n",
       "369   Putatively Actionable                              NaN   \n",
       "424   Putatively Actionable                              NaN   \n",
       "442   Putatively Actionable                              NaN   \n",
       "510   Putatively Actionable                              NaN   \n",
       "533   Putatively Actionable                              NaN   \n",
       "564   Putatively Actionable                              NaN   \n",
       "699   Putatively Actionable                              NaN   \n",
       "879   Putatively Actionable                      Preclinical   \n",
       "1026  Putatively Actionable                              NaN   \n",
       "1425  Putatively Actionable                              NaN   \n",
       "1505  Putatively Actionable                              NaN   \n",
       "1533  Putatively Actionable                      Preclinical   \n",
       "1663  Putatively Actionable                      Preclinical   \n",
       "1733  Putatively Actionable                     FDA-Approved   \n",
       "1925  Putatively Actionable                     FDA-Approved   \n",
       "2016  Putatively Actionable                     FDA-Approved   \n",
       "2050  Putatively Actionable                     FDA-Approved   \n",
       "2130  Putatively Actionable                     FDA-Approved   \n",
       "2289  Putatively Actionable                     FDA-Approved   \n",
       "2313  Putatively Actionable                Clinical evidence   \n",
       "2423  Putatively Actionable                      Preclinical   \n",
       "2446  Putatively Actionable                     FDA-Approved   \n",
       "2546  Putatively Actionable                     FDA-Approved   \n",
       "2595  Putatively Actionable                     FDA-Approved   \n",
       "2620  Putatively Actionable                     FDA-Approved   \n",
       "2653  Putatively Actionable                     FDA-Approved   \n",
       "2666  Putatively Actionable                     FDA-Approved   \n",
       "2680  Putatively Actionable                     FDA-Approved   \n",
       "2842  Putatively Actionable                     FDA-Approved   \n",
       "3027  Putatively Actionable                     FDA-Approved   \n",
       "3232  Putatively Actionable                     FDA-Approved   \n",
       "3288  Putatively Actionable                     FDA-Approved   \n",
       "3367  Putatively Actionable                Clinical evidence   \n",
       "3420  Putatively Actionable                Clinical evidence   \n",
       "3432  Putatively Actionable                Clinical evidence   \n",
       "3539  Putatively Actionable                Clinical evidence   \n",
       "3549  Putatively Actionable                        Guideline   \n",
       "3610  Putatively Actionable                     FDA-Approved   \n",
       "3636  Putatively Actionable                     FDA-Approved   \n",
       "4217  Putatively Actionable                   Clinical trial   \n",
       "4298  Putatively Actionable                     FDA-Approved   \n",
       "4343  Putatively Actionable                      Preclinical   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "70                   Clinical evidence                               NaN   \n",
       "119                  Clinical evidence                               NaN   \n",
       "137                                NaN                    Clinical trial   \n",
       "369                  Clinical evidence                               NaN   \n",
       "424                  Clinical evidence                               NaN   \n",
       "442                  Clinical evidence                               NaN   \n",
       "510                  Clinical evidence                               NaN   \n",
       "533                  Clinical evidence                               NaN   \n",
       "564                  Clinical evidence                               NaN   \n",
       "699                  Clinical evidence                               NaN   \n",
       "879                                NaN                    Clinical trial   \n",
       "1026                 Clinical evidence                               NaN   \n",
       "1425                 Clinical evidence                               NaN   \n",
       "1505                 Clinical evidence                               NaN   \n",
       "1533                               NaN                               NaN   \n",
       "1663                               NaN                               NaN   \n",
       "1733                         Guideline                         Guideline   \n",
       "1925                         Guideline                         Guideline   \n",
       "2016                         Guideline                         Guideline   \n",
       "2050                         Guideline                         Guideline   \n",
       "2130                         Guideline                         Guideline   \n",
       "2289                         Guideline                         Guideline   \n",
       "2313                               NaN                         Guideline   \n",
       "2423                       Preclinical                               NaN   \n",
       "2446                         Guideline                         Guideline   \n",
       "2546                         Guideline                         Guideline   \n",
       "2595                         Guideline                         Guideline   \n",
       "2620                         Guideline                         Guideline   \n",
       "2653                         Guideline                         Guideline   \n",
       "2666                         Guideline                         Guideline   \n",
       "2680                         Guideline                         Guideline   \n",
       "2842                         Guideline                         Guideline   \n",
       "3027                         Guideline                         Guideline   \n",
       "3232                         Guideline                         Guideline   \n",
       "3288                         Guideline                         Guideline   \n",
       "3367                               NaN                         Guideline   \n",
       "3420                               NaN                         Guideline   \n",
       "3432                               NaN                         Guideline   \n",
       "3539                               NaN                         Guideline   \n",
       "3549                               NaN                         Guideline   \n",
       "3610                         Guideline                         Guideline   \n",
       "3636                         Guideline                         Guideline   \n",
       "4217                    Clinical trial                 Clinical evidence   \n",
       "4298                         Guideline                         Guideline   \n",
       "4343                               NaN                    Clinical trial   \n",
       "\n",
       "         feature_type feature alteration_type alteration  tumor_f  \\\n",
       "70    Somatic Variant      AR        Missense    p.T878A   0.6073   \n",
       "119   Somatic Variant      AR        Missense    p.T878A   0.4224   \n",
       "137   Somatic Variant  PIK3CA        Missense    p.E545K   0.3825   \n",
       "369   Somatic Variant      AR        Missense    p.L702H   0.5703   \n",
       "424   Somatic Variant      AR        Missense    p.L702H   0.0800   \n",
       "442   Somatic Variant      AR        Missense    p.L702H   0.5806   \n",
       "510   Somatic Variant      AR        Missense    p.L702H   0.1190   \n",
       "533   Somatic Variant      AR        Missense    p.L702H   0.2063   \n",
       "564   Somatic Variant      AR        Missense    p.L702H   0.8983   \n",
       "699   Somatic Variant      AR        Missense    p.L702H   0.0758   \n",
       "879   Somatic Variant  PIK3CA        Missense    p.E545A   0.4378   \n",
       "1026  Somatic Variant      AR        Missense    p.T878A   0.5784   \n",
       "1425  Somatic Variant      AR        Missense    p.T878A   0.7194   \n",
       "1505  Somatic Variant      AR        Missense    p.T878A   0.5587   \n",
       "1533  Somatic Variant    AKT1        Missense     p.E17K   0.5348   \n",
       "1663  Somatic Variant    AKT1        Missense     p.E17K   0.2219   \n",
       "1733  Somatic Variant    BRAF        Missense    p.V600E   0.1910   \n",
       "1925  Somatic Variant    BRAF        Missense    p.V600E   0.3716   \n",
       "2016  Somatic Variant    BRAF        Missense    p.V600E   0.8286   \n",
       "2050  Somatic Variant    BRAF        Missense    p.V600E   0.3945   \n",
       "2130  Somatic Variant    BRAF        Missense    p.V600E   0.6027   \n",
       "2289  Somatic Variant    BRAF        Missense    p.V600E   0.6505   \n",
       "2313  Somatic Variant    NRAS        Missense     p.Q61R   0.4222   \n",
       "2423  Somatic Variant  MAP2K1        Missense     p.Q56P   0.2955   \n",
       "2446  Somatic Variant    BRAF        Missense    p.V600E   0.4085   \n",
       "2546  Somatic Variant    BRAF        Missense    p.V600E   0.6174   \n",
       "2595  Somatic Variant    BRAF        Missense    p.V600E   0.6393   \n",
       "2620  Somatic Variant    BRAF        Missense    p.V600E   0.6593   \n",
       "2653  Somatic Variant    BRAF        Missense    p.V600E   0.7023   \n",
       "2666  Somatic Variant    BRAF        Missense    p.V600E   0.2834   \n",
       "2680  Somatic Variant    BRAF        Missense    p.V600E   0.8049   \n",
       "2842  Somatic Variant    BRAF        Missense    p.V600E   0.2387   \n",
       "3027  Somatic Variant    BRAF        Missense    p.V600E   0.6316   \n",
       "3232  Somatic Variant    BRAF        Missense    p.V600E   0.2340   \n",
       "3288  Somatic Variant    BRAF        Missense    p.V600E   0.4659   \n",
       "3367  Somatic Variant    NRAS        Missense     p.Q61L   0.6401   \n",
       "3420  Somatic Variant    NRAS        Missense     p.Q61R   0.4370   \n",
       "3432  Somatic Variant    NRAS        Missense     p.Q61R   0.4457   \n",
       "3539  Somatic Variant    NRAS        Missense     p.Q61L   0.3958   \n",
       "3549  Somatic Variant     KIT        Missense    p.V559G   0.8811   \n",
       "3610  Somatic Variant    BRAF        Missense    p.V600E   0.2803   \n",
       "3636  Somatic Variant    BRAF        Missense    p.V600E   0.4099   \n",
       "4217  Somatic Variant    KRAS        Missense     p.G12D   0.4828   \n",
       "4298  Somatic Variant    BRAF        Missense    p.V600E   0.3839   \n",
       "4343  Somatic Variant  PIK3CA        Missense    p.E542K   0.4722   \n",
       "\n",
       "      total_coverage  ...                                     prognostic_url  \\\n",
       "70             219.0  ...                                                NaN   \n",
       "119            232.0  ...                                                NaN   \n",
       "137            217.0  ...           https://doi.org/10.1200/JCO.2017.75.7310   \n",
       "369            128.0  ...                                                NaN   \n",
       "424             50.0  ...                                                NaN   \n",
       "442             31.0  ...                                                NaN   \n",
       "510            210.0  ...                                                NaN   \n",
       "533             63.0  ...                                                NaN   \n",
       "564             59.0  ...                                                NaN   \n",
       "699            356.0  ...                                                NaN   \n",
       "879            201.0  ...           https://doi.org/10.1200/JCO.2017.75.7310   \n",
       "1026           306.0  ...                                                NaN   \n",
       "1425           449.0  ...                                                NaN   \n",
       "1505           213.0  ...                                                NaN   \n",
       "1533           445.0  ...                                                NaN   \n",
       "1663           329.0  ...                                                NaN   \n",
       "1733           267.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "1925           218.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2016           245.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2050           436.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2130           297.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2289           372.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2313           334.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2423           264.0  ...                                                NaN   \n",
       "2446           213.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2546           311.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2595           718.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2620           226.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2653           393.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2666           247.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2680           123.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "2842           331.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3027           152.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3232           376.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3288           410.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3367          1592.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3420           270.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3432           341.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3539           331.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3549           858.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3610           635.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "3636           405.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "4217            29.0  ...                https://doi.org/10.1038/ctg.2016.18   \n",
       "4298           112.0  ...  https://www.nccn.org/professionals/physician_g...   \n",
       "4343           449.0  ...           https://doi.org/10.1200/JCO.2017.75.7310   \n",
       "\n",
       "      number_germline_mutations_in_gene validation_total_coverage  \\\n",
       "70                                  1.0                    1921.0   \n",
       "119                                 NaN                     290.0   \n",
       "137                                 NaN                      21.0   \n",
       "369                                 NaN                      21.0   \n",
       "424                                 NaN                       4.0   \n",
       "442                                 NaN                      33.0   \n",
       "510                                 1.0                      81.0   \n",
       "533                                 2.0                    1305.0   \n",
       "564                                 NaN                      66.0   \n",
       "699                                 1.0                     640.0   \n",
       "879                                 NaN                       1.0   \n",
       "1026                                1.0                      64.0   \n",
       "1425                                1.0                     383.0   \n",
       "1505                                2.0                     493.0   \n",
       "1533                                NaN                     102.0   \n",
       "1663                                NaN                     445.0   \n",
       "1733                                NaN                      63.0   \n",
       "1925                                NaN                       6.0   \n",
       "2016                                NaN                     166.0   \n",
       "2050                                NaN                       0.0   \n",
       "2130                                NaN                     109.0   \n",
       "2289                                NaN                      92.0   \n",
       "2313                                NaN                     118.0   \n",
       "2423                                NaN                       6.0   \n",
       "2446                                NaN                       9.0   \n",
       "2546                                NaN                       0.0   \n",
       "2595                                NaN                       5.0   \n",
       "2620                                NaN                       0.0   \n",
       "2653                                NaN                      20.0   \n",
       "2666                                NaN                     118.0   \n",
       "2680                                NaN                       0.0   \n",
       "2842                                NaN                     137.0   \n",
       "3027                                NaN                       4.0   \n",
       "3232                                NaN                     114.0   \n",
       "3288                                NaN                     122.0   \n",
       "3367                                NaN                     493.0   \n",
       "3420                                NaN                     152.0   \n",
       "3432                                NaN                       0.0   \n",
       "3539                                NaN                       0.0   \n",
       "3549                                NaN                       0.0   \n",
       "3610                                NaN                       9.0   \n",
       "3636                                NaN                     106.0   \n",
       "4217                                NaN                      17.0   \n",
       "4298                                NaN                      17.0   \n",
       "4343                                1.0                       4.0   \n",
       "\n",
       "     validation_tumor_f validation_detection_power            feature_display  \\\n",
       "70               0.9724                     1.0000      AR p.T878A (Missense)   \n",
       "119              0.4759                     1.0000      AR p.T878A (Missense)   \n",
       "137              0.2857                     0.9799  PIK3CA p.E545K (Missense)   \n",
       "369              0.7619                     0.9998      AR p.L702H (Missense)   \n",
       "424              0.0000                     0.0002      AR p.L702H (Missense)   \n",
       "442              0.9394                     1.0000      AR p.L702H (Missense)   \n",
       "510              0.2222                     0.9822      AR p.L702H (Missense)   \n",
       "533              0.5586                     1.0000      AR p.L702H (Missense)   \n",
       "564              1.0000                     1.0000      AR p.L702H (Missense)   \n",
       "699              0.1125                     1.0000      AR p.L702H (Missense)   \n",
       "879              0.0000                     0.0000  PIK3CA p.E545A (Missense)   \n",
       "1026             0.2812                     1.0000      AR p.T878A (Missense)   \n",
       "1425             0.9922                     1.0000      AR p.T878A (Missense)   \n",
       "1505             0.9899                     1.0000      AR p.T878A (Missense)   \n",
       "1533             0.0000                     1.0000     AKT1 p.E17K (Missense)   \n",
       "1663             0.2921                     1.0000     AKT1 p.E17K (Missense)   \n",
       "1733             0.4444                     0.9979    BRAF p.V600E (Missense)   \n",
       "1925             0.0000                     0.1468    BRAF p.V600E (Missense)   \n",
       "2016             0.8795                     1.0000    BRAF p.V600E (Missense)   \n",
       "2050             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2130             0.5596                     1.0000    BRAF p.V600E (Missense)   \n",
       "2289             0.6739                     1.0000    BRAF p.V600E (Missense)   \n",
       "2313             0.4661                     1.0000     NRAS p.Q61R (Missense)   \n",
       "2423             0.3333                     0.0703   MAP2K1 p.Q56P (Missense)   \n",
       "2446             0.4444                     0.5392    BRAF p.V600E (Missense)   \n",
       "2546             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2595             0.4000                     0.4079    BRAF p.V600E (Missense)   \n",
       "2620             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2653             0.4000                     1.0000    BRAF p.V600E (Missense)   \n",
       "2666             0.2542                     1.0000    BRAF p.V600E (Missense)   \n",
       "2680             0.0000                     0.0000    BRAF p.V600E (Missense)   \n",
       "2842             0.2044                     1.0000    BRAF p.V600E (Missense)   \n",
       "3027             0.0000                     0.1610    BRAF p.V600E (Missense)   \n",
       "3232             0.2456                     1.0000    BRAF p.V600E (Missense)   \n",
       "3288             0.5574                     1.0000    BRAF p.V600E (Missense)   \n",
       "3367             0.7201                     1.0000     NRAS p.Q61L (Missense)   \n",
       "3420             0.4803                     1.0000     NRAS p.Q61R (Missense)   \n",
       "3432             0.0000                     0.0000     NRAS p.Q61R (Missense)   \n",
       "3539             0.0000                     0.0000     NRAS p.Q61L (Missense)   \n",
       "3549             0.0000                     0.0000     KIT p.V559G (Missense)   \n",
       "3610             0.0000                     0.2294    BRAF p.V600E (Missense)   \n",
       "3636             0.6698                     1.0000    BRAF p.V600E (Missense)   \n",
       "4217             0.4706                     0.9721     KRAS p.G12D (Missense)   \n",
       "4298             0.2353                     0.9280    BRAF p.V600E (Missense)   \n",
       "4343             0.5000                     0.0505  PIK3CA p.E542K (Missense)   \n",
       "\n",
       "     preclinical_efficacy_observed                patient_id  \\\n",
       "70                             NaN                   SC_9065   \n",
       "119                            NaN         PROS01448-1115161   \n",
       "137                            1.0                   MO_1410   \n",
       "369                            NaN  PROS12319B-SU2C-06115114   \n",
       "424                            NaN         PROS01448-6115250   \n",
       "442                            NaN         PROS01448-1115244   \n",
       "510                            NaN         PROS01448-6115219   \n",
       "533                            NaN                   MO_1316   \n",
       "564                            NaN         PROS01448-6115227   \n",
       "699                            NaN                   SC_9046   \n",
       "879                            1.0                   MO_1012   \n",
       "1026                           NaN                   MO_1040   \n",
       "1425                           NaN                   TP_2032   \n",
       "1505                           NaN                   SC_9011   \n",
       "1533                           0.0                   MO_1096   \n",
       "1663                           0.0                   SC_9019   \n",
       "1733                           1.0             MEL-IPI_Pat38   \n",
       "1925                           1.0             MEL-IPI_Pat39   \n",
       "2016                           1.0             MEL-IPI_Pat08   \n",
       "2050                           1.0             MEL-IPI_Pat37   \n",
       "2130                           1.0             MEL-IPI_Pat90   \n",
       "2289                           1.0             MEL-IPI_Pat15   \n",
       "2313                           1.0            MEL-IPI_Pat119   \n",
       "2423                           0.0            MEL-IPI_Pat117   \n",
       "2446                           1.0            MEL-IPI_Pat128   \n",
       "2546                           1.0             MEL-IPI_Pat49   \n",
       "2595                           1.0            MEL-IPI_Pat129   \n",
       "2620                           1.0             MEL-IPI_Pat25   \n",
       "2653                           1.0            MEL-IPI_Pat118   \n",
       "2666                           1.0             MEL-IPI_Pat14   \n",
       "2680                           1.0             MEL-IPI_Pat50   \n",
       "2842                           1.0             MEL-IPI_Pat03   \n",
       "3027                           1.0             MEL-IPI_Pat02   \n",
       "3232                           1.0             MEL-IPI_Pat27   \n",
       "3288                           1.0             MEL-IPI_Pat80   \n",
       "3367                           1.0             MEL-IPI_Pat19   \n",
       "3420                           1.0            MEL-IPI_Pat123   \n",
       "3432                           1.0             MEL-IPI_Pat28   \n",
       "3539                           1.0             MEL-IPI_Pat44   \n",
       "3549                           0.0             MEL-IPI_Pat88   \n",
       "3610                           1.0             MEL-IPI_Pat43   \n",
       "3636                           1.0             MEL-IPI_Pat86   \n",
       "4217                           1.0              KIRP-A4-8312   \n",
       "4298                           1.0              KIRP-AL-3467   \n",
       "4343                           1.0              KIRP-5P-A9K3   \n",
       "\n",
       "                         tumor_sample_barcode  \\\n",
       "70                              SC_9065-Tumor   \n",
       "119          PROS01448-1115161-Tumor-SM-5SGU1   \n",
       "137                             MO_1410-Tumor   \n",
       "369   PROS12319B-SU2C-06115114-Tumor-SM-4W2N9   \n",
       "424          PROS01448-6115250-Tumor-SM-67ES4   \n",
       "442          PROS01448-1115244-Tumor-SM-5SGU6   \n",
       "510          PROS01448-6115219-Tumor-SM-67ERS   \n",
       "533                                MO_1316-TM   \n",
       "564          PROS01448-6115227-Tumor-SM-67ERU   \n",
       "699                             SC_9046-Tumor   \n",
       "879         MO_1012-Tumor-Subcutaneous_nodule   \n",
       "1026                            MO_1040-Tumor   \n",
       "1425                            TP_2032-Tumor   \n",
       "1505                            SC_9011-Tumor   \n",
       "1533                            MO_1096-Tumor   \n",
       "1663                            SC_9019-Tumor   \n",
       "1733             MEL-IPI_Pat38-Tumor-SM-53U3Z   \n",
       "1925             MEL-IPI_Pat39-Tumor-SM-4DK1Z   \n",
       "2016             MEL-IPI_Pat08-Tumor-SM-4DK14   \n",
       "2050             MEL-IPI_Pat37-Tumor-SM-53U3Y   \n",
       "2130             MEL-IPI_Pat90-Tumor-SM-4DK3G   \n",
       "2289             MEL-IPI_Pat15-Tumor-SM-4DK1B   \n",
       "2313            MEL-IPI_Pat119-Tumor-SM-7459N   \n",
       "2423            MEL-IPI_Pat117-Tumor-SM-5X2QU   \n",
       "2446            MEL-IPI_Pat128-Tumor-SM-5X2R6   \n",
       "2546             MEL-IPI_Pat49-Tumor-SM-4DK2A   \n",
       "2595            MEL-IPI_Pat129-Tumor-SM-5X2R7   \n",
       "2620             MEL-IPI_Pat25-Tumor-SM-4DK1L   \n",
       "2653            MEL-IPI_Pat118-Tumor-SM-5X2QV   \n",
       "2666             MEL-IPI_Pat14-Tumor-SM-53U3C   \n",
       "2680             MEL-IPI_Pat50-Tumor-SM-53U69   \n",
       "2842             MEL-IPI_Pat03-Tumor-SM-4DJZY   \n",
       "3027             MEL-IPI_Pat02-Tumor-SM-4DJZX   \n",
       "3232             MEL-IPI_Pat27-Tumor-SM-4DK1N   \n",
       "3288             MEL-IPI_Pat80-Tumor-SM-4DK36   \n",
       "3367             MEL-IPI_Pat19-Tumor-SM-4DK1F   \n",
       "3420            MEL-IPI_Pat123-Tumor-SM-5X2R1   \n",
       "3432             MEL-IPI_Pat28-Tumor-SM-4DK1O   \n",
       "3539             MEL-IPI_Pat44-Tumor-SM-4DK25   \n",
       "3549             MEL-IPI_Pat88-Tumor-SM-4DK3E   \n",
       "3610             MEL-IPI_Pat43-Tumor-SM-4DK24   \n",
       "3636             MEL-IPI_Pat86-Tumor-SM-4DK3C   \n",
       "4217                          KIRP-A4-8312-TP   \n",
       "4298                          KIRP-AL-3467-TP   \n",
       "4343                          KIRP-5P-A9K3-TP   \n",
       "\n",
       "                         normal_sample_barcode  \n",
       "70                                  SC_9065-NB  \n",
       "119          PROS01448-1115161-Normal-SM-6CAJL  \n",
       "137                             MO_1410-Normal  \n",
       "369   PROS12319B-SU2C-06115114-Normal-SM-4W2NS  \n",
       "424          PROS01448-6115250-Normal-SM-67ET3  \n",
       "442          PROS01448-1115244-Normal-SM-6CAJP  \n",
       "510          PROS01448-6115219-Normal-SM-67ESS  \n",
       "533                                 MO_1316-NB  \n",
       "564          PROS01448-6115227-Normal-SM-67ESU  \n",
       "699                             SC_9046-Normal  \n",
       "879                             MO_1012-Normal  \n",
       "1026                            MO_1040-Normal  \n",
       "1425                            TP_2032-Normal  \n",
       "1505                            SC_9011-Normal  \n",
       "1533                            MO_1096-Normal  \n",
       "1663                            SC_9019-Normal  \n",
       "1733             MEL-IPI_Pat38-Normal-SM-53U5L  \n",
       "1925             MEL-IPI_Pat39-Normal-SM-4NFV6  \n",
       "2016             MEL-IPI_Pat08-Normal-SM-4NFUA  \n",
       "2050             MEL-IPI_Pat37-Normal-SM-4NFV4  \n",
       "2130             MEL-IPI_Pat90-Normal-SM-4NFWM  \n",
       "2289             MEL-IPI_Pat15-Normal-SM-4NFUH  \n",
       "2313            MEL-IPI_Pat119-Normal-SM-7459Q  \n",
       "2423            MEL-IPI_Pat117-Normal-SM-5X2RC  \n",
       "2446            MEL-IPI_Pat128-Normal-SM-5VWHQ  \n",
       "2546             MEL-IPI_Pat49-Normal-SM-53U5W  \n",
       "2595            MEL-IPI_Pat129-Normal-SM-5X2RI  \n",
       "2620             MEL-IPI_Pat25-Normal-SM-4NFUR  \n",
       "2653            MEL-IPI_Pat118-Normal-SM-5X2RD  \n",
       "2666             MEL-IPI_Pat14-Normal-SM-4NFUG  \n",
       "2680             MEL-IPI_Pat50-Normal-SM-53U5X  \n",
       "2842             MEL-IPI_Pat03-Normal-SM-4NFU5  \n",
       "3027             MEL-IPI_Pat02-Normal-SM-4NFU4  \n",
       "3232             MEL-IPI_Pat27-Normal-SM-4NFUT  \n",
       "3288             MEL-IPI_Pat80-Normal-SM-53U53  \n",
       "3367             MEL-IPI_Pat19-Normal-SM-4NFUL  \n",
       "3420            MEL-IPI_Pat123-Normal-SM-5VWHL  \n",
       "3432             MEL-IPI_Pat28-Normal-SM-4NFUU  \n",
       "3539             MEL-IPI_Pat44-Normal-SM-4NFVB  \n",
       "3549             MEL-IPI_Pat88-Normal-SM-53U4C  \n",
       "3610             MEL-IPI_Pat43-Normal-SM-4NFVA  \n",
       "3636             MEL-IPI_Pat86-Normal-SM-53U59  \n",
       "4217                           KIRP-A4-8312-NB  \n",
       "4298                           KIRP-AL-3467-NB  \n",
       "4343                           KIRP-5P-A9K3-NB  \n",
       "\n",
       "[45 rows x 41 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handles = (glob.glob('2015-Robinson/data/*/*/*.actionable.txt') + \n",
    "           glob.glob('2015-VanAllen/data/*/*/*.actionable.txt') + \n",
    "           glob.glob('2016-TCGA/data/*/*/*.actionable.txt') + \n",
    "           glob.glob('2014-Perry/data/*/*/*.actionable.txt')\n",
    "          )\n",
    "tmp = pd.concat([pd.read_csv(handle, sep='\\t') for handle in handles], ignore_index=True)\n",
    "tmp = tmp[tmp['feature_type'].eq('Somatic Variant') & tmp['score_bin'].isin(['Putatively Actionable', 'Investigate Actionability - High', 'Investigate Actionability - Low']) & ~tmp['validation_total_coverage'].isnull()]\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(29, 41)\n"
     ]
    }
   ],
   "source": [
    "print(tmp[tmp['validation_detection_power'].ge(0.80)].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BRAF p.V600E (Missense)      11\n",
       "AR p.L702H (Missense)         6\n",
       "AR p.T878A (Missense)         5\n",
       "AKT1 p.E17K (Missense)        2\n",
       "NRAS p.Q61R (Missense)        2\n",
       "KRAS p.G12D (Missense)        1\n",
       "PIK3CA p.E545K (Missense)     1\n",
       "NRAS p.Q61L (Missense)        1\n",
       "Name: feature_display, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['validation_detection_power'].ge(0.80)]['feature_display'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extended features\n",
    "Features considered: Germline variants, fusions, mutational burden, mutational signatures, aneuploidy  \n",
    "Evidence considered: FDA-approved, clinical trials, or clinical evidence  \n",
    "Matches considered: First putatively actionable and then also investigate actionability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(557, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>KIRP-2Z-A9J2</td>\n",
       "      <td>MLH3.p.G370C.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Germline</td>\n",
       "      <td>MLH3 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>KIRP-2Z-A9J5</td>\n",
       "      <td>COSMIC Signature 3</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>KIRP-2Z-A9J7</td>\n",
       "      <td>COSMIC Signature 2</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>KIRP-2Z-A9J8</td>\n",
       "      <td>ATM.p.R1875*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>ATM Germline</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      patient_id         feature_string                almanac_bin phial_bin  \\\n",
       "4   KIRP-2K-A9WE     COSMIC Signature 4  Investigate Actionability       NaN   \n",
       "23  KIRP-2Z-A9J2  MLH3.p.G370C.Germline  Investigate Actionability       NaN   \n",
       "34  KIRP-2Z-A9J5     COSMIC Signature 3  Investigate Actionability       NaN   \n",
       "53  KIRP-2Z-A9J7     COSMIC Signature 2  Investigate Actionability       NaN   \n",
       "56  KIRP-2Z-A9J8  ATM.p.R1875*.Germline  Investigate Actionability       NaN   \n",
       "\n",
       "    sum cohort sensitive_predictive_implication  \\\n",
       "4     1   KIRP                      Inferential   \n",
       "23    1   KIRP                Clinical evidence   \n",
       "34    1   KIRP                      Inferential   \n",
       "53    1   KIRP                      Inferential   \n",
       "56    1   KIRP                      Preclinical   \n",
       "\n",
       "   resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "4                                NaN                               NaN   \n",
       "23                               NaN                               NaN   \n",
       "34                               NaN                               NaN   \n",
       "53                               NaN                               NaN   \n",
       "56                               NaN                 Clinical evidence   \n",
       "\n",
       "       clinvar          feature_type  feature_str_simple  \n",
       "4          NaN  Mutational signature  COSMIC Signature 4  \n",
       "23         NaN              Germline       MLH3 Germline  \n",
       "34         NaN  Mutational signature  COSMIC Signature 3  \n",
       "53         NaN  Mutational signature  COSMIC Signature 2  \n",
       "56  Pathogenic              Germline        ATM Germline  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx_f = df['feature_type'].isin(['Germline', 'Rearrangement', 'Aneuploidy', \n",
    "                                 'Tumor mutational burden', 'Mutational signature'])\n",
    "idx_a = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "\n",
    "tmp = df[idx_f & idx_a]\n",
    "print(tmp.shape)\n",
    "tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'By considering these feature types, MOAlmanac identified an additional 557 clinically relevant molecular features in 329 patients, resulting in 395 patients with at least one event associated with therapeutic sensitivity, resistance, or prognosis.'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_cr_total = n_almanac_mel + n_almanac_su2c + n_almanac_kirp + n_almanac_os\n",
    "\n",
    "f\"By considering these feature types, MOAlmanac identified an additional {tmp.shape[0]} clinically relevant \\\n",
    "molecular features in {tmp['patient_id'].drop_duplicates().shape[0]} patients, resulting in \\\n",
    "{n_cr_total} patients with at least one event associated with therapeutic sensitivity, resistance, or prognosis.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(557, 12)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(329,)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13, 12)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_germline = tmp[tmp['feature_type'].eq('Germline') & tmp['clinvar'].isin(['Pathogenic', 'Pathogenic/Likely pathogenic', 'Likely pathogenic'])]\n",
    "tmp_germline.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13,)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_germline['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>KIRP-2Z-A9J8</td>\n",
       "      <td>ATM.p.R1875*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>ATM Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>KIRP-SX-A7SM</td>\n",
       "      <td>MLH1..GermlineSplice_Site</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>MLH1 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3073</th>\n",
       "      <td>MO_1071</td>\n",
       "      <td>BRCA1.p.C61G.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA1 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3197</th>\n",
       "      <td>MO_1176</td>\n",
       "      <td>BRCA2.p.LK1491fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3266</th>\n",
       "      <td>MO_1221</td>\n",
       "      <td>ATM..GermlineSplice_Site</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>ATM Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3745</th>\n",
       "      <td>Osteosarcoma-SJDOSTEOS009</td>\n",
       "      <td>BRCA2.p.T2412fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3981</th>\n",
       "      <td>PROS01448-6115250</td>\n",
       "      <td>BRCA2.p.L698fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4280</th>\n",
       "      <td>SC_9022</td>\n",
       "      <td>BRCA2.p.S1982fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4355</th>\n",
       "      <td>SC_9030</td>\n",
       "      <td>ATM.p.Q1852*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>ATM Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4594</th>\n",
       "      <td>SC_9060</td>\n",
       "      <td>RAD51D.p.A122fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>Pathogenic/Likely pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>RAD51D Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4688</th>\n",
       "      <td>SC_9072</td>\n",
       "      <td>ATM.p.W2638*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>Pathogenic/Likely pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>ATM Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4810</th>\n",
       "      <td>SC_9090</td>\n",
       "      <td>BRCA2.p.S1955*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5040</th>\n",
       "      <td>TP_2054</td>\n",
       "      <td>BRCA2.p.S1882*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     patient_id             feature_string  \\\n",
       "56                 KIRP-2Z-A9J8      ATM.p.R1875*.Germline   \n",
       "808                KIRP-SX-A7SM  MLH1..GermlineSplice_Site   \n",
       "3073                    MO_1071      BRCA1.p.C61G.Germline   \n",
       "3197                    MO_1176  BRCA2.p.LK1491fs.Germline   \n",
       "3266                    MO_1221   ATM..GermlineSplice_Site   \n",
       "3745  Osteosarcoma-SJDOSTEOS009   BRCA2.p.T2412fs.Germline   \n",
       "3981          PROS01448-6115250    BRCA2.p.L698fs.Germline   \n",
       "4280                    SC_9022   BRCA2.p.S1982fs.Germline   \n",
       "4355                    SC_9030      ATM.p.Q1852*.Germline   \n",
       "4594                    SC_9060   RAD51D.p.A122fs.Germline   \n",
       "4688                    SC_9072      ATM.p.W2638*.Germline   \n",
       "4810                    SC_9090    BRCA2.p.S1955*.Germline   \n",
       "5040                    TP_2054    BRCA2.p.S1882*.Germline   \n",
       "\n",
       "                    almanac_bin phial_bin  sum cohort  \\\n",
       "56    Investigate Actionability       NaN    1   KIRP   \n",
       "808   Investigate Actionability       NaN    1   KIRP   \n",
       "3073  Investigate Actionability       NaN    1   SU2C   \n",
       "3197  Investigate Actionability       NaN    1   SU2C   \n",
       "3266  Investigate Actionability       NaN    1   SU2C   \n",
       "3745  Investigate Actionability       NaN    1     OS   \n",
       "3981  Investigate Actionability       NaN    1   SU2C   \n",
       "4280  Investigate Actionability       NaN    1   SU2C   \n",
       "4355  Investigate Actionability       NaN    1   SU2C   \n",
       "4594  Investigate Actionability       NaN    1   SU2C   \n",
       "4688  Investigate Actionability       NaN    1   SU2C   \n",
       "4810  Investigate Actionability       NaN    1   SU2C   \n",
       "5040  Investigate Actionability       NaN    1   SU2C   \n",
       "\n",
       "     sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "56                        Preclinical                               NaN   \n",
       "808                               NaN                               NaN   \n",
       "3073                     FDA-Approved                               NaN   \n",
       "3197                     FDA-Approved                               NaN   \n",
       "3266                      Preclinical                               NaN   \n",
       "3745                     FDA-Approved                               NaN   \n",
       "3981                     FDA-Approved                               NaN   \n",
       "4280                     FDA-Approved                               NaN   \n",
       "4355                      Preclinical                               NaN   \n",
       "4594                              NaN                               NaN   \n",
       "4688                      Preclinical                               NaN   \n",
       "4810                     FDA-Approved                               NaN   \n",
       "5040                     FDA-Approved                               NaN   \n",
       "\n",
       "     prognostic_predictive_implication                       clinvar  \\\n",
       "56                   Clinical evidence                    Pathogenic   \n",
       "808                          Guideline                    Pathogenic   \n",
       "3073                               NaN                    Pathogenic   \n",
       "3197                               NaN                    Pathogenic   \n",
       "3266                 Clinical evidence                    Pathogenic   \n",
       "3745                               NaN                    Pathogenic   \n",
       "3981                               NaN                    Pathogenic   \n",
       "4280                               NaN                    Pathogenic   \n",
       "4355                 Clinical evidence             Likely pathogenic   \n",
       "4594                       Inferential  Pathogenic/Likely pathogenic   \n",
       "4688                 Clinical evidence  Pathogenic/Likely pathogenic   \n",
       "4810                               NaN                    Pathogenic   \n",
       "5040                               NaN                    Pathogenic   \n",
       "\n",
       "     feature_type feature_str_simple  \n",
       "56       Germline       ATM Germline  \n",
       "808      Germline      MLH1 Germline  \n",
       "3073     Germline     BRCA1 Germline  \n",
       "3197     Germline     BRCA2 Germline  \n",
       "3266     Germline       ATM Germline  \n",
       "3745     Germline     BRCA2 Germline  \n",
       "3981     Germline     BRCA2 Germline  \n",
       "4280     Germline     BRCA2 Germline  \n",
       "4355     Germline       ATM Germline  \n",
       "4594     Germline    RAD51D Germline  \n",
       "4688     Germline       ATM Germline  \n",
       "4810     Germline     BRCA2 Germline  \n",
       "5040     Germline     BRCA2 Germline  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_germline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(137, 12)\n",
      "(91,)\n"
     ]
    }
   ],
   "source": [
    "tmp_rearrangement = tmp[tmp['feature_type'].eq('Rearrangement')]\n",
    "print(tmp_rearrangement.shape)\n",
    "print(tmp_rearrangement['patient_id'].drop_duplicates().shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erg_patients = tmp[tmp['feature_string'].eq('TMPRSS2--ERG')]['patient_id'].drop_duplicates()\n",
    "len(pd.Index(erg_patients).difference(pd.Index(pa_patients)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2932</th>\n",
       "      <td>MO_1008</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3053</th>\n",
       "      <td>MO_1015</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3066</th>\n",
       "      <td>MO_1040</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3081</th>\n",
       "      <td>MO_1071</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3117</th>\n",
       "      <td>MO_1095</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3142</th>\n",
       "      <td>MO_1114</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3150</th>\n",
       "      <td>MO_1118</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3161</th>\n",
       "      <td>MO_1124</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3193</th>\n",
       "      <td>MO_1162</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>MO_1176</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3230</th>\n",
       "      <td>MO_1192</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3245</th>\n",
       "      <td>MO_1202</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>MO_1241</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>MO_1244</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>MO_1249</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3339</th>\n",
       "      <td>MO_1263</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3348</th>\n",
       "      <td>MO_1277</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3362</th>\n",
       "      <td>MO_1316</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3381</th>\n",
       "      <td>MO_1337</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3898</th>\n",
       "      <td>PROS01448-1115183</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4011</th>\n",
       "      <td>PROS10448-1115154</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4118</th>\n",
       "      <td>SC_9003</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4188</th>\n",
       "      <td>SC_9009</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4223</th>\n",
       "      <td>SC_9016</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4241</th>\n",
       "      <td>SC_9018</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4260</th>\n",
       "      <td>SC_9020</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4285</th>\n",
       "      <td>SC_9022</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4399</th>\n",
       "      <td>SC_9035</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4416</th>\n",
       "      <td>SC_9037</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4455</th>\n",
       "      <td>SC_9043</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4466</th>\n",
       "      <td>SC_9046</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4508</th>\n",
       "      <td>SC_9049</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4548</th>\n",
       "      <td>SC_9056</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4583</th>\n",
       "      <td>SC_9059</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4596</th>\n",
       "      <td>SC_9060</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4611</th>\n",
       "      <td>SC_9061</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4638</th>\n",
       "      <td>SC_9063</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4660</th>\n",
       "      <td>SC_9066</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4792</th>\n",
       "      <td>SC_9086</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4839</th>\n",
       "      <td>SC_9092</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4910</th>\n",
       "      <td>SC_9097</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4959</th>\n",
       "      <td>SC_9104</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4969</th>\n",
       "      <td>SC_9107</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4984</th>\n",
       "      <td>SC_9109</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4997</th>\n",
       "      <td>TP_2001</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5048</th>\n",
       "      <td>TP_2054</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical evidence</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>TMPRSS2--ERG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient_id feature_string            almanac_bin phial_bin  sum  \\\n",
       "2932            MO_1008   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3053            MO_1015   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3066            MO_1040   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3081            MO_1071   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3117            MO_1095   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3142            MO_1114   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3150            MO_1118   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3161            MO_1124   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3193            MO_1162   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3201            MO_1176   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3230            MO_1192   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3245            MO_1202   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3298            MO_1241   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3311            MO_1244   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3319            MO_1249   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3339            MO_1263   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3348            MO_1277   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3362            MO_1316   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3381            MO_1337   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "3898  PROS01448-1115183   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4011  PROS10448-1115154   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4118            SC_9003   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4188            SC_9009   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4223            SC_9016   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4241            SC_9018   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4260            SC_9020   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4285            SC_9022   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4399            SC_9035   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4416            SC_9037   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4455            SC_9043   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4466            SC_9046   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4508            SC_9049   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4548            SC_9056   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4583            SC_9059   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4596            SC_9060   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4611            SC_9061   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4638            SC_9063   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4660            SC_9066   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4792            SC_9086   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4839            SC_9092   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4910            SC_9097   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4959            SC_9104   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4969            SC_9107   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4984            SC_9109   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "4997            TP_2001   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "5048            TP_2054   TMPRSS2--ERG  Putatively Actionable       NaN    1   \n",
       "\n",
       "     cohort sensitive_predictive_implication  \\\n",
       "2932   SU2C                      Preclinical   \n",
       "3053   SU2C                      Preclinical   \n",
       "3066   SU2C                      Preclinical   \n",
       "3081   SU2C                      Preclinical   \n",
       "3117   SU2C                      Preclinical   \n",
       "3142   SU2C                      Preclinical   \n",
       "3150   SU2C                      Preclinical   \n",
       "3161   SU2C                      Preclinical   \n",
       "3193   SU2C                      Preclinical   \n",
       "3201   SU2C                      Preclinical   \n",
       "3230   SU2C                      Preclinical   \n",
       "3245   SU2C                      Preclinical   \n",
       "3298   SU2C                      Preclinical   \n",
       "3311   SU2C                      Preclinical   \n",
       "3319   SU2C                      Preclinical   \n",
       "3339   SU2C                      Preclinical   \n",
       "3348   SU2C                      Preclinical   \n",
       "3362   SU2C                      Preclinical   \n",
       "3381   SU2C                      Preclinical   \n",
       "3898   SU2C                      Preclinical   \n",
       "4011   SU2C                      Preclinical   \n",
       "4118   SU2C                      Preclinical   \n",
       "4188   SU2C                      Preclinical   \n",
       "4223   SU2C                      Preclinical   \n",
       "4241   SU2C                      Preclinical   \n",
       "4260   SU2C                      Preclinical   \n",
       "4285   SU2C                      Preclinical   \n",
       "4399   SU2C                      Preclinical   \n",
       "4416   SU2C                      Preclinical   \n",
       "4455   SU2C                      Preclinical   \n",
       "4466   SU2C                      Preclinical   \n",
       "4508   SU2C                      Preclinical   \n",
       "4548   SU2C                      Preclinical   \n",
       "4583   SU2C                      Preclinical   \n",
       "4596   SU2C                      Preclinical   \n",
       "4611   SU2C                      Preclinical   \n",
       "4638   SU2C                      Preclinical   \n",
       "4660   SU2C                      Preclinical   \n",
       "4792   SU2C                      Preclinical   \n",
       "4839   SU2C                      Preclinical   \n",
       "4910   SU2C                      Preclinical   \n",
       "4959   SU2C                      Preclinical   \n",
       "4969   SU2C                      Preclinical   \n",
       "4984   SU2C                      Preclinical   \n",
       "4997   SU2C                      Preclinical   \n",
       "5048   SU2C                      Preclinical   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "2932                               NaN                 Clinical evidence   \n",
       "3053                               NaN                 Clinical evidence   \n",
       "3066                               NaN                 Clinical evidence   \n",
       "3081                               NaN                 Clinical evidence   \n",
       "3117                               NaN                 Clinical evidence   \n",
       "3142                               NaN                 Clinical evidence   \n",
       "3150                               NaN                 Clinical evidence   \n",
       "3161                               NaN                 Clinical evidence   \n",
       "3193                               NaN                 Clinical evidence   \n",
       "3201                               NaN                 Clinical evidence   \n",
       "3230                               NaN                 Clinical evidence   \n",
       "3245                               NaN                 Clinical evidence   \n",
       "3298                               NaN                 Clinical evidence   \n",
       "3311                               NaN                 Clinical evidence   \n",
       "3319                               NaN                 Clinical evidence   \n",
       "3339                               NaN                 Clinical evidence   \n",
       "3348                               NaN                 Clinical evidence   \n",
       "3362                               NaN                 Clinical evidence   \n",
       "3381                               NaN                 Clinical evidence   \n",
       "3898                               NaN                 Clinical evidence   \n",
       "4011                               NaN                 Clinical evidence   \n",
       "4118                               NaN                 Clinical evidence   \n",
       "4188                               NaN                 Clinical evidence   \n",
       "4223                               NaN                 Clinical evidence   \n",
       "4241                               NaN                 Clinical evidence   \n",
       "4260                               NaN                 Clinical evidence   \n",
       "4285                               NaN                 Clinical evidence   \n",
       "4399                               NaN                 Clinical evidence   \n",
       "4416                               NaN                 Clinical evidence   \n",
       "4455                               NaN                 Clinical evidence   \n",
       "4466                               NaN                 Clinical evidence   \n",
       "4508                               NaN                 Clinical evidence   \n",
       "4548                               NaN                 Clinical evidence   \n",
       "4583                               NaN                 Clinical evidence   \n",
       "4596                               NaN                 Clinical evidence   \n",
       "4611                               NaN                 Clinical evidence   \n",
       "4638                               NaN                 Clinical evidence   \n",
       "4660                               NaN                 Clinical evidence   \n",
       "4792                               NaN                 Clinical evidence   \n",
       "4839                               NaN                 Clinical evidence   \n",
       "4910                               NaN                 Clinical evidence   \n",
       "4959                               NaN                 Clinical evidence   \n",
       "4969                               NaN                 Clinical evidence   \n",
       "4984                               NaN                 Clinical evidence   \n",
       "4997                               NaN                 Clinical evidence   \n",
       "5048                               NaN                 Clinical evidence   \n",
       "\n",
       "     clinvar   feature_type feature_str_simple  \n",
       "2932     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3053     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3066     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3081     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3117     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3142     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3150     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3161     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3193     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3201     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3230     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3245     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3298     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3311     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3319     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3339     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3348     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3362     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3381     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "3898     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4011     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4118     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4188     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4223     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4241     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4260     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4285     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4399     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4416     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4455     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4466     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4508     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4548     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4583     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4596     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4611     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4638     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4660     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4792     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4839     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4910     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4959     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4969     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4984     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "4997     NaN  Rearrangement       TMPRSS2--ERG  \n",
       "5048     NaN  Rearrangement       TMPRSS2--ERG  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_string'].eq('TMPRSS2--ERG')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MEL 44\n",
      "SU2C 4\n"
     ]
    }
   ],
   "source": [
    "tmb = tmp[tmp['feature_string'].eq('High Mutational Burden')]\n",
    "for label, group in tmb.groupby('cohort'):\n",
    "    print(label, group['patient_id'].drop_duplicates().shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(116,)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_type'].eq('Mutational signature')]['patient_id'].drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "180"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_type'].eq('Aneuploidy')]['patient_id'].drop_duplicates().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.0\n",
      "count    329.000000\n",
      "mean       1.693009\n",
      "std        1.504053\n",
      "min        1.000000\n",
      "25%        1.000000\n",
      "50%        1.000000\n",
      "75%        2.000000\n",
      "max       23.000000\n",
      "Name: patient_id, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "# Range of additional considerations per patient\n",
    "print(tmp['patient_id'].value_counts().median())\n",
    "print(tmp['patient_id'].value_counts().describe())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.loc[tmp['cohort'].isin(['MEL', 'SU2C']), 'patient_id'].value_counts().median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    217.000000\n",
       "mean       1.834101\n",
       "std        1.776946\n",
       "min        1.000000\n",
       "25%        1.000000\n",
       "50%        1.000000\n",
       "75%        2.000000\n",
       "max       23.000000\n",
       "Name: patient_id, dtype: float64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.loc[tmp['cohort'].isin(['MEL', 'SU2C']), 'patient_id'].value_counts().describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAOAAAAEoCAYAAABB4f2MAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3XmcHHWd//HXmzHMBIIoVwIKhEscEkBIQITIfcrqTwVxAypgxEVM8IDlcOTeICDhUNAgBoMLO7qA/gRRgZgEdgSBBHU5AigQrpgQlCM3AT77x/fbSU3TM/2dTnfXTPfn+XjUI91V1VWfmfRnqup7ysxwzuVjrbwDcK6ZeQI6lyNPQOdy5AnoXI48AZ3LkSegcznyBHQuR56AzuXIE9C5HL0r7wCqZaONNrLhw4fnHYZzAMyePftlM9u43H4Nk4DDhw9n1qxZeYfhHACSnk3Zz29BncuRJ6BzOfIEdC5HSQkoaR9JH868P05Sl6RrJA2pXXjONbbUK+AVwDAASdsD1wD/C3wE+G5tQnOu8aUm4DbAw/H1EcBdZnYScALw8VoE5lwzSE1AA1ri6wOA38XX84ENqx2Uq7/Ozk5GjhxJS0sLI0eOpLOzM++QmkJqPeCDwFmS7gI+Cnw5rh9OSEI3gHV2dtLR0cGUKVMYM2YMXV1djBs3DoCxY8fmHF2DM7OyCzCS8Mz3GnBOZv1VwI0px6j1MmrUKHOVGTFihE2fPr3buunTp9uIESNyimjgA2ZZwvdWtgaDMklqA94ys5Vr/JdgDY0ePdq8JUxlWlpaWL58OYMGDVq1buXKlbS1tfHWW2/lGNnAJWm2mY0ut1+f6gEljZb0WUnrxlUthOdDN4C1t7fT1dXVbV1XVxft7e05RdQ8UusBh0q6H3gA+C9gaNx0GTCpRrG5Ouno6GDcuHHMmDGDlStXMmPGDMaNG0dHR0feoTW81EKYy1ld4vlcZv1NwPerHZSrr0JBy4QJE5gzZw7t7e1MnDjRC2DqIDUBDwAOMLNXJGXXPwVsUfWoXN2NHTvWEy4Hqc+Ag4E3SqzfGFhevXCcay6pCXgPcFzmvUlqAU4Hfl/toJxrFqm3oKcBd0vaDWglFLyMANYH9qpRbM41vKQroJk9BuwI3AfcCbQRCmB2MbOnaheec40teUgKM5sPnF3DWJxrOskJKGlT4CvADnHVY8BkM5tXi8CcawapFfEHEaocPgssjctRwN8kHVy78JxrbKlXwO8BPwa+ZpnGo5KuBK4EvM2ScxVIrYYYDlxl72y5fTWwZVUjcq6JpCbgLEIpaLEdgT9VLxznmkvqLegPgMslbQf8Ma7bg1Aoc4akXQs7mtlD1Q3RucaVmoA3xn8v7GUbdB+6wjlXRmoCblXTKJxrUqkJuNTMFtY0EueaUGohzIuSbpF0mIr6IznnKpeagIcDK4BbgOclXSBpm9qF5VxzSG2MfZeZHQ1sBnwHOAx4UtJ0ScfEwZmcc33Up0GZzOxVM7s6jvZ0MrAn8J/APEkX+TwRzvVNnybojA2yjwWOB94H/AyYQrgyngmMBg6scozONaykBJT0aeCLwMHAI4T2nzeY2euZfR4G/lyLIJ1rVKlXwJ8AncBHzGx2D/s8A0ysSlTONYnUZ8BNzezEXpIPM1tmZuf1tF3SmZIelPS6pIWSbpM0smgfSTpX0jxJyyTNlDQiMUbnBpzUBFwkaZPilZI2lJQ6dvm+hDalewL7A28C0yRtkNnnNOAUYAKwG/AScJek9RLP4dyAknoL2lPleyulhyt8BzM7pNsBpc8TJnvZC7gtVvB/HbjIzG6J+xxLSMKjCZOCOtdQek1ASd+MLw04UdLizOYWwlRlj1d47vUIV+BX4vutCLPw3lnYwcyWSbqHcNX0BHQNp9wVcEL8V8CXgOzt5hvAXODECs99JaHU9L74flj8d0HRfgsIVR7ONZxeE9DMtgKQNAP4tJm90tv+qSRdBowBxphZ8TNkca97lVjnXENIbYq2XxWT73JgLLC/mT2d2VSYaXdY0Uc24Z1XRecaQp+aoq2pOIjT0YTkK352fIaQhAdl9m8jPGfeW7cgnaujPjVFWxOSrgY+D3wSeEVS4Uq32MwWm5lJugLokPQ48CTwbWAxYU5C5xpO3RIQOCn+WzyZy3nAufH1JYSZmK4G3gvcDxxsZovqEaBz9Va3BDSzsh1547CH57I6IZ1raMnPgHGa6lMl/VDSRnHdXpJ8vBjnKpQ6NP0o4AngGGAc8O646SC8AbZzFUu9Al4KXGlmuxCGpii4A58f0LmKpSbgKOD6Euv/DgytXjjONZfUBFxGKJUs9kFCY2nnXAVSE/BXwDmSWuN7kzQcuJgwUppzrgKpCXgqsAGwEFgH6AL+BrxKqCx3zlUgqR4wjv0yRtL+wK6ExH3IzKbVMjjnGl3ZBJQ0iHDF+4KZTQem1zwq55pE2VtQM1tJ6CzrXYKcq7LUZ8DrgRNqGYhzzSi1Lei6wDGSDgJmA0uyG83s5GoH5lwzSE3AdqAw8+3WRdv81tS5CqWWgu5X60Cca0Z17RHvnOsudW6IW3vbbmafqE44zjWX1GfAfxS9HwTsDGwO/KKqETnXRFKfAY8vtV7SJMCHi3CuQmv6DHgNq8d6cc710Zom4PZVicK5JpVaCPO94lXApoS54q+rdlDONYvUQpgdi96/Teia9A08AZ2rmFfEO5ej1FHRris1SaakdSX5FdC5CqUWwhxLGLG62GDgC9ULx7nmUm6Czg0IBS4C3ivpzczmFuBwfOYi5ypW7hnwZUJvBwMeK7HdgHOqHZRzzaJcAu5HuPpNB44A/pnZ9gbwrJnNq1FszjW8cjPk3g0Q53943szerktUzjWJ1GqIZwEkbQZsAaxdtP2e6ofmXONLbQmzGWGSzL0Jz33F87a3VD805xpfajXEFcBbwA7AUsK00Z8B5gCH1iY05xpfalO0fYDDzexxSQYsNLM/SFoBXADcVbMInWtgqVfAwYQqCQgloZvE148BO1U7KOeaRWoCPk6YCQngz8CJkrYEvgq8WIvAnGsGqbegVwLD4uvzgd8BYwmTdR5bg7icawqp1RA3Zl4/FKcm+yDwnJm93NPnnHO9S70CriJpKKEQ5qGyOzvnepXaHWmQpEskLSI88w2P6y+W5GPCOFeh1EKYc4CPA58jPPcVPAAcl3oySXtLulXSi5JM0nFF26fG9dnlj6nHd26gSb0FHQt80czulpRtD/oI8IE+nG9I/MxP41LKNODzmfdv9OH4zg0oqQm4GfBsD59Pfo40s98Av4FwtethtxVmNj/1mM4NZKm3oI8S2oEWO4owXVk1jZH0kqQnJV0raZPyH3FuYEq9ep0H3CBpc0LD689I+iBwNKFXfLX8jjDU/TOEgp7/AKZLGmVmK3r7oHMDUWo94G2SjgK+RRiS8BzCfIEfN7Np1QrGzH6WefuwpNmEW9/D8TkoXAPqMQElnQ1camZLJW0B3Glmd9QvNDCzeZJeALar53mdq5fengHPJpRaQrgl3Lj24XQnaSPgfcDf631u5+qht1vQF4EjJd1O6ID7fkltpXY0s+dSTiZpCLBtfLsWsIWkDxF6WPwTOBe4hZBww4HvAC8Bv0w5vnMDTW8JOBG4Cvg+off7gyX2KfSMT+0RPxqYkXl/XlyuB75CGAL/C8B7CEk4AzjKzHwKNNeQekxAM/uRpP8mXIkeIvR8L56os0/MbCYhaXtyyJoc37mBptyoaK8Cf5Z0PHC3VwU4V12p1RDX1zoQ55rRmk7Q6ZxbA56AzuXIE9C5HHkCOpejck3RkpjZ+dUJx7nm0lsp6GeK3m8JrAMUZkPajDBK9lzCSGnOuT7qrSJ+x8LrWA/4BeDYQrOz2ED7J8CNpY/gnCsn9RnwbODr2Taf8fUp+ASdzlUsNQGHUnqO+DZgo+qF41xzSU3Au4BrJe0hqSUuewDX4BOzOFex1AT8EvA8cC+wPC5/IHRZOqE2oTnX+FLbgi4EPibpA4Qh6QXMMbMnaxmcc42uT0PTm9mTkl4jDE3v88U7t4Z8aHrnclTXoemdc93Ve2h651xG6hWwKkPTO+e6649D0zvXNPrb0PTONZWkK6CZ3Ua42h3M6qHpt6PKQ9M712z6MrXYHUBdh6Z3rtF5j3jnctRbj/i3CaNel2VmqSNjO+cyersFPYrVCTiU0Ov9l8B9cd1HgE/i/QGdq1hvPeJvLryWdCtwppldm9nlOkkPEJLwB7UL0bnGlfoMuD/dJ1UpmAHsW7VonGsyqQn4MnBkifVHAgurF45zzSW1GuJs4CeS9mP1M+AewIHAuFoE5lwzSO2Q+1NJTwAnA58gdMh9DNjLzO6vYXzONbS+VMTfDxxTw1icazp96skgaTNgE4qeHc3soWoG5VyzSEpASbsAN7B6PJisvkxR7ZzLSL0C/ogwKtoJhKHpk1rIOOd6l5qAOwC7+ChozlVXaj3gw8CwWgbiXDNKTcBvAZdIOlDSUEkbZJdaBuhcI0u9BS10ur2T7s9/wgthnKtYagLuV42TSdobOBUYRRjo6Xgzm5rZLkLvii8D7wXuB75qZo9W4/zO9TepLWHurtL5hhCGMvxpXIqdRpjy7DjgCUITuLskbW9mi6oUg3P9RnKPeEk7SrpK0m8lbRrXfTLWESYxs9+Y2bdiV6duQ9vHq9/XgYvM7BYzewQ4FliPMPiTcw0ndWj6g4EHgfcRuiYV5grchup1yN2KUNJ6Z2GFmS0D7gH2rNI5nOtXUq+AFwDfNLNPAW9k1s8Edq9SLIVqjgVF6xfgVSCuQaUm4AjgNyXW/xOodjVEcSsblVjnXENITcBXCLefxXYFXqhSLPPjv8VXu01451XRuYaQmoD/BXxX0vsJV6N3SdoHuJTSpZmVeIaQhAcVVkhqAz5KmJnXuYaTWg/4bWAqYYKWQmdcERJzYurJJA0Bto1v1wK2kPQh4J9m9pykK4AOSY8DT8bzLo7nca7hpNYDrgSOkXQW4bZzLeBPZvbXPp5vNN0HdzovLtcT6v4uIZSwXs3qiviDvQ7QNarU/oBnA5ea2dPA05n1g4F/N7PzU45jZjN5Z3/C7HYDzo2Lcw2vLzPkDimxfh18YN6G0NnZyciRI2lpaWHkyJF0dnbmHVJTSH0G7KkqYBdCVYQbwDo7O+no6GDKlCmMGTOGrq4uxo0Lg92NHTs25+ganJn1uACLgNeBtzKvC8uSuP7q3o5Rr2XUqFHmKjNixAibPn16t3XTp0+3ESNG5BTRwAfMsoTvrcK+pUk6lnD1u47QTvO1zOY3gLlmdl+pz9bb6NGjbdasWXmHMSC1tLSwfPlyBg0atGrdypUraWtr46233soxsoFL0mwzG11uv15vQc3s+niwZ4B7LZSGugbT3t5OV1cX++23utdZV1cX7e3tOUbVHFILYUaVSj5JbZJ+WOWYXJ11dHQwbtw4ZsyYwcqVK5kxYwbjxo2jo6Mj79AaXmohzJmSDgW+YGbzYdVQhZ0UdStyA0+hoGXChAnMmTOH9vZ2Jk6c6AUwddDrM+CqncKAvD8Fdgb+jTA//PmE1jFft9BtKFf+DOj6k6o8AxaY2TzgQEmTgJsIpZ+fNbNfrlmYzjW3vvSIPwI4ntAH8GXgG5K2qFFczjWF1B7xUwhD059nZgcAOxG6KP1Fkk/Y4lyFUq+AuwO7m9mVAGb2spn9P+BMYHKtgnOu0aWWgo42sxXFK81ssqTpVY7JuaaRdAUsJJ+k0ZI+K2nd+H5dMr0jnHN9k9odaShwK7AboVH2doTEuwxYDnytVgE618hSnwEvJwwXsSGwNLP+JuDgagflXLNIfQY8ADjAzF4J4+eu8hTgVRHOVSj1CjiY7uOBFmxMuAV1zlUgNQHvIYzZUmCSWoDTgd9XOyjnmkXqLehpwN2SdgNagUmEwXrXB/aqUWzONbzUaojHgB0J43PeCbQRCmB2MbOnaheec40t9QpI7IbkAzA5V0XJjbGdc9XnCehcjjwBnctRjwkoaW9Jyc+Izrm+6+0KOIM495+kpyVtWJ+QnGsevSXgK4RpowGGl9nXOVeB3m4xbyFUvv+d0ANilqSSo7Sa2da1CM65RtdbAp5I6IK0HaHb0U8Iw9M756qkxwSM49vfDiBpZ2CS+Tx9Dauzs5OJEyeuGhe0o6PDxwWtg9RhCY+HVVNGb0u4JX3KzLwnRAPw2ZFylDKDCyFRvwssI4wJ+nZ8fQkwKOUYtV58dqTK+exI1Ufi7Eip9XyXAGMJz4Vdcd1Hge8QSkdPrdYfBFd/c+bMYcyYMd3WjRkzhjlz5uQUUfNIrVo4GhhnZteb2VNxmQp8CfBxQQe4wuxIWT47Un2kJuD6hOEnij0FvKd64bg8+OxI+Um9Bf0LcDLw1aL1XwP+XNWIXN357Ej5SZ0daW/gN8A84D5CKehHgM2Aw8ysq5eP14XPjuT6k9TZkVJ7xN8DfIDQC34I8O74evv+kHzODVR96RE/D6jpQ4Gkc3lnr/sFZjaslud1Li/9sYH1E8CmmWXHfMNpDhMmTKCtrQ1JtLW1MWHChLxDagr9MQHfNLP5mWVh3gE1ugkTJjB58mQuvPBClixZwoUXXsjkyZM9Ceshpba+XgtwLmHo+xeBZ4CfAVunfNZbwlSutbXVJk2a1G3dpEmTrLW1NaeIBj4SW8IklYLWi6TDgPWAx4FNgG8DHwRGmNk/evusl4JWThJLlixhnXXWWbVu6dKlrLvuuvSn78dAUtVS0BIHHyzpQElbVvL5npjZb83sv83sf81sGvAvMcZjq3ke111rayuTJ3efZ3Xy5Mm0trbmFFHzSJ2ieqqkk+LrtYEHCAP0PhGvWjVhZouBRwl9El2NnHDCCZx++ulcdtllLF26lMsuu4zTTz+dE044Ie/QGl/KfSrwd2DX+PpIYC7hFvEM4P6UY1SyEEbg/jtwdrl9/RlwzYwfP95aW1sNsNbWVhs/fnzeIQ1oVPMZUNJyYFsze0HSj4HXzOwUScOBh81svWr8MZB0KXAb8FxM8LOAvYEdzezZ3j7rz4CuP6n2M+B8YGScEekQYFpcPwRYWVmIJb0f6CTUBf4CWAHsUS75nBuoUlvCXAf8nNAW9C1WT0n2YUKJZVWY2b9W61jODQSpQ1KcL+lRwmy4N5lZYbLON4GLaxWcq5+2tjZWrFix6n1rayvLl/uII7WWWgq6N/ArM7vczF7IbLqRMH6oG8AKyTd06FDmzJnD0KFDWbFiBW1tbXmH1vBSb0FnENplvlS0fv24raWaQbn6KiTf/PnzAZg/fz7Dhg1jwYIFOUfW+FILYUToA1hsQ2BJ9cJxeZk5c2av711t9HoFlHRrfGnADZJWZDa3ACMJs+a6AW7fffdddQUsvHe1V+4K+I+4iPCs94/M8gIwGfhcLQN0tdfa2sqCBQsYNmwYjz/++KrbT2+KVnu9XgFt9YC8c4FLzcxvNxvQ8uXLaWtrY8GCBatGQvNS0PpIrYY4r9aBuHx5suUjKQElbQBMBA4gNBHrdutqZu+ufmjONb7UaogpwC7AjwitYbyTmHNVkJqABwAHmdn9tQzGuWaTWg/4ErC4loE414xSE7ADOF/SkFoG41yzSb0F/TZhnviXJD1LURckM9upynG5GpPU58+k9B11fZOagDfXNApXdz0l0/AzbmfuRYfXOZrm5fWAzuWoPw7M61zTSK2IX0QvdX9eEe9cZVKfAccXvR9EqJg/gtBCxjlXgdRnwOtLrZf0EKGS/vvVDMq5ZrGmz4AzgI9XIxDnmtGaJuC/Ai9XIxDnmlFqIczDdC+EETAU2AD4Sg3icq4pVFoR/zawEJhpZlUbF9RV387n3clry/o2dvLwM25P3nf9wYP4yzkH9zUsF3lFfIN7bdnKmrZs6UuyundKniMeQNL+wA6E29FHzWxmLYJyrlmkPgO+D/glMIrQIRdgM0mzgE+Z2bweP+yc61FqKej3CHNCbGtmm5vZ5oQ5+96K25xzFUi9BT0I2NfMnimsMLOnJZ3M6olanHN9tKb1gG9XJQrnmlRqAv4e+J6kzQsrJG0BXIlfAZ2rWGoCngysAzwt6dk4UO9Tcd3JNYrNuYaXWg/4PLCrpIOADxJawjxmZtN6/6Rzrjd9qgc0s7uAu2oUi6uB9drPYMfrz6jh8QF8CItKlZsd6TDgh8DOZvZa0bb1gb8AXzazO2sXolsTDx/7cN4huF6UewYcD3y3OPkA4rqLga/VIjDnmkG5BNwJ6O05bzqwc/XCca65lEvAjem9rs8Is+Q65ypQLgFfIFwFe7IT8GL1wnGuuZRLwNuBCyQNLt4gaR3g/LhPVUk6SdIzkpZLmi3po9U+h3P9QblqiInAkcBfJX0fKHS+bScU0Ai4sJoBSfosoYXNSUBX/Pe3knYws+eqeS7n8lZuiuqXJO1JqIq4kJBwEJ797gBOMrMFVY7pm8BUM7s2vp8g6VDC0BdnVvlczuWqbEW8mT0LfEzSe4FtCUn4VzN7pdrBSFqb0Ofw0qJNdwJ7Vvt8zuUtuSVMTLgHaxgLwEZAC1B8VV0AHFjjcztXd31qilZHxcPgq8S6bmbPnv1ynDrNuf5gy5Sd+lsCvkzoZT+saP0mvPOq2I2ZbVyroJyrlX41O5KZvQHMJvTAzzoIuLf+ETlXW/3tCghwGfCfkh4A/gCcCGwGTM41KudqoN8loJn9XNKGhGmxNwUeAT4WS2Odayjyeb+dy0+/egZ0rtl4AjYASb+WNHUNjzFakkka3ss+60i6WdJr5fZ1aTwBi0iaGr9cxcsf+3CMmZKuqmWcOfkisDcwhvB8/nw1Dhp/57+uxrEGmn5XCNNPTAM+X7TujTwC6We2BeaYWb8d50LSIDPr23RQeTIzXzILMBX4dS/b9wFWEkYKL6w7EXgd2Dp+3oqW4XG/HQjdtxYBLwGdwLDicxOG+XgReAX4CbBOZp914n6LCY0TvhU/MzWzz9qE4UJeAJYQmhAeUvRzHEro3bIc+B/g6GysJX7umUU/08yUcxGaFk4BngGWAX8FTgPWitvPLfH72hcYHl+PLorDgCPj68I+YwmjMywDxsdtewJ3A0vj7/KHwLszx9kb+GP8Pb4G3A+MrPv3Le8vfH9byiVg3OdCwu3XBoRhGpcAx8Zt6xMaDVxHaNEzLH4JNyW09LmY0J1rJ+A24IHMl3Fq/DJcG/c5GHgVODNz7h/EL9QhwEjgJkLyZxPwxvjl2pvwR2E84Qq+c9y+eUy878f4j4oJ1FsCbhB/pnvjz7RB4rkGEfqN7hYT5qj4M42L24cAPyeMtlf4fa3dxwScS+g2txXwfmDHmFinEOYw+TBwH3Bz/Ny7CH/cLgW2ib+Do4F2T8D+kYBvxv/A7HJxZp9BhL/0vwAeAn5edIyZwFVF684Hfl+07r3xC7R75tzPA+/K7HMtMC3zZV0BHJPZPiR+oafG99sQhhHZouhc/x/4QXx9IfAksRoqrvt2bwkY97mKeOVLPVcPx7mo8DNlfu5fF+3TlwQ8pWifnwJTitZ9KO67CeGPiQH75P1982fA0u4Bvly07tXCCzNbKelo4FHCreT+CcccBewtaXGJbdsQroQQBjx+M7NtHuEveGG/tQl/zQuxLI5TiBfsShw4WVJmNa2E2zQIV9c/WvxmRvfRdynnQtKJwJcIDZQHE/6AVbNhxayi96OAbWPn7lVhxH+3MbP7YqnxHZJ+T5he4SYLA1DXlSdgaUvN7G9l9tmDUIr8HsLgVa/2vjtrEZ7/Ti2xLdvQvLgAwVhdWi3KWyt+ZrcSx1rWh+OkKHuumARXEH7uewm3y18FPlXm2IXBwFbFKmlQD/suKRHXj4HLS+z7IoCZHS/pCsKz8CeAiZI+aWZ3lImrqjwBKxDrv64ifJEOBW6UtGfmyvUG4bkv6yHC88+zVnkp3d8IX/Q9gKdjLOsSngWfivv8ifClHWZmM3o4zmPAEZKUuQruUUE8KecaA9xvZquqZSRtU7RPqd/Xwvjvppl1H0qM6yFgRLk/omb2F8Lg0hdL+i1wLGGkh7rxesDSWiUNK1o2BpDUAtwA3G1m1xBurd4PnJP5/Fxgd0nDJW0kaS3gakIBzc8lfVjS1pIOlPQjSeulBGVmiwklihdLOkjSCELBSEtmnycJBSNTJR0ZzzNa0qmSPh13m0x4frpC0vaSjiSU5PZJ4rmeJMwrcpik7SSdRShJzpoLjIyxbBSrEpYRCndOlzQiDo1SPFJCTy4m/P4nS9pF0raS/kXSNQCStpJ0kaQ9JW0paT9Codhjff0drLG8H0L720LpagQDXojbzwLmAxtnPnMQ4co0Jr7/AOGZaindqyG2A24mlMAtA54glESunTl3cWHEucAjmffrEgoZFhOeP8/indUQg+LnniZcXeYDtwKjMvscHs+/nNDr5Bj6WAiTci7CM+uU+DO/Gl+fDczNHGNjwrAji2IM+8b17TG2pcDDwEcpXQgzukSso4HfEW55l8TPnx+3DSUUoL1IKNR6DrgEGFTv75s3xnYuR34L6lyOPAGdy5EnoHM58gR0LkeegG6NSZorqVQDA1eGJ6BLJulcSY+U2LQboZF4Pc7VULwlTIPJoz+cmS0sv5crxa+AOYm95idLulLSK3H5bmw1U9hnbUkXS3pB0hJJD0o6JLN939hb/2OSHpD0BqGbUqnzmaTxkm6XtFTSs5I+V7TPRZKekLQs3lZeIqktbjuO0NpnRGaUgOPitm63oJLWjy18XpK0SNLdkkZnth8nabGkAyQ9En+2GZK2KneuhpN3y5NmXQhdlhbRvU/ea8A3M/uU62u3L6ElyMOEvoNbk2mhU3Q+A/4B/BuhpU4HocHz6Mw+ZwF7EVqYfIzQQuSCuG0woSnY46zutzc4bpsLnBpfizCt3O3A7oRe9BcQWqRsGvc5jtByaFrcZydCu9I7yp2r0ZbcA2jWJSZgqT55hSZvKf36Cgl4RML5DLi2aN004IZePnMi8LfM+3PJNIvLrM8m4P6EZnKDi/b5M3BafH1cjGf7zPZj4h+XtXo7V6Mt/gyYr1J98i6Q9G4S+9pFxf3helLc5+8+QptQAGKj7K8TrlpDCI28i3splDOKMGzGwqK42wh/VApWmNkTmffzCO1K3wP8s49GM1GCAAABi0lEQVTnHLA8AfuvlH59BcX94fpM0h7Az4DzgG8QGk5/gvQeCAVrEfo3lppW/PXM6zeLthX+EDVVuYQnYL4+XKJP3jwze11SSl+7vtqD0H0p+35OfL0X8KKZXVDYKGnLos+X6rdX7CFCb4O3zezpNYg15VwDnidgvjYj9Mn7AWEgoX8H/gNCXztJhb52pxC+2BsQnvueNrNfVHC+T0t6kPD8eSRwAKuHu3gSeJ+kYwi3pocQRhvLmgtsKWlXQgHNIjNbUbTPNEIXol9JOo3VBSmHEsaB+Z/EWFPONfDl/RDarAshCSYT+ti9SugvNwloyexTrq/dvoRbt40SzmeEUtTfEW5hnyOO5JbZ5zuEnuiLCf3lvhK+Iqu2t7K6P6MBx8X1c4mFMPH9esCVhJHW3iAMNPUzwngsEAphFhedu9vP0tO5Gm3x/oA5kTSTUMo3vk7nM+AzZnZzPc7n0jTVA69z/Y0noHM58ltQ53LkV0DncuQJ6FyOPAGdy5EnoHM58gR0LkeegM7l6P8Afw6RFncuIDYAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 216x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "\n",
    "fig = plt.figure(figsize = (3, 4), constrained_layout=True)\n",
    "ax = plt.subplot()\n",
    "plt.tick_params(axis = 'both', which = 'both',\n",
    "            top = False, bottom = False, right = False, left = False, \n",
    "            labeltop = False, labelbottom = True, labelright = False, labelleft = True)\n",
    "\n",
    "p = tmp['patient_id'].value_counts().plot.box()\n",
    "plt.ylabel('Counts of extended feature types', fontsize=14)\n",
    "\n",
    "ax.set_xticklabels(['Extended features\\nper patient'], fontsize = 14)\n",
    "ax.tick_params(axis='y', which='major', labelsize=14)\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Therapeutic sensitivity by evidence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4       KIRP-2K-A9WE\n",
       "23      KIRP-2Z-A9J2\n",
       "34      KIRP-2Z-A9J5\n",
       "53      KIRP-2Z-A9J7\n",
       "56      KIRP-2Z-A9J8\n",
       "            ...     \n",
       "5051         TP_2060\n",
       "5053         TP_2060\n",
       "5066         TP_2061\n",
       "5078         TP_2061\n",
       "5090         TP_2064\n",
       "Name: patient_id, Length: 557, dtype: object"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['patient_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda3/envs/moalmanac-paper/lib/python3.7/site-packages/ipykernel_launcher.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  del sys.path[0]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "FDA-Approved         70\n",
       "Clinical evidence    16\n",
       "Preclinical          10\n",
       "Guideline             9\n",
       "Inferential           1\n",
       "Clinical trial        1\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp = df[df['sensitive_predictive_implication'].notnull()]\n",
    "tmp['sensitive_predictive_implication'].value_counts()\n",
    "\n",
    "evidence_map = {\n",
    "    'FDA-Approved': 0,\n",
    "    'Guideline': 1,\n",
    "    'Clinical trial': 2,\n",
    "    'Clinical evidence': 3,\n",
    "    'Preclinical': 4,\n",
    "    'Inferential': 5\n",
    "}\n",
    "\n",
    "tmp['evidence_map'] = tmp['sensitive_predictive_implication'].replace(evidence_map)\n",
    "tmp = (tmp\n",
    " .loc[:, ['patient_id', 'sensitive_predictive_implication', 'evidence_map']]\n",
    " .sort_values('evidence_map')\n",
    " .drop_duplicates(subset='patient_id', keep='first')\n",
    ")\n",
    "\n",
    "idx_mel = tmp['patient_id'].str.contains('MEL-IPI')\n",
    "idx_kirp = tmp['patient_id'].str.contains('KIRP')\n",
    "idx_os = tmp['patient_id'].str.contains('Osteosarcoma')\n",
    "idx_prad = tmp.index.difference(idx_mel[idx_mel].index.union(idx_kirp[idx_kirp].index).union(idx_os[idx_os].index))\n",
    "tmp_mel_vcs = tmp[idx_mel]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_mel_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "79"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_mel_vcs['FDA-Approved'] + tmp_mel_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Guideline         74\n",
       "FDA-Approved      35\n",
       "Preclinical       28\n",
       "Clinical trial     4\n",
       "Inferential        1\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_su2c_vcs = tmp.loc[idx_prad, :]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_su2c_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "109"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_su2c_vcs['FDA-Approved'] + tmp_su2c_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Guideline            19\n",
       "Inferential          18\n",
       "FDA-Approved         17\n",
       "Preclinical          14\n",
       "Clinical evidence    11\n",
       "Clinical trial        4\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_kirp_vcs = tmp[idx_kirp]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_kirp_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_kirp_vcs['FDA-Approved'] + tmp_kirp_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Preclinical          16\n",
       "Inferential           8\n",
       "FDA-Approved          7\n",
       "Guideline             4\n",
       "Clinical evidence     3\n",
       "Clinical trial        2\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_os_vcs = tmp[idx_os]['sensitive_predictive_implication'].value_counts()\n",
    "tmp_os_vcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_os_vcs['FDA-Approved'] + tmp_os_vcs['Guideline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "235"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "tmp_mel_vcs['FDA-Approved'] + tmp_mel_vcs['Guideline'] +\n",
    "tmp_su2c_vcs['FDA-Approved'] + tmp_su2c_vcs['Guideline'] +\n",
    "tmp_kirp_vcs['FDA-Approved'] + tmp_kirp_vcs['Guideline'] + \n",
    "tmp_os_vcs['FDA-Approved'] + tmp_os_vcs['Guideline']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "419"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "110 + 150 + 100 + 59"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FDA-Approved         129\n",
       "Guideline            106\n",
       "Preclinical           68\n",
       "Clinical evidence     30\n",
       "Inferential           28\n",
       "Clinical trial        11\n",
       "Name: sensitive_predictive_implication, dtype: int64"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['sensitive_predictive_implication'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "patients_fda_guideline = tmp[tmp['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline'])]['patient_id'].tolist()\n",
    "patients_all = tmp['patient_id'].tolist()\n",
    "patients_fda_guideline_clinical = tmp[tmp['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence'])]['patient_id'].tolist()\n",
    "patients_preclinical = tmp[tmp['sensitive_predictive_implication'].isin(['Preclinical'])]['patient_id'].tolist()\n",
    "patients_inferential = tmp[tmp['sensitive_predictive_implication'].isin(['Inferential'])]['patient_id'].tolist()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "patients_all = tmp['patient_id'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/miniconda3/envs/moalmanac-paper/lib/python3.7/site-packages/ipykernel_launcher.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  del sys.path[0]\n"
     ]
    }
   ],
   "source": [
    "tmp = df[df['sensitive_predictive_implication'].notnull() & df['feature_type'].isin(['Somatic variant', 'Somatic copy number'])]\n",
    "tmp['sensitive_predictive_implication'].value_counts()\n",
    "\n",
    "evidence_map = {\n",
    "    'FDA-Approved': 0,\n",
    "    'Guideline': 1,\n",
    "    'Clinical trial': 2,\n",
    "    'Clinical evidence': 3,\n",
    "    'Preclinical': 4,\n",
    "    'Inferential': 5\n",
    "}\n",
    "\n",
    "tmp['evidence_map'] = tmp['sensitive_predictive_implication'].replace(evidence_map)\n",
    "tmp = (tmp\n",
    " .loc[:, ['patient_id', 'sensitive_predictive_implication', 'evidence_map']]\n",
    " .sort_values('evidence_map')\n",
    " .drop_duplicates(subset='patient_id', keep='first')\n",
    ")\n",
    "\n",
    "patients_fda_guideline_somatic = (tmp[\n",
    "    tmp['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline'])]['patient_id'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['KIRP-2Z-A9JJ', 'MEL-IPI_Pat06', 'Osteosarcoma-CCM201',\n",
      "       'Osteosarcoma-IAS513', 'Osteosarcoma-NCF101', 'Osteosarcoma-RCP466',\n",
      "       'Osteosarcoma-SJDOSTEOS009', 'Osteosarcoma-WABS378',\n",
      "       'PROS01448-1115244', 'PROS01448-6115250', 'SC_9034', 'SC_9066',\n",
      "       'SC_9086', 'TP_2032', 'TP_2054', 'TP_2060'],\n",
      "      dtype='object')\n",
      "(16,)\n"
     ]
    }
   ],
   "source": [
    "tmp_patients = pd.Index(patients_fda_guideline).difference(pd.Index(patients_fda_guideline_somatic))\n",
    "print(tmp_patients)\n",
    "print(tmp_patients.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>KIRP-2Z-A9JJ</td>\n",
       "      <td>EML4--ALK</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>EML4--ALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>MEL-IPI_Pat06</td>\n",
       "      <td>COL1A1--TGM2</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>MEL</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--TGM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3513</th>\n",
       "      <td>Osteosarcoma-CCM201</td>\n",
       "      <td>COL1A1--MMP9</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--MMP9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3549</th>\n",
       "      <td>Osteosarcoma-IAS513</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3650</th>\n",
       "      <td>Osteosarcoma-NCF101</td>\n",
       "      <td>BRCA2.p.F1460fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3671</th>\n",
       "      <td>Osteosarcoma-RCP466</td>\n",
       "      <td>COL1A1--USP6</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--USP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3745</th>\n",
       "      <td>Osteosarcoma-SJDOSTEOS009</td>\n",
       "      <td>BRCA2.p.T2412fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3850</th>\n",
       "      <td>Osteosarcoma-WABS378</td>\n",
       "      <td>ABL1--TTF1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>ABL1--TTF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3912</th>\n",
       "      <td>PROS01448-1115244</td>\n",
       "      <td>COL1A1--KLK3</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--KLK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3981</th>\n",
       "      <td>PROS01448-6115250</td>\n",
       "      <td>BRCA2.p.L698fs.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4385</th>\n",
       "      <td>SC_9034</td>\n",
       "      <td>COL1A1--CPE</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--CPE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4654</th>\n",
       "      <td>SC_9066</td>\n",
       "      <td>CDK5RAP2--COL1A1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>CDK5RAP2--COL1A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4655</th>\n",
       "      <td>SC_9066</td>\n",
       "      <td>COL1A1--C1R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--C1R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4786</th>\n",
       "      <td>SC_9086</td>\n",
       "      <td>COL1A1--LRP1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--LRP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4790</th>\n",
       "      <td>SC_9086</td>\n",
       "      <td>MT-RNR2--COL1A1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>MT-RNR2--COL1A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5024</th>\n",
       "      <td>TP_2032</td>\n",
       "      <td>COL1A1--MT-CO2</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--MT-CO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5040</th>\n",
       "      <td>TP_2054</td>\n",
       "      <td>BRCA2.p.S1882*.Germline</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Germline</td>\n",
       "      <td>BRCA2 Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5051</th>\n",
       "      <td>TP_2060</td>\n",
       "      <td>COL1A1--BCL2L2-PABPN1</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>COL1A1--BCL2L2-PABPN1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     patient_id            feature_string  \\\n",
       "117                KIRP-2Z-A9JJ                 EML4--ALK   \n",
       "871               MEL-IPI_Pat06              COL1A1--TGM2   \n",
       "3513        Osteosarcoma-CCM201              COL1A1--MMP9   \n",
       "3549        Osteosarcoma-IAS513                NTRK2--AUH   \n",
       "3650        Osteosarcoma-NCF101  BRCA2.p.F1460fs.Germline   \n",
       "3671        Osteosarcoma-RCP466              COL1A1--USP6   \n",
       "3745  Osteosarcoma-SJDOSTEOS009  BRCA2.p.T2412fs.Germline   \n",
       "3850       Osteosarcoma-WABS378                ABL1--TTF1   \n",
       "3912          PROS01448-1115244              COL1A1--KLK3   \n",
       "3981          PROS01448-6115250   BRCA2.p.L698fs.Germline   \n",
       "4385                    SC_9034               COL1A1--CPE   \n",
       "4654                    SC_9066          CDK5RAP2--COL1A1   \n",
       "4655                    SC_9066               COL1A1--C1R   \n",
       "4786                    SC_9086              COL1A1--LRP1   \n",
       "4790                    SC_9086           MT-RNR2--COL1A1   \n",
       "5024                    TP_2032            COL1A1--MT-CO2   \n",
       "5040                    TP_2054   BRCA2.p.S1882*.Germline   \n",
       "5051                    TP_2060     COL1A1--BCL2L2-PABPN1   \n",
       "\n",
       "                    almanac_bin phial_bin  sum cohort  \\\n",
       "117       Putatively Actionable       NaN    1   KIRP   \n",
       "871   Investigate Actionability       NaN    1    MEL   \n",
       "3513  Investigate Actionability       NaN    1     OS   \n",
       "3549  Investigate Actionability       NaN    1     OS   \n",
       "3650  Investigate Actionability       NaN    1     OS   \n",
       "3671  Investigate Actionability       NaN    1     OS   \n",
       "3745  Investigate Actionability       NaN    1     OS   \n",
       "3850  Investigate Actionability       NaN    1     OS   \n",
       "3912  Investigate Actionability       NaN    1   SU2C   \n",
       "3981  Investigate Actionability       NaN    1   SU2C   \n",
       "4385  Investigate Actionability       NaN    1   SU2C   \n",
       "4654  Investigate Actionability       NaN    1   SU2C   \n",
       "4655  Investigate Actionability       NaN    1   SU2C   \n",
       "4786  Investigate Actionability       NaN    1   SU2C   \n",
       "4790  Investigate Actionability       NaN    1   SU2C   \n",
       "5024  Investigate Actionability       NaN    1   SU2C   \n",
       "5040  Investigate Actionability       NaN    1   SU2C   \n",
       "5051  Investigate Actionability       NaN    1   SU2C   \n",
       "\n",
       "     sensitive_predictive_implication resistance_predictive_implication  \\\n",
       "117                         Guideline                               NaN   \n",
       "871                         Guideline                               NaN   \n",
       "3513                        Guideline                               NaN   \n",
       "3549                     FDA-Approved                               NaN   \n",
       "3650                     FDA-Approved                               NaN   \n",
       "3671                        Guideline                               NaN   \n",
       "3745                     FDA-Approved                               NaN   \n",
       "3850                     FDA-Approved                               NaN   \n",
       "3912                        Guideline                               NaN   \n",
       "3981                     FDA-Approved                               NaN   \n",
       "4385                        Guideline                               NaN   \n",
       "4654                        Guideline                               NaN   \n",
       "4655                        Guideline                               NaN   \n",
       "4786                        Guideline                               NaN   \n",
       "4790                        Guideline                               NaN   \n",
       "5024                        Guideline                               NaN   \n",
       "5040                     FDA-Approved                               NaN   \n",
       "5051                        Guideline                               NaN   \n",
       "\n",
       "     prognostic_predictive_implication     clinvar   feature_type  \\\n",
       "117                                NaN         NaN  Rearrangement   \n",
       "871                                NaN         NaN  Rearrangement   \n",
       "3513                               NaN         NaN  Rearrangement   \n",
       "3549                               NaN         NaN  Rearrangement   \n",
       "3650                               NaN         NaN       Germline   \n",
       "3671                               NaN         NaN  Rearrangement   \n",
       "3745                               NaN  Pathogenic       Germline   \n",
       "3850                               NaN         NaN  Rearrangement   \n",
       "3912                               NaN         NaN  Rearrangement   \n",
       "3981                               NaN  Pathogenic       Germline   \n",
       "4385                               NaN         NaN  Rearrangement   \n",
       "4654                               NaN         NaN  Rearrangement   \n",
       "4655                               NaN         NaN  Rearrangement   \n",
       "4786                               NaN         NaN  Rearrangement   \n",
       "4790                               NaN         NaN  Rearrangement   \n",
       "5024                               NaN         NaN  Rearrangement   \n",
       "5040                               NaN  Pathogenic       Germline   \n",
       "5051                               NaN         NaN  Rearrangement   \n",
       "\n",
       "         feature_str_simple  \n",
       "117               EML4--ALK  \n",
       "871            COL1A1--TGM2  \n",
       "3513           COL1A1--MMP9  \n",
       "3549             NTRK2--AUH  \n",
       "3650         BRCA2 Germline  \n",
       "3671           COL1A1--USP6  \n",
       "3745         BRCA2 Germline  \n",
       "3850             ABL1--TTF1  \n",
       "3912           COL1A1--KLK3  \n",
       "3981         BRCA2 Germline  \n",
       "4385            COL1A1--CPE  \n",
       "4654       CDK5RAP2--COL1A1  \n",
       "4655            COL1A1--C1R  \n",
       "4786           COL1A1--LRP1  \n",
       "4790        MT-RNR2--COL1A1  \n",
       "5024         COL1A1--MT-CO2  \n",
       "5040         BRCA2 Germline  \n",
       "5051  COL1A1--BCL2L2-PABPN1  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp = (df[df['patient_id'].isin(tmp_patients) \n",
    "          & df['sensitive_predictive_implication'].isin(['FDA-Approved', 'Guideline'])])\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "COL1A1--LRP1                1\n",
       "MT-RNR2--COL1A1             1\n",
       "BRCA2.p.F1460fs.Germline    1\n",
       "COL1A1--MMP9                1\n",
       "COL1A1--USP6                1\n",
       "CDK5RAP2--COL1A1            1\n",
       "COL1A1--C1R                 1\n",
       "COL1A1--BCL2L2-PABPN1       1\n",
       "BRCA2.p.S1882*.Germline     1\n",
       "EML4--ALK                   1\n",
       "BRCA2.p.T2412fs.Germline    1\n",
       "COL1A1--CPE                 1\n",
       "COL1A1--KLK3                1\n",
       "BRCA2.p.L698fs.Germline     1\n",
       "NTRK2--AUH                  1\n",
       "COL1A1--MT-CO2              1\n",
       "COL1A1--TGM2                1\n",
       "ABL1--TTF1                  1\n",
       "Name: feature_string, dtype: int64"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp['feature_string'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3549</th>\n",
       "      <td>Osteosarcoma-IAS513</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>OS</td>\n",
       "      <td>FDA-Approved</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rearrangement</td>\n",
       "      <td>NTRK2--AUH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               patient_id feature_string                almanac_bin phial_bin  \\\n",
       "3549  Osteosarcoma-IAS513     NTRK2--AUH  Investigate Actionability       NaN   \n",
       "\n",
       "      sum cohort sensitive_predictive_implication  \\\n",
       "3549    1     OS                     FDA-Approved   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "3549                               NaN                               NaN   \n",
       "\n",
       "     clinvar   feature_type feature_str_simple  \n",
       "3549     NaN  Rearrangement         NTRK2--AUH  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp[tmp['feature_string'].str.contains('NTRK')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(68,)"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(pd.Index(patients_preclinical)\n",
    " .difference(pd.Index(patients_fda_guideline_clinical))\n",
    " .shape\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(28,)"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(pd.Index(patients_inferential)\n",
    " .difference(pd.Index(patients_fda_guideline_clinical))\n",
    " .difference(pd.Index(patients_preclinical))\n",
    " .shape\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PTEN.Deletion          18\n",
       "CDKN2A.Deletion        12\n",
       "TMPRSS2--ERG           10\n",
       "AURKB.Amplification     4\n",
       "FGFR2.Deletion          3\n",
       "Name: feature_string, dtype: int64"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(df[df['patient_id'].isin(patients_preclinical) & df['sensitive_predictive_implication'].eq('Preclinical')]['feature_string']\n",
    " .value_counts()\n",
    " .head()\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table of sample counts by assertion type, evidence, and match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Somatic variant        17\n",
       "Germline                6\n",
       "Somatic copy number     3\n",
       "Name: feature_type, dtype: int64"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(df[(df['cohort'].eq('KIRP') &\n",
    " df['almanac_bin'].isin(['Investigate Actionability', 'Putatively Actionable']) &\n",
    " df['prognostic_predictive_implication'].isin(['Guideline'])\n",
    ")]\n",
    "#.shape\n",
    "#['patient_id'].value_counts()\n",
    "['feature_type'].value_counts()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       KIRP-2K-A9WE\n",
       "1       KIRP-2K-A9WE\n",
       "2       KIRP-2K-A9WE\n",
       "3       KIRP-2K-A9WE\n",
       "4       KIRP-2K-A9WE\n",
       "            ...     \n",
       "5086         TP_2064\n",
       "5087         TP_2064\n",
       "5088         TP_2064\n",
       "5089         TP_2064\n",
       "5090         TP_2064\n",
       "Name: patient_id, Length: 5091, dtype: object"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['patient_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "columns = df.columns[df.columns.str.contains('predictive_implication')]\n",
    "\n",
    "evidence_columns = ['FDA-Approved', 'Guideline', 'Clinical trial', 'Clinical evidence', 'Preclinical', 'Inferential']\n",
    "match_columns = ['Putatively Actionable', 'Investigate Actionability', 'Biologically Relevant']\n",
    "types_columns = ['Somatic variant', 'Somatic copy number', 'Germline', 'Rearrangement', 'Mutational signature', 'Tumor mutational burden', 'Aneuploidy']\n",
    "\n",
    "outname = 'retrospective.feature_types_by_evidence.xlsx'\n",
    "\n",
    "with pd.ExcelWriter(outname) as writer: \n",
    "    for label, group in df.groupby('cohort'):\n",
    "        for column in columns:\n",
    "            sheetname = '{} {} profile counts'.format(label, column.split('_')[0])\n",
    "            (\n",
    "                df[df[column].isin(evidence_columns) & df['cohort'].eq(label)]\n",
    "                .loc[:, ['patient_id', 'feature_type', column]]\n",
    "                .drop_duplicates()\n",
    "                .pivot_table(index='feature_type', columns=column, values='patient_id', aggfunc=lambda x: len(x.unique()))\n",
    "                .reindex(evidence_columns, axis=1)\n",
    "                .reindex(types_columns)\n",
    "                .fillna(0)\n",
    "                .astype(int)\n",
    "                .to_excel(writer, sheet_name=sheetname)\n",
    "            )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## In conclusion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP 85 0.85\n",
      "MEL 110 1.0\n",
      "OS 51 0.864\n",
      "SU2C 149 0.993\n"
     ]
    }
   ],
   "source": [
    "n_size = {\n",
    "    'KIRP': 100,\n",
    "    'MEL': 110,\n",
    "    'SU2C': 150,\n",
    "    'OS': 59\n",
    "}\n",
    "\n",
    "tmp = df[df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])]\n",
    "for label, group in tmp.groupby('cohort'):\n",
    "    print(label, \n",
    "          group['patient_id'].drop_duplicates().shape[0], \n",
    "          round(group['patient_id'].drop_duplicates().shape[0] / n_size[label], 3)\n",
    "         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP 27 0.27\n",
      "MEL 101 0.918\n",
      "OS 36 0.61\n",
      "SU2C 131 0.873\n"
     ]
    }
   ],
   "source": [
    "tmp = df[df['phial_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])]\n",
    "for label, group in tmp.groupby('cohort'):\n",
    "    print(label, \n",
    "          group['patient_id'].drop_duplicates().shape[0], \n",
    "          round(group['patient_id'].drop_duplicates().shape[0] / n_size[label], 3)\n",
    "         )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2 caption"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(365,)\n",
      "(295,)\n"
     ]
    }
   ],
   "source": [
    "idx_v = df['feature_type'].eq('Somatic variant')\n",
    "idx_cn = df['feature_type'].eq('Somatic copy number')\n",
    "idx_a = df['almanac_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "idx_p = df['phial_bin'].fillna('').isin(['Putatively Actionable', 'Investigate Actionability'])\n",
    "\n",
    "idx = (idx_a) & (idx_v | idx_cn)\n",
    "tmp = df[idx]['patient_id'].drop_duplicates()\n",
    "print(tmp.shape)\n",
    "\n",
    "idx = (idx_p) & (idx_v | idx_cn)\n",
    "tmp = df[idx]['patient_id'].drop_duplicates()\n",
    "print(tmp.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>KIRP-2Z-A9J3</td>\n",
       "      <td>FGFR1.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>FGFR1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>KIRP-2Z-A9JE</td>\n",
       "      <td>SMARCA4.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>SMARCA4 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>KIRP-2Z-A9JG</td>\n",
       "      <td>ABL1.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ABL1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>KIRP-2Z-A9JP</td>\n",
       "      <td>SMARCA4.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>SMARCA4 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>KIRP-2Z-A9JR</td>\n",
       "      <td>EWSR1.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>EWSR1 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4994</th>\n",
       "      <td>TP_2001</td>\n",
       "      <td>MITF.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MITF Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5010</th>\n",
       "      <td>TP_2010</td>\n",
       "      <td>TMPRSS2.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>TMPRSS2 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5016</th>\n",
       "      <td>TP_2020</td>\n",
       "      <td>AR.p.Q58L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>AR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5021</th>\n",
       "      <td>TP_2020</td>\n",
       "      <td>NOTCH2.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>NOTCH2 Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5069</th>\n",
       "      <td>TP_2061</td>\n",
       "      <td>MAP2K4.Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>MAP2K4 Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>225 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id    feature_string almanac_bin                  phial_bin  \\\n",
       "27    KIRP-2Z-A9J3    FGFR1.Deletion         NaN  Investigate Actionability   \n",
       "90    KIRP-2Z-A9JE  SMARCA4.Deletion         NaN  Investigate Actionability   \n",
       "91    KIRP-2Z-A9JG     ABL1.Deletion         NaN  Investigate Actionability   \n",
       "168   KIRP-2Z-A9JP  SMARCA4.Deletion         NaN  Investigate Actionability   \n",
       "189   KIRP-2Z-A9JR    EWSR1.Deletion         NaN  Investigate Actionability   \n",
       "...            ...               ...         ...                        ...   \n",
       "4994       TP_2001     MITF.Deletion         NaN  Investigate Actionability   \n",
       "5010       TP_2010  TMPRSS2.Deletion         NaN  Investigate Actionability   \n",
       "5016       TP_2020         AR.p.Q58L         NaN  Investigate Actionability   \n",
       "5021       TP_2020   NOTCH2.Deletion         NaN  Investigate Actionability   \n",
       "5069       TP_2061   MAP2K4.Deletion         NaN  Investigate Actionability   \n",
       "\n",
       "      sum cohort sensitive_predictive_implication  \\\n",
       "27      1   KIRP                              NaN   \n",
       "90      1   KIRP                              NaN   \n",
       "91      1   KIRP                              NaN   \n",
       "168     1   KIRP                              NaN   \n",
       "189     1   KIRP                              NaN   \n",
       "...   ...    ...                              ...   \n",
       "4994    1   SU2C                              NaN   \n",
       "5010    1   SU2C                              NaN   \n",
       "5016    1   SU2C                              NaN   \n",
       "5021    1   SU2C                              NaN   \n",
       "5069    1   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "27                                 NaN                               NaN   \n",
       "90                                 NaN                               NaN   \n",
       "91                                 NaN                               NaN   \n",
       "168                                NaN                               NaN   \n",
       "189                                NaN                               NaN   \n",
       "...                                ...                               ...   \n",
       "4994                               NaN                               NaN   \n",
       "5010                               NaN                               NaN   \n",
       "5016                               NaN                               NaN   \n",
       "5021                               NaN                               NaN   \n",
       "5069                               NaN                               NaN   \n",
       "\n",
       "     clinvar         feature_type feature_str_simple  \n",
       "27       NaN  Somatic copy number          FGFR1 Del  \n",
       "90       NaN  Somatic copy number        SMARCA4 Del  \n",
       "91       NaN  Somatic copy number           ABL1 Del  \n",
       "168      NaN  Somatic copy number        SMARCA4 Del  \n",
       "189      NaN  Somatic copy number          EWSR1 Del  \n",
       "...      ...                  ...                ...  \n",
       "4994     NaN  Somatic copy number           MITF Del  \n",
       "5010     NaN  Somatic copy number        TMPRSS2 Del  \n",
       "5016     NaN      Somatic variant                 AR  \n",
       "5021     NaN  Somatic copy number         NOTCH2 Del  \n",
       "5069     NaN  Somatic copy number         MAP2K4 Del  \n",
       "\n",
       "[225 rows x 12 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[idx & df['almanac_bin'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_patients_preclinical = (pd.Index(patients_preclinical)\n",
    "                               .difference(pd.Index(patients_fda_guideline_clinical))\n",
    "                              )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SU2C    28\n",
       "OS      16\n",
       "KIRP    14\n",
       "MEL     10\n",
       "Name: cohort, dtype: int64"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    df[df['patient_id'].isin(\n",
    "        pd.Index(patients_preclinical)\n",
    "        .difference(pd.Index(patients_fda_guideline_clinical))\n",
    "    )]\n",
    " .loc[:, ['patient_id', 'cohort']]\n",
    " .drop_duplicates()\n",
    " .loc[:, 'cohort']\n",
    " .value_counts()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>feature_string</th>\n",
       "      <th>almanac_bin</th>\n",
       "      <th>phial_bin</th>\n",
       "      <th>sum</th>\n",
       "      <th>cohort</th>\n",
       "      <th>sensitive_predictive_implication</th>\n",
       "      <th>resistance_predictive_implication</th>\n",
       "      <th>prognostic_predictive_implication</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>feature_str_simple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AR.p.QQQQQ73del.Germline</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Germline</td>\n",
       "      <td>AR Germline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>ASXL1.Amplification</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>ASXL1 Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>AURKA.Amplification</td>\n",
       "      <td>Putatively Actionable</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic copy number</td>\n",
       "      <td>AURKA Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIRP-2K-A9WE</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>KIRP</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5086</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>COSMIC Signature 1</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5087</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>COSMIC Signature 6</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutational signature</td>\n",
       "      <td>COSMIC Signature 6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5088</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>MPL.p.A134T</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Biologically Relevant</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>MPL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5089</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>TP53.p.H179R</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>2</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>Guideline</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic variant</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5090</th>\n",
       "      <td>TP_2064</td>\n",
       "      <td>Whole genome doubling</td>\n",
       "      <td>Investigate Actionability</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SU2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Inferential</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aneuploidy</td>\n",
       "      <td>WGD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5091 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient_id            feature_string                almanac_bin  \\\n",
       "0     KIRP-2K-A9WE  AR.p.QQQQQ73del.Germline      Biologically Relevant   \n",
       "1     KIRP-2K-A9WE       ASXL1.Amplification      Biologically Relevant   \n",
       "2     KIRP-2K-A9WE       AURKA.Amplification      Putatively Actionable   \n",
       "3     KIRP-2K-A9WE       COSMIC Signature 12      Biologically Relevant   \n",
       "4     KIRP-2K-A9WE        COSMIC Signature 4  Investigate Actionability   \n",
       "...            ...                       ...                        ...   \n",
       "5086       TP_2064        COSMIC Signature 1      Biologically Relevant   \n",
       "5087       TP_2064        COSMIC Signature 6      Biologically Relevant   \n",
       "5088       TP_2064               MPL.p.A134T  Investigate Actionability   \n",
       "5089       TP_2064              TP53.p.H179R  Investigate Actionability   \n",
       "5090       TP_2064     Whole genome doubling  Investigate Actionability   \n",
       "\n",
       "                      phial_bin  sum cohort sensitive_predictive_implication  \\\n",
       "0                           NaN    1   KIRP                              NaN   \n",
       "1                           NaN    1   KIRP                              NaN   \n",
       "2                           NaN    1   KIRP                      Preclinical   \n",
       "3                           NaN    1   KIRP                              NaN   \n",
       "4                           NaN    1   KIRP                      Inferential   \n",
       "...                         ...  ...    ...                              ...   \n",
       "5086                        NaN    1   SU2C                              NaN   \n",
       "5087                        NaN    1   SU2C                              NaN   \n",
       "5088      Biologically Relevant    2   SU2C                      Preclinical   \n",
       "5089  Investigate Actionability    2   SU2C                              NaN   \n",
       "5090                        NaN    1   SU2C                              NaN   \n",
       "\n",
       "     resistance_predictive_implication prognostic_predictive_implication  \\\n",
       "0                                  NaN                               NaN   \n",
       "1                                  NaN                               NaN   \n",
       "2                                  NaN                       Preclinical   \n",
       "3                                  NaN                               NaN   \n",
       "4                                  NaN                               NaN   \n",
       "...                                ...                               ...   \n",
       "5086                               NaN                               NaN   \n",
       "5087                               NaN                               NaN   \n",
       "5088                               NaN                               NaN   \n",
       "5089                         Guideline                         Guideline   \n",
       "5090                               NaN                       Inferential   \n",
       "\n",
       "     clinvar          feature_type   feature_str_simple  \n",
       "0        NaN              Germline          AR Germline  \n",
       "1        NaN   Somatic copy number            ASXL1 Amp  \n",
       "2        NaN   Somatic copy number            AURKA Amp  \n",
       "3        NaN  Mutational signature  COSMIC Signature 12  \n",
       "4        NaN  Mutational signature   COSMIC Signature 4  \n",
       "...      ...                   ...                  ...  \n",
       "5086     NaN  Mutational signature   COSMIC Signature 1  \n",
       "5087     NaN  Mutational signature   COSMIC Signature 6  \n",
       "5088     NaN       Somatic variant                  MPL  \n",
       "5089     NaN       Somatic variant                 TP53  \n",
       "5090     NaN            Aneuploidy                  WGD  \n",
       "\n",
       "[5091 rows x 12 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p ../../tables/supplementary-table-03\n",
    "\n",
    "outname = '../../tables/supplementary-table-03/supplementary-table-3.xlsx'\n",
    "with pd.ExcelWriter(outname) as writer:\n",
    "    df.drop('sum', axis=1).to_excel(writer, sheet_name='MOAlmanac & PHIAL calls', index=False)\n",
    "    counts = pd.read_csv('retrospective.counts-by-category.txt', sep='\\t')\n",
    "    counts.to_excel(writer, sheet_name='summarized counts', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "moalmanac-paper",
   "language": "python",
   "name": "moalmanac-paper"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
