{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We create a supplementary table that is an excel file with the following sheets:\n",
    "- target - annotated for updating to moalmanac\n",
    "- moalmanac - aneuploidy\n",
    "- moalmanac - copy_number\n",
    "- moalmanac - germline variant\n",
    "- moalmanac - knockdown\n",
    "- moalmanac - microsatellite stability\n",
    "- moalmanac - mutational burden\n",
    "- moalmanac - mutational signature\n",
    "- moalmanac - neoantigen burden\n",
    "- moalmanac - rearrangement\n",
    "- moalmanac - silencing\n",
    "- moalmanac - somatic variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import openpyxl\n",
    "import pandas as pd\n",
    "import glob \n",
    "\n",
    "options = {}\n",
    "options['strings_to_formulas'] = False\n",
    "options['strings_to_urls'] = False\n",
    "\n",
    "column_order = ['disease', \n",
    "                'context', \n",
    "                'oncotree_term', \n",
    "                'oncotree_code', \n",
    "                'therapy_name', \n",
    "                'therapy_strategy', \n",
    "                'therapy_type', \n",
    "                'therapy_sensitivity',\n",
    "                'therapy_resistance', \n",
    "                'favorable_prognosis', \n",
    "                'predictive_implication',\n",
    "                'description', \n",
    "                'preferred_assertion', \n",
    "                'source_type', \n",
    "                'citation', \n",
    "                'url', \n",
    "                'doi',\n",
    "                'pmid', \n",
    "                'nct', \n",
    "                'last_updated', \n",
    "                'adverse_event_risk', \n",
    "                'feature_type', \n",
    "                'gene', \n",
    "                'chromosome', \n",
    "                'start_position', \n",
    "                'end_position', \n",
    "                'reference_allele', \n",
    "                'alternate_allele', \n",
    "                'cdna_change', \n",
    "                'protein_change', \n",
    "                'variant_annotation', \n",
    "                'exon', \n",
    "                'rsid', \n",
    "                'pathogenic', \n",
    "                'direction', \n",
    "                'cytoband',\n",
    "                'gene1', \n",
    "                'gene2', \n",
    "                'rearrangement_type', \n",
    "                'classification', \n",
    "                'minimum_mutations', \n",
    "                'mutations_per_mb',\n",
    "                #'minimum_neoantigens',\n",
    "                'cosmic_signature_number',\n",
    "                'cosmic_signature_version',\n",
    "                'event', \n",
    "                'locus', \n",
    "                'status', \n",
    "                'technique'\n",
    "]\n",
    "\n",
    "dtype_map = {\n",
    "    'aneuploidy': 'Aneuploidy',\n",
    "    'copy_number': 'Copy Number',\n",
    "    'germline_variant': 'Germline variant',\n",
    "    'knockdown': 'Knockdown',\n",
    "    'microsatellite_stability': 'Microsatellite stability',\n",
    "    'mutational_burden': 'Mutational burden',\n",
    "    'mutational_signature': 'Mutational signature',\n",
    "    'neoantigen_burden': 'Neoantigen burden',\n",
    "    'rearrangement': 'Rearrangement',\n",
    "    'silencing': 'Silencing',\n",
    "    'somatic_variant': 'Somatic variant'\n",
    "}\n",
    "\n",
    "sheets = []\n",
    "handles = glob.glob('../moalmanac-db/content/*')\n",
    "for handle in sorted(handles):\n",
    "    dtype = handle.split('/')[-1].split('.tsv')[0]\n",
    "    feature_type = dtype_map[dtype]  \n",
    "    tmp = pd.read_csv(handle, sep='\\t')\n",
    "    tmp['feature_type'] = feature_type\n",
    "    sheets.append(tmp)\n",
    "\n",
    "\n",
    "sort_columns = ['gene', 'gene1', 'gene2', 'therapy_sensitivity', 'therapy_resistance', 'favorable_prognosis', 'disease', 'therapy_name']\n",
    "df = (pd\n",
    "      .concat(sheets, ignore_index=True)\n",
    "      .loc[:, column_order]\n",
    "      .sort_values(sort_columns, ascending=True)\n",
    "     )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "filename = 'supplementary-table-2'\n",
    "folder = f'../../../tables/supplementary-table-02'\n",
    "!mkdir -p $folder\n",
    "outname = f'{folder}/{filename}.xlsx'\n",
    "\n",
    "writer = pd.ExcelWriter(outname)\n",
    "\n",
    "target = pd.read_excel('../target/almanac-comparison.xlsx')\n",
    "target.to_excel(writer, sheet_name='TableS2-TARGET', index=False)\n",
    "df.to_excel(writer, sheet_name='TableS2-MOAlmanac', index=False)\n",
    "\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "moalmanac-paper",
   "language": "python",
   "name": "moalmanac-paper"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
